// @wix/auto_sdk_ecom_inventory/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  DecrementNotPossibleWixError: () => DecrementNotPossibleWixError,
  IdentityType: () => IdentityType,
  IncrementNotPossibleWixError: () => IncrementNotPossibleWixError,
  Reason: () => Reason,
  provideHandlers: () => provideHandlers2,
  publicProvideHandlers: () => provideHandlers
});
module.exports = __toCommonJS(index_exports);

// src/service-plugins-types.ts
var Reason = /* @__PURE__ */ ((Reason2) => {
  Reason2["UNKNOWN_REASON"] = "UNKNOWN_REASON";
  Reason2["ORDER_PLACED"] = "ORDER_PLACED";
  Reason2["ORDER_PAID"] = "ORDER_PAID";
  Reason2["ORDER_CANCELED"] = "ORDER_CANCELED";
  Reason2["ORDER_REFUNDED"] = "ORDER_REFUNDED";
  Reason2["ORDER_EDITED"] = "ORDER_EDITED";
  Reason2["ORDER_REJECTED"] = "ORDER_REJECTED";
  return Reason2;
})(Reason || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});

// src/interfaces-ecommerce-v1-inventory-provider.public.ts
var import_sdk_types = require("@wix/sdk-types");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");
var provideHandlers = (0, import_sdk_types.ServicePluginDefinition)("ECOM_INVENTORY", [
  {
    name: "decrementAvailability",
    primaryHttpMappingPath: "/v1/decrement",
    transformations: {
      toREST: (payload) => {
        const toRestResponse = payload;
        return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
      },
      fromREST: (payload) => {
        const fromRestRequest = payload;
        return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
      }
    }
  },
  {
    name: "incrementAvailability",
    primaryHttpMappingPath: "/v1/increment",
    transformations: {
      toREST: (payload) => {
        const toRestResponse = payload;
        return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
      },
      fromREST: (payload) => {
        const fromRestRequest = payload;
        return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
      }
    }
  }
]);

// src/interfaces-ecommerce-v1-inventory-provider.context.ts
var import_service_plugin_modules = require("@wix/sdk-runtime/service-plugin-modules");
var provideHandlers2 = (0, import_service_plugin_modules.createServicePluginModule)(
  provideHandlers
);

// src/service-plugins-error-classes.ts
var DecrementNotPossibleWixError = class extends Error {
  constructor() {
    super("DecrementNotPossible");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "DECREMENT_NOT_POSSIBLE";
    this.name = "DecrementNotPossible";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "DecrementNotPossible",
      applicationCode: "DECREMENT_NOT_POSSIBLE"
    };
  }
};
/** @hidden */
DecrementNotPossibleWixError.__type = "wix_spi_error";
var IncrementNotPossibleWixError = class extends Error {
  constructor() {
    super("IncrementNotPossible");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "INCREMENT_NOT_POSSIBLE";
    this.name = "IncrementNotPossible";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "IncrementNotPossible",
      applicationCode: "INCREMENT_NOT_POSSIBLE"
    };
  }
};
/** @hidden */
IncrementNotPossibleWixError.__type = "wix_spi_error";
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DecrementNotPossibleWixError,
  IdentityType,
  IncrementNotPossibleWixError,
  Reason,
  provideHandlers,
  publicProvideHandlers
});
