// @wix/auto_sdk_ecom_subscription-contracts/build/internal/es/meta.mjs
// src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_subscription-contracts";
function getSubscriptionContract(payload) {
  function __getSubscriptionContract({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.subscription_contracts.v1.subscription_contract",
      method: "GET",
      methodFqn: "wix.ecom.subscription_contracts.v1.SubscriptionContracts.GetSubscriptionContract",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
        protoPath: "/v1/subscription-contracts/{subscriptionContractId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "subscriptionContract.createdDate" },
            { path: "subscriptionContract.updatedDate" },
            { path: "subscriptionContract.subscriptionSettings.startDate" },
            {
              path: "subscriptionContract.lineItems.image.urlExpirationDate"
            },
            {
              path: "subscriptionContract.lineItems.digitalFile.expirationDate"
            },
            {
              path: "subscriptionContract.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "subscriptionContract.lineItems.image.focalPoint.x" },
            { path: "subscriptionContract.lineItems.image.focalPoint.y" },
            {
              path: "subscriptionContract.lineItems.physicalProperties.weight"
            },
            {
              path: "subscriptionContract.billingInfo.address.geocode.latitude"
            },
            {
              path: "subscriptionContract.billingInfo.address.geocode.longitude"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "subscriptionContract.recipientInfo.address.geocode.latitude"
            },
            {
              path: "subscriptionContract.recipientInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSubscriptionContract;
}
function querySubscriptionContracts(payload) {
  function __querySubscriptionContracts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.subscription_contracts.v1.subscription_contract",
      method: "GET",
      methodFqn: "wix.ecom.subscription_contracts.v1.SubscriptionContracts.QuerySubscriptionContracts",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
        protoPath: "/v1/subscription-contracts/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "subscriptionContracts.createdDate" },
            { path: "subscriptionContracts.updatedDate" },
            { path: "subscriptionContracts.subscriptionSettings.startDate" },
            {
              path: "subscriptionContracts.lineItems.image.urlExpirationDate"
            },
            {
              path: "subscriptionContracts.lineItems.digitalFile.expirationDate"
            },
            {
              path: "subscriptionContracts.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "subscriptionContracts.lineItems.image.focalPoint.x" },
            { path: "subscriptionContracts.lineItems.image.focalPoint.y" },
            {
              path: "subscriptionContracts.lineItems.physicalProperties.weight"
            },
            {
              path: "subscriptionContracts.billingInfo.address.geocode.latitude"
            },
            {
              path: "subscriptionContracts.billingInfo.address.geocode.longitude"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "subscriptionContracts.recipientInfo.address.geocode.latitude"
            },
            {
              path: "subscriptionContracts.recipientInfo.address.geocode.longitude"
            }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
            protoPath: "/v1/subscription-contracts/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __querySubscriptionContracts;
}

// src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.meta.ts
function getSubscriptionContract2() {
  const payload = { subscriptionContractId: ":subscriptionContractId" };
  const getRequestOptions = getSubscriptionContract(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/subscription-contracts/{subscriptionContractId}",
    pathParams: { subscriptionContractId: "subscriptionContractId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function querySubscriptionContracts2() {
  const payload = {};
  const getRequestOptions = querySubscriptionContracts(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/subscription-contracts/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getSubscriptionContract2 as getSubscriptionContract,
  querySubscriptionContracts2 as querySubscriptionContracts
};
