// @wix/auto_sdk_ecom_draft-orders/build/internal/es/meta.mjs
// src/ecom-v1-draft-order-draft-orders.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixEcomOrdersDraftV1DraftOrdersUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/draft-orders",
        destPath: "/v1/draft-orders"
      },
      {
        srcPath: "/_api/v1/draft-orders",
        destPath: "/v1/draft-orders"
      },
      {
        srcPath: "/v1/draft-orders",
        destPath: "/v1/draft-orders"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/draft-orders",
        destPath: "/v1/draft-orders"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_draft-orders";
function createDraftOrder(payload) {
  function __createDraftOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.CreateDraftOrder",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDraftOrder;
}
function createEmptyDraftOrder(payload) {
  function __createEmptyDraftOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.CreateEmptyDraftOrder",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/create-empty-draft-order",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createEmptyDraftOrder;
}
function addLineItemsToDraftOrder(payload) {
  function __addLineItemsToDraftOrder({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "customLineItems.media.urlExpirationDate" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "customLineItems.media.focalPoint.x" },
          { path: "customLineItems.media.focalPoint.y" },
          { path: "customLineItems.physicalProperties.weight" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.AddLineItemsToDraftOrder",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/add-line-items",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addLineItemsToDraftOrder;
}
function updateLineItems(payload) {
  function __updateLineItems({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.UpdateLineItems",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/line-items",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateLineItems;
}
function setDiscounts(payload) {
  function __setDiscounts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.SetDiscounts",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/set-discounts",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setDiscounts;
}
function createCustomDiscounts(payload) {
  function __createCustomDiscounts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.CreateCustomDiscounts",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/create-custom-discounts",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCustomDiscounts;
}
function deleteCustomDiscounts(payload) {
  function __deleteCustomDiscounts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.DeleteCustomDiscounts",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/delete-custom-discounts",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteCustomDiscounts;
}
function setAdditionalFees(payload) {
  function __setAdditionalFees({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.SetAdditionalFees",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/set-additional-fees",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setAdditionalFees;
}
function createCustomAdditionalFees(payload) {
  function __createCustomAdditionalFees({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.CreateCustomAdditionalFees",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/create-additional-fees",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCustomAdditionalFees;
}
function deleteCustomAdditionalFees(payload) {
  function __deleteCustomAdditionalFees({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.DeleteCustomAdditionalFees",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/delete-custom-additional-fees",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteCustomAdditionalFees;
}
function setShippingInfo(payload) {
  function __setShippingInfo({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "shippingInfo.logistics.deliverByDate" },
          { path: "shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "shippingInfo.logistics.deliveryTimeSlot.to" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "shippingInfo.logistics.shippingDestination.address.geocode.latitude"
          },
          {
            path: "shippingInfo.logistics.shippingDestination.address.geocode.longitude"
          },
          {
            path: "shippingInfo.logistics.pickupDetails.address.geocode.latitude"
          },
          {
            path: "shippingInfo.logistics.pickupDetails.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "PUT",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.SetShippingInfo",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/set-shipping-info",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setShippingInfo;
}
function setBuyerInfo(payload) {
  function __setBuyerInfo({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "PUT",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.SetBuyerInfo",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/set-buyer-info",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setBuyerInfo;
}
function setRecipientInfo(payload) {
  function __setRecipientInfo({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "recipientInfo.address.geocode.latitude" },
          { path: "recipientInfo.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "PUT",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.SetRecipientInfo",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/set-recipient-info",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setRecipientInfo;
}
function setBillingInfo(payload) {
  function __setBillingInfo({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "billingInfo.address.geocode.latitude" },
          { path: "billingInfo.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "PUT",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.SetBillingInfo",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/set-billing-info",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setBillingInfo;
}
function getDraftOrder(payload) {
  function __getDraftOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "GET",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.GetDraftOrder",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
            protoPath: "/v1/draft-orders/{draftOrderId}",
            data: payload,
            host
          }),
          params: toURLSearchParams(payload)
        }
      ]
    };
    return metadata;
  }
  return __getDraftOrder;
}
function getOrderDraftabilityStatus(payload) {
  function __getOrderDraftabilityStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "GET",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.GetOrderDraftabilityStatus",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{orderId}/draftability",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getOrderDraftabilityStatus;
}
function commitDraftOrder(payload) {
  function __commitDraftOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.CommitDraftOrder",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/commit",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "committedDraftOrder.createdDate" },
            { path: "committedDraftOrder.updatedDate" },
            {
              path: "committedDraftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "committedDraftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "committedDraftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            { path: "orderAfterCommit.createdDate" },
            { path: "orderAfterCommit.updatedDate" },
            { path: "orderAfterCommit.purchasedDate" },
            { path: "orderAfterCommit.lineItems.image.urlExpirationDate" },
            { path: "orderAfterCommit.lineItems.digitalFile.expirationDate" },
            {
              path: "orderAfterCommit.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "orderAfterCommit.shippingInfo.logistics.deliverByDate" },
            {
              path: "orderAfterCommit.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "orderAfterCommit.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            { path: "orderAfterCommit.activities.createdDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "committedDraftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "committedDraftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            { path: "orderAfterCommit.lineItems.image.focalPoint.x" },
            { path: "orderAfterCommit.lineItems.image.focalPoint.y" },
            {
              path: "committedDraftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            { path: "orderAfterCommit.lineItems.physicalProperties.weight" },
            { path: "orderAfterCommit.billingInfo.address.geocode.latitude" },
            {
              path: "orderAfterCommit.billingInfo.address.geocode.longitude"
            },
            {
              path: "orderAfterCommit.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "orderAfterCommit.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "orderAfterCommit.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "orderAfterCommit.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "orderAfterCommit.recipientInfo.address.geocode.latitude"
            },
            {
              path: "orderAfterCommit.recipientInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __commitDraftOrder;
}
function createOrderFromDraft(payload) {
  function __createOrderFromDraft({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.CreateOrderFromDraft",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}/create-order",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "committedDraftOrder.createdDate" },
            { path: "committedDraftOrder.updatedDate" },
            {
              path: "committedDraftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "committedDraftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "committedDraftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            { path: "createdOrder.createdDate" },
            { path: "createdOrder.updatedDate" },
            { path: "createdOrder.purchasedDate" },
            { path: "createdOrder.lineItems.image.urlExpirationDate" },
            { path: "createdOrder.lineItems.digitalFile.expirationDate" },
            {
              path: "createdOrder.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "createdOrder.shippingInfo.logistics.deliverByDate" },
            {
              path: "createdOrder.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "createdOrder.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            { path: "createdOrder.activities.createdDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "committedDraftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "committedDraftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            { path: "createdOrder.lineItems.image.focalPoint.x" },
            { path: "createdOrder.lineItems.image.focalPoint.y" },
            {
              path: "committedDraftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "committedDraftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "committedDraftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            { path: "createdOrder.lineItems.physicalProperties.weight" },
            { path: "createdOrder.billingInfo.address.geocode.latitude" },
            { path: "createdOrder.billingInfo.address.geocode.longitude" },
            {
              path: "createdOrder.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "createdOrder.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "createdOrder.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "createdOrder.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "createdOrder.recipientInfo.address.geocode.latitude" },
            { path: "createdOrder.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOrderFromDraft;
}
function deleteDraftOrder(payload) {
  function __deleteDraftOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "DELETE",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.DeleteDraftOrder",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{draftOrderId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteDraftOrder;
}
function queryDraftOrders(payload) {
  function __queryDraftOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.QueryDraftOrders",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftOrders.createdDate" },
            { path: "draftOrders.updatedDate" },
            {
              path: "draftOrders.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "draftOrders.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "draftOrders.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "draftOrders.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "draftOrders.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "draftOrders.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "draftOrders.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "draftOrders.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "draftOrders.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftOrders.lineItems.lineItem.image.focalPoint.x" },
            { path: "draftOrders.lineItems.lineItem.image.focalPoint.y" },
            {
              path: "draftOrders.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "draftOrders.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "draftOrders.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "draftOrders.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "draftOrders.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "draftOrders.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "draftOrders.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "draftOrders.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "draftOrders.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "draftOrders.billingInfo.address.geocode.latitude" },
            { path: "draftOrders.billingInfo.address.geocode.longitude" },
            {
              path: "draftOrders.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "draftOrders.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "draftOrders.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "draftOrders.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "draftOrders.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "draftOrders.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "draftOrders.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "draftOrders.billingDetails.billingInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryDraftOrders;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftOrder.createdDate" },
            { path: "draftOrder.updatedDate" },
            { path: "draftOrder.lineItems.lineItem.image.urlExpirationDate" },
            {
              path: "draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftOrder.lineItems.lineItem.image.focalPoint.x" },
            { path: "draftOrder.lineItems.lineItem.image.focalPoint.y" },
            {
              path: "draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "draftOrder.billingInfo.address.geocode.latitude" },
            { path: "draftOrder.billingInfo.address.geocode.longitude" },
            {
              path: "draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function setTaxExemption(payload) {
  function __setTaxExemption({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "PUT",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.SetTaxExemption",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/set-tax-exemption",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setTaxExemption;
}
function setBusinessLocation(payload) {
  function __setBusinessLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.draft_order",
      method: "PUT",
      methodFqn: "com.wix.ecom.orders.draft.v1.DraftOrders.SetBusinessLocation",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersDraftV1DraftOrdersUrl({
        protoPath: "/v1/draft-orders/set-business-location",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "calculatedDraftOrder.draftOrder.createdDate" },
            { path: "calculatedDraftOrder.draftOrder.updatedDate" },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.urlExpirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.digitalFile.expirationDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.x"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.image.focalPoint.y"
            },
            {
              path: "calculatedDraftOrder.draftOrder.lineItems.lineItem.physicalProperties.weight"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.changedDetails.shippingInfoBeforeChange.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.shippingInfo.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.changedDetails.recipientInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.recipientInfo.recipientInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.changedDetails.billingInfoBeforeChange.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.draftOrder.billingDetails.billingInfo.address.geocode.longitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "calculatedDraftOrder.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setBusinessLocation;
}

// src/ecom-v1-draft-order-draft-orders.meta.ts
function createDraftOrder2() {
  const payload = {};
  const getRequestOptions = createDraftOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createEmptyDraftOrder2() {
  const payload = {};
  const getRequestOptions = createEmptyDraftOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/create-empty-draft-order",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addLineItemsToDraftOrder2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = addLineItemsToDraftOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/add-line-items",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateLineItems2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = updateLineItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/line-items",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setDiscounts2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = setDiscounts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/set-discounts",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createCustomDiscounts2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = createCustomDiscounts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/create-custom-discounts",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCustomDiscounts2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = deleteCustomDiscounts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/delete-custom-discounts",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setAdditionalFees2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = setAdditionalFees(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/set-additional-fees",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createCustomAdditionalFees2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = createCustomAdditionalFees(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/create-additional-fees",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCustomAdditionalFees2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = deleteCustomAdditionalFees(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/delete-custom-additional-fees",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setShippingInfo2() {
  const payload = {};
  const getRequestOptions = setShippingInfo(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v1/draft-orders/set-shipping-info",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setBuyerInfo2() {
  const payload = {};
  const getRequestOptions = setBuyerInfo(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v1/draft-orders/set-buyer-info",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setRecipientInfo2() {
  const payload = {};
  const getRequestOptions = setRecipientInfo(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v1/draft-orders/set-recipient-info",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setBillingInfo2() {
  const payload = {};
  const getRequestOptions = setBillingInfo(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v1/draft-orders/set-billing-info",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDraftOrder2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = getDraftOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/draft-orders/{draftOrderId}",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getOrderDraftabilityStatus2() {
  const payload = { orderId: ":orderId" };
  const getRequestOptions = getOrderDraftabilityStatus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/draft-orders/{orderId}/draftability",
    pathParams: { orderId: "orderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function commitDraftOrder2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = commitDraftOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/commit",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createOrderFromDraft2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = createOrderFromDraft(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{draftOrderId}/create-order",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteDraftOrder2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = deleteDraftOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/draft-orders/{draftOrderId}",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDraftOrders2() {
  const payload = {};
  const getRequestOptions = queryDraftOrders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/draft-orders/{id}/update-extended-fields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setTaxExemption2() {
  const payload = {};
  const getRequestOptions = setTaxExemption(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v1/draft-orders/set-tax-exemption",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setBusinessLocation2() {
  const payload = {};
  const getRequestOptions = setBusinessLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v1/draft-orders/set-business-location",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  addLineItemsToDraftOrder2 as addLineItemsToDraftOrder,
  commitDraftOrder2 as commitDraftOrder,
  createCustomAdditionalFees2 as createCustomAdditionalFees,
  createCustomDiscounts2 as createCustomDiscounts,
  createDraftOrder2 as createDraftOrder,
  createEmptyDraftOrder2 as createEmptyDraftOrder,
  createOrderFromDraft2 as createOrderFromDraft,
  deleteCustomAdditionalFees2 as deleteCustomAdditionalFees,
  deleteCustomDiscounts2 as deleteCustomDiscounts,
  deleteDraftOrder2 as deleteDraftOrder,
  getDraftOrder2 as getDraftOrder,
  getOrderDraftabilityStatus2 as getOrderDraftabilityStatus,
  queryDraftOrders2 as queryDraftOrders,
  setAdditionalFees2 as setAdditionalFees,
  setBillingInfo2 as setBillingInfo,
  setBusinessLocation2 as setBusinessLocation,
  setBuyerInfo2 as setBuyerInfo,
  setDiscounts2 as setDiscounts,
  setRecipientInfo2 as setRecipientInfo,
  setShippingInfo2 as setShippingInfo,
  setTaxExemption2 as setTaxExemption,
  updateExtendedFields2 as updateExtendedFields,
  updateLineItems2 as updateLineItems
};
