// @wix/auto_sdk_ecom_tax-groups/build/internal/es/meta.mjs
// src/billing-v1-tax-group-tax-groups.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBillingV1TaxGroupsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/billing/v1/tax-groups",
        destPath: "/v1/tax-groups"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/billing/v1/tax-groups",
        destPath: "/v1/tax-groups"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_tax-groups";
function createTaxGroup(payload) {
  function __createTaxGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "taxGroup.createdDate" },
          { path: "taxGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "POST",
      methodFqn: "wix.billing.v1.TaxGroups.CreateTaxGroup",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroup.createdDate" },
            { path: "taxGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createTaxGroup;
}
function getTaxGroup(payload) {
  function __getTaxGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "GET",
      methodFqn: "wix.billing.v1.TaxGroups.GetTaxGroup",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/{taxGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroup.createdDate" },
            { path: "taxGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTaxGroup;
}
function updateTaxGroup(payload) {
  function __updateTaxGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "taxGroup.createdDate" },
          { path: "taxGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "PATCH",
      methodFqn: "wix.billing.v1.TaxGroups.UpdateTaxGroup",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/{taxGroup.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroup.createdDate" },
            { path: "taxGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateTaxGroup;
}
function deleteTaxGroup(payload) {
  function __deleteTaxGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "DELETE",
      methodFqn: "wix.billing.v1.TaxGroups.DeleteTaxGroup",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/{taxGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteTaxGroup;
}
function queryTaxGroups(payload) {
  function __queryTaxGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "POST",
      methodFqn: "wix.billing.v1.TaxGroups.QueryTaxGroups",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroups.createdDate" },
            { path: "taxGroups.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTaxGroups;
}
function listDefaultTaxGroups(payload) {
  function __listDefaultTaxGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "GET",
      methodFqn: "wix.billing.v1.TaxGroups.ListDefaultTaxGroups",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/default-tax-groups",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroups.createdDate" },
            { path: "taxGroups.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDefaultTaxGroups;
}
function listDefaultTaxGroupsByAppIds(payload) {
  function __listDefaultTaxGroupsByAppIds({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "POST",
      methodFqn: "wix.billing.v1.TaxGroups.ListDefaultTaxGroupsByAppIds",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/list-default-tax-groups-by-app-ids",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.taxGroup.createdDate" },
            { path: "results.taxGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDefaultTaxGroupsByAppIds;
}

// src/billing-v1-tax-group-tax-groups.meta.ts
function createTaxGroup2() {
  const payload = {};
  const getRequestOptions = createTaxGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tax-groups",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getTaxGroup2() {
  const payload = { taxGroupId: ":taxGroupId" };
  const getRequestOptions = getTaxGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/tax-groups/{taxGroupId}",
    pathParams: { taxGroupId: "taxGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateTaxGroup2() {
  const payload = { taxGroup: { id: ":taxGroupId" } };
  const getRequestOptions = updateTaxGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/tax-groups/{taxGroup.id}",
    pathParams: { taxGroupId: "taxGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteTaxGroup2() {
  const payload = { taxGroupId: ":taxGroupId" };
  const getRequestOptions = deleteTaxGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/tax-groups/{taxGroupId}",
    pathParams: { taxGroupId: "taxGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryTaxGroups2() {
  const payload = {};
  const getRequestOptions = queryTaxGroups(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tax-groups/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDefaultTaxGroups2() {
  const payload = {};
  const getRequestOptions = listDefaultTaxGroups(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/tax-groups/default-tax-groups",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDefaultTaxGroupsByAppIds2() {
  const payload = {};
  const getRequestOptions = listDefaultTaxGroupsByAppIds(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tax-groups/list-default-tax-groups-by-app-ids",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createTaxGroup2 as createTaxGroup,
  deleteTaxGroup2 as deleteTaxGroup,
  getTaxGroup2 as getTaxGroup,
  listDefaultTaxGroups2 as listDefaultTaxGroups,
  listDefaultTaxGroupsByAppIds2 as listDefaultTaxGroupsByAppIds,
  queryTaxGroups2 as queryTaxGroups,
  updateTaxGroup2 as updateTaxGroup
};
