// @wix/auto_sdk_ecom_checkout-templates/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  createAndRedirectToCheckout: () => createAndRedirectToCheckout2,
  createCheckoutFromTemplate: () => createCheckoutFromTemplate2,
  createCheckoutTemplate: () => createCheckoutTemplate2,
  deleteCheckoutTemplate: () => deleteCheckoutTemplate2,
  getCheckoutTemplate: () => getCheckoutTemplate2,
  queryCheckoutTemplates: () => queryCheckoutTemplates2,
  updateCheckoutTemplate: () => updateCheckoutTemplate2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-v1-checkout-template-checkout-templates.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_bytes = require("@wix/sdk-runtime/transformations/bytes");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomCheckoutTemplateApiV1CheckoutTemplateServiceUrl(opts) {
  const domainToMappings = {
    "www._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-templates",
        destPath: "/v1/checkout-templates"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/checkout-templates",
        destPath: "/v1/checkout-templates"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/checkout-templates",
        destPath: "/v1/checkout-templates"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/checkout-templates",
        destPath: "/v1/checkout-templates"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/checkout-template",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_checkout-templates";
function createCheckoutTemplate(payload) {
  function __createCheckoutTemplate({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "checkoutTemplate.lineItems.catalogOverrideFields.physicalProperties.weight"
          },
          {
            path: "checkoutTemplate.customLineItems.physicalProperties.weight"
          }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          {
            path: "checkoutTemplate.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate"
          },
          {
            path: "checkoutTemplate.customLineItems.serviceProperties.scheduledDate"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.checkout_template",
      method: "POST",
      methodFqn: "wix.ecom.checkout_template.api.v1.CheckoutTemplateService.CreateCheckoutTemplate",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomCheckoutTemplateApiV1CheckoutTemplateServiceUrl({
        protoPath: "/v1/checkout-templates",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "checkoutTemplate.lineItems.catalogOverrideFields.physicalProperties.weight"
            },
            {
              path: "checkoutTemplate.customLineItems.physicalProperties.weight"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "checkoutTemplate.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "checkoutTemplate.customLineItems.serviceProperties.scheduledDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCheckoutTemplate;
}
function getCheckoutTemplate(payload) {
  function __getCheckoutTemplate({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.checkout_template",
      method: "GET",
      methodFqn: "wix.ecom.checkout_template.api.v1.CheckoutTemplateService.GetCheckoutTemplate",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomCheckoutTemplateApiV1CheckoutTemplateServiceUrl({
        protoPath: "/v1/checkout-templates/{checkoutTemplateId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "checkoutTemplate.lineItems.catalogOverrideFields.physicalProperties.weight"
            },
            {
              path: "checkoutTemplate.customLineItems.physicalProperties.weight"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "checkoutTemplate.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "checkoutTemplate.customLineItems.serviceProperties.scheduledDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCheckoutTemplate;
}
function updateCheckoutTemplate(payload) {
  function __updateCheckoutTemplate({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "checkoutTemplate.lineItems.catalogOverrideFields.physicalProperties.weight"
          },
          {
            path: "checkoutTemplate.customLineItems.physicalProperties.weight"
          }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          {
            path: "checkoutTemplate.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate"
          },
          {
            path: "checkoutTemplate.customLineItems.serviceProperties.scheduledDate"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.checkout_template",
      method: "PATCH",
      methodFqn: "wix.ecom.checkout_template.api.v1.CheckoutTemplateService.UpdateCheckoutTemplate",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomCheckoutTemplateApiV1CheckoutTemplateServiceUrl({
        protoPath: "/v1/checkout-templates/{checkoutTemplate.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "checkoutTemplate.lineItems.catalogOverrideFields.physicalProperties.weight"
            },
            {
              path: "checkoutTemplate.customLineItems.physicalProperties.weight"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "checkoutTemplate.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "checkoutTemplate.customLineItems.serviceProperties.scheduledDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCheckoutTemplate;
}
function deleteCheckoutTemplate(payload) {
  function __deleteCheckoutTemplate({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.checkout_template",
      method: "DELETE",
      methodFqn: "wix.ecom.checkout_template.api.v1.CheckoutTemplateService.DeleteCheckoutTemplate",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomCheckoutTemplateApiV1CheckoutTemplateServiceUrl({
        protoPath: "/v1/checkout-templates/{checkoutTemplateId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteCheckoutTemplate;
}
function queryCheckoutTemplates(payload) {
  function __queryCheckoutTemplates({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.checkout_template",
      method: "POST",
      methodFqn: "wix.ecom.checkout_template.api.v1.CheckoutTemplateService.QueryCheckoutTemplates",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomCheckoutTemplateApiV1CheckoutTemplateServiceUrl({
        protoPath: "/v1/checkout-templates/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "checkoutTemplates.lineItems.catalogOverrideFields.physicalProperties.weight"
            },
            {
              path: "checkoutTemplates.customLineItems.physicalProperties.weight"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "checkoutTemplates.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "checkoutTemplates.customLineItems.serviceProperties.scheduledDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryCheckoutTemplates;
}
function createCheckoutFromTemplate(payload) {
  function __createCheckoutFromTemplate({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.checkout_template",
      method: "POST",
      methodFqn: "wix.ecom.checkout_template.api.v1.CheckoutTemplateService.CreateCheckoutFromTemplate",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomCheckoutTemplateApiV1CheckoutTemplateServiceUrl({
        protoPath: "/v1/checkout-templates/{checkoutTemplateId}/create-checkout-from-template",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createCheckoutFromTemplate;
}
function createAndRedirectToCheckout(payload) {
  function __createAndRedirectToCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.checkout_template",
      method: "GET",
      methodFqn: "wix.ecom.checkout_template.api.v1.CheckoutTemplateService.CreateAndRedirectToCheckout",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomCheckoutTemplateApiV1CheckoutTemplateServiceUrl({
        protoPath: "/v1/checkout-templates/{checkoutTemplateId}/create-and-redirect-to-checkout",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_bytes.transformRESTBytesToSDKBytes,
          paths: [{ path: "body" }]
        }
      ])
    };
    return metadata;
  }
  return __createAndRedirectToCheckout;
}

// src/ecom-v1-checkout-template-checkout-templates.meta.ts
function createCheckoutTemplate2() {
  const payload = {};
  const getRequestOptions = createCheckoutTemplate(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/checkout-templates",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCheckoutTemplate2() {
  const payload = { checkoutTemplateId: ":checkoutTemplateId" };
  const getRequestOptions = getCheckoutTemplate(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/checkout-templates/{checkoutTemplateId}",
    pathParams: { checkoutTemplateId: "checkoutTemplateId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCheckoutTemplate2() {
  const payload = { checkoutTemplate: { id: ":checkoutTemplateId" } };
  const getRequestOptions = updateCheckoutTemplate(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/checkout-templates/{checkoutTemplate.id}",
    pathParams: { checkoutTemplateId: "checkoutTemplateId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCheckoutTemplate2() {
  const payload = { checkoutTemplateId: ":checkoutTemplateId" };
  const getRequestOptions = deleteCheckoutTemplate(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/checkout-templates/{checkoutTemplateId}",
    pathParams: { checkoutTemplateId: "checkoutTemplateId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCheckoutTemplates2() {
  const payload = {};
  const getRequestOptions = queryCheckoutTemplates(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/checkout-templates/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createCheckoutFromTemplate2() {
  const payload = { checkoutTemplateId: ":checkoutTemplateId" };
  const getRequestOptions = createCheckoutFromTemplate(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/checkout-templates/{checkoutTemplateId}/create-checkout-from-template",
    pathParams: { checkoutTemplateId: "checkoutTemplateId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createAndRedirectToCheckout2() {
  const payload = { checkoutTemplateId: ":checkoutTemplateId" };
  const getRequestOptions = createAndRedirectToCheckout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/checkout-templates/{checkoutTemplateId}/create-and-redirect-to-checkout",
    pathParams: { checkoutTemplateId: "checkoutTemplateId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createAndRedirectToCheckout,
  createCheckoutFromTemplate,
  createCheckoutTemplate,
  deleteCheckoutTemplate,
  getCheckoutTemplate,
  queryCheckoutTemplates,
  updateCheckoutTemplate
});
