// @wix/auto_sdk_ecom_discounts/build/es/index.mjs
// src/service-plugins-types.ts
var TargetType = /* @__PURE__ */ ((TargetType2) => {
  TargetType2["UNKNOWN_TARGET"] = "UNKNOWN_TARGET";
  TargetType2["GLOBAL"] = "GLOBAL";
  TargetType2["SHIPPING"] = "SHIPPING";
  TargetType2["LINE_ITEMS"] = "LINE_ITEMS";
  return TargetType2;
})(TargetType || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_RATE"] = "UNKNOWN_RATE";
  Type2["FIXED_AMOUNT"] = "FIXED_AMOUNT";
  Type2["PERCENTAGE"] = "PERCENTAGE";
  return Type2;
})(Type || {});
var DiscountSource = /* @__PURE__ */ ((DiscountSource2) => {
  DiscountSource2["UNKNOWN_SOURCE"] = "UNKNOWN_SOURCE";
  DiscountSource2["COUPON"] = "COUPON";
  DiscountSource2["AUTOMATIC"] = "AUTOMATIC";
  DiscountSource2["CUSTOM"] = "CUSTOM";
  return DiscountSource2;
})(DiscountSource || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});

// src/interfaces-ecommerce-v1-discounts-provider.public.ts
import { ServicePluginDefinition } from "@wix/sdk-types";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";
var provideHandlers = ServicePluginDefinition("ECOM_DISCOUNTS", [
  {
    name: "getEligibleDiscounts",
    primaryHttpMappingPath: "/v1/eligible-discounts",
    transformations: {
      toREST: (payload) => {
        const toRestResponse = payload;
        return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
      },
      fromREST: (payload) => {
        const fromRestRequest = payload;
        return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
      }
    }
  }
]);

// src/interfaces-ecommerce-v1-discounts-provider.context.ts
import { createServicePluginModule } from "@wix/sdk-runtime/service-plugin-modules";
var provideHandlers2 = createServicePluginModule(
  provideHandlers
);

// src/service-plugins-error-classes.ts
var CouponLimitPerCustomerExceededWixError = class extends Error {
  constructor() {
    super("CouponLimitPerCustomerExceeded");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_LIMIT_PER_CUSTOMER_EXCEEDED";
    this.name = "CouponLimitPerCustomerExceeded";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponLimitPerCustomerExceeded",
      applicationCode: "ERROR_COUPON_LIMIT_PER_CUSTOMER_EXCEEDED"
    };
  }
};
/** @hidden */
CouponLimitPerCustomerExceededWixError.__type = "wix_spi_error";
var CouponIsDisabledWixError = class extends Error {
  constructor() {
    super("CouponIsDisabled");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_IS_DISABLED";
    this.name = "CouponIsDisabled";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponIsDisabled",
      applicationCode: "ERROR_COUPON_IS_DISABLED"
    };
  }
};
/** @hidden */
CouponIsDisabledWixError.__type = "wix_spi_error";
var CouponUsageExceededWixError = class extends Error {
  constructor() {
    super("CouponUsageExceeded");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_USAGE_EXCEEDED";
    this.name = "CouponUsageExceeded";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponUsageExceeded",
      applicationCode: "ERROR_COUPON_USAGE_EXCEEDED"
    };
  }
};
/** @hidden */
CouponUsageExceededWixError.__type = "wix_spi_error";
var CouponHasExpiredWixError = class extends Error {
  constructor() {
    super("CouponHasExpired");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_HAS_EXPIRED";
    this.name = "CouponHasExpired";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponHasExpired",
      applicationCode: "ERROR_COUPON_HAS_EXPIRED"
    };
  }
};
/** @hidden */
CouponHasExpiredWixError.__type = "wix_spi_error";
var CouponIsNotActiveYetWixError = class extends Error {
  constructor() {
    super("CouponIsNotActiveYet");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_IS_NOT_ACTIVE_YET";
    this.name = "CouponIsNotActiveYet";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponIsNotActiveYet",
      applicationCode: "ERROR_COUPON_IS_NOT_ACTIVE_YET"
    };
  }
};
/** @hidden */
CouponIsNotActiveYetWixError.__type = "wix_spi_error";
var CouponDoesNotExistWixError = class extends Error {
  constructor() {
    super("CouponDoesNotExist");
    this.httpCode = 404;
    this.statusCode = "NOT_FOUND";
    this.applicationCode = "ERROR_COUPON_DOES_NOT_EXIST";
    this.name = "CouponDoesNotExist";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponDoesNotExist",
      applicationCode: "ERROR_COUPON_DOES_NOT_EXIST"
    };
  }
};
/** @hidden */
CouponDoesNotExistWixError.__type = "wix_spi_error";
export {
  CouponDoesNotExistWixError,
  CouponHasExpiredWixError,
  CouponIsDisabledWixError,
  CouponIsNotActiveYetWixError,
  CouponLimitPerCustomerExceededWixError,
  CouponUsageExceededWixError,
  DiscountSource,
  IdentityType,
  TargetType,
  Type,
  provideHandlers2 as provideHandlers,
  provideHandlers as publicProvideHandlers
};
