// @wix/auto_sdk_ecom_discounts-custom-trigger/build/internal/es/meta.mjs
// src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.http.ts
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixEcomCustomTriggerV1CustomTriggersServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/custom-triggers-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/discount-custom-triggers",
        destPath: "/v1/discount-custom-triggers"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/discount-custom-triggers",
        destPath: "/v1/discount-custom-triggers"
      },
      {
        srcPath: "/v1/discount-custom-triggers",
        destPath: "/v1/discount-custom-triggers"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_discounts-custom-trigger";
function listTriggers(payload) {
  function __listTriggers({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.custom_trigger.v1.custom_trigger",
      method: "POST",
      methodFqn: "com.wix.ecom.custom_trigger.v1.CustomTriggersService.ListTriggers",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomCustomTriggerV1CustomTriggersServiceUrl({
        protoPath: "/v1/discount-custom-triggers/list",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listTriggers;
}

// src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.meta.ts
function listTriggers2() {
  const payload = {};
  const getRequestOptions = listTriggers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/discount-custom-triggers/list",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  listTriggers2 as listTriggers
};
