// @wix/auto_sdk_ecom_abandoned-checkouts/build/es/index.typings.mjs
// src/ecom-v1-abandoned-checkout-abandoned-checkouts.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTBytesToSDKBytes } from "@wix/sdk-runtime/transformations/bytes";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/abandoned-checkout",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_abandoned-checkouts";
function getAbandonedCheckout(payload) {
  function __getAbandonedCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "GET",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.GetAbandonedCheckout",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckout.createdDate" },
            { path: "abandonedCheckout.updatedDate" },
            { path: "abandonedCheckout.checkoutRecoveredDate" },
            { path: "abandonedCheckout.activities.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAbandonedCheckout;
}
function deleteAbandonedCheckout(payload) {
  function __deleteAbandonedCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "DELETE",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.DeleteAbandonedCheckout",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteAbandonedCheckout;
}
function queryAbandonedCheckouts(payload) {
  function __queryAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.QueryAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.createdDate" },
            { path: "results.updatedDate" },
            { path: "results.checkoutRecoveredDate" },
            { path: "results.activities.createdDate" },
            { path: "abandonedCheckouts.createdDate" },
            { path: "abandonedCheckouts.updatedDate" },
            { path: "abandonedCheckouts.checkoutRecoveredDate" },
            { path: "abandonedCheckouts.activities.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAbandonedCheckouts;
}
function searchAbandonedCheckouts(payload) {
  function __searchAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.SearchAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckouts.createdDate" },
            { path: "abandonedCheckouts.updatedDate" },
            { path: "abandonedCheckouts.checkoutRecoveredDate" },
            { path: "abandonedCheckouts.activities.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchAbandonedCheckouts;
}
function addAbandonedCheckoutActivity(payload) {
  function __addAbandonedCheckoutActivity({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.AddAbandonedCheckoutActivity",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkouts/{abandonedCheckoutId}/add-activity",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckout.createdDate" },
            { path: "abandonedCheckout.updatedDate" },
            { path: "abandonedCheckout.checkoutRecoveredDate" },
            { path: "abandonedCheckout.activities.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addAbandonedCheckoutActivity;
}
function redirectToCheckout(payload) {
  function __redirectToCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "GET",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.RedirectToCheckout",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}/redirect-to-checkout",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTBytesToSDKBytes,
          paths: [{ path: "body" }]
        }
      ])
    };
    return metadata;
  }
  return __redirectToCheckout;
}

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var ActivityType = /* @__PURE__ */ ((ActivityType2) => {
  ActivityType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  ActivityType2["SCHEDULED"] = "SCHEDULED";
  ActivityType2["EMAIL_SENT"] = "EMAIL_SENT";
  ActivityType2["EMAIL_NOT_SENT"] = "EMAIL_NOT_SENT";
  ActivityType2["NOTIFICATION_SENT"] = "NOTIFICATION_SENT";
  ActivityType2["TASK_CREATED"] = "TASK_CREATED";
  return ActivityType2;
})(ActivityType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["ABANDONED"] = "ABANDONED";
  Status2["RECOVERED"] = "RECOVERED";
  return Status2;
})(Status || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var Identity = /* @__PURE__ */ ((Identity2) => {
  Identity2["ADMIN"] = "ADMIN";
  Identity2["MEMBER"] = "MEMBER";
  Identity2["VISITOR"] = "VISITOR";
  Identity2["CONTACT"] = "CONTACT";
  return Identity2;
})(Identity || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var CommonSortOrder = /* @__PURE__ */ ((CommonSortOrder2) => {
  CommonSortOrder2["ASC"] = "ASC";
  CommonSortOrder2["DESC"] = "DESC";
  return CommonSortOrder2;
})(CommonSortOrder || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getAbandonedCheckout2(abandonedCheckoutId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    abandonedCheckoutId
  });
  const reqOpts = getAbandonedCheckout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.abandonedCheckout;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { abandonedCheckoutId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["abandonedCheckoutId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteAbandonedCheckout2(abandonedCheckoutId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    abandonedCheckoutId
  });
  const reqOpts = deleteAbandonedCheckout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { abandonedCheckoutId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["abandonedCheckoutId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryAbandonedCheckouts2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryAbandonedCheckouts(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.abandonedCheckouts,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function searchAbandonedCheckouts2(search) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ search });
  const reqOpts = searchAbandonedCheckouts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { search: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["search"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function addAbandonedCheckoutActivity2(abandonedCheckoutId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    abandonedCheckoutId,
    activityType: options?.activityType
  });
  const reqOpts = addAbandonedCheckoutActivity(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          abandonedCheckoutId: "$[0]",
          activityType: "$[1].activityType"
        },
        singleArgumentUnchanged: false
      },
      ["abandonedCheckoutId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function redirectToCheckout2(abandonedCheckoutId, metasiteId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    abandonedCheckoutId,
    metasiteId
  });
  const reqOpts = redirectToCheckout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          abandonedCheckoutId: "$[0]",
          metasiteId: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["abandonedCheckoutId", "metasiteId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  ActivityType,
  CommonSortOrder,
  Identity,
  Mode,
  SortOrder,
  Status,
  VatType,
  WebhookIdentityType,
  addAbandonedCheckoutActivity2 as addAbandonedCheckoutActivity,
  deleteAbandonedCheckout2 as deleteAbandonedCheckout,
  getAbandonedCheckout2 as getAbandonedCheckout,
  queryAbandonedCheckouts2 as queryAbandonedCheckouts,
  redirectToCheckout2 as redirectToCheckout,
  searchAbandonedCheckouts2 as searchAbandonedCheckouts
};
