// @wix/auto_sdk_ecom_totals-calculator/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  calculateTotals: () => calculateTotals2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-v1-totals-calculation-totals-calculator.http.ts
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomTotalsCalculatorV1TotalsCalculatorUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/totals-calculator",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/totals-calculator",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/totals-calculator",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/calculate-totals",
        destPath: "/v1/calculate-totals"
      }
    ],
    _: [
      {
        srcPath: "/_api/totals-calculator",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/totals-calculator",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_totals-calculator";
function calculateTotals(payload) {
  function __calculateTotals({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "lineItems.physicalProperties.weight" },
          { path: "shippingAddress.geocode.latitude" },
          { path: "shippingAddress.geocode.longitude" },
          { path: "billingAddress.geocode.latitude" },
          { path: "billingAddress.geocode.longitude" },
          {
            path: "shippingOptions.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
          },
          {
            path: "shippingOptions.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
          },
          { path: "appliedDiscounts.merchantDiscount.percentage" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "lineItems.serviceProperties.scheduledDate" },
          {
            path: "shippingOptions.shippingOptions.logistics.deliveryTimeSlot.from"
          },
          {
            path: "shippingOptions.shippingOptions.logistics.deliveryTimeSlot.to"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.totals_calculation",
      method: "POST",
      methodFqn: "com.wix.ecom.totals_calculator.v1.TotalsCalculator.CalculateTotals",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomTotalsCalculatorV1TotalsCalculatorUrl({
        protoPath: "/v1/calculate-totals",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "shippingInfo.carrierServiceOptions.shippingOptions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "shippingInfo.carrierServiceOptions.shippingOptions.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "appliedDiscounts.merchantDiscount.percentage" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.from"
            },
            {
              path: "shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.to"
            },
            {
              path: "shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.from"
            },
            {
              path: "shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.to"
            },
            { path: "membershipOptions.eligibleMemberships.expirationDate" },
            {
              path: "membershipOptions.invalidMemberships.membership.expirationDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateTotals;
}

// src/ecom-v1-totals-calculation-totals-calculator.meta.ts
function calculateTotals2() {
  const payload = {};
  const getRequestOptions = calculateTotals(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/calculate-totals",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  calculateTotals
});
