// @wix/auto_sdk_ecom_abandoned-checkouts/build/internal/es/meta.mjs
// src/ecom-v1-abandoned-checkout-abandoned-checkouts.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTBytesToSDKBytes } from "@wix/sdk-runtime/transformations/bytes";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/abandoned-checkout",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_abandoned-checkouts";
function getAbandonedCheckout(payload) {
  function __getAbandonedCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "GET",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.GetAbandonedCheckout",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckout.createdDate" },
            { path: "abandonedCheckout.updatedDate" },
            { path: "abandonedCheckout.checkoutRecoveredDate" },
            { path: "abandonedCheckout.activities.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAbandonedCheckout;
}
function deleteAbandonedCheckout(payload) {
  function __deleteAbandonedCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "DELETE",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.DeleteAbandonedCheckout",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteAbandonedCheckout;
}
function queryAbandonedCheckouts(payload) {
  function __queryAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.QueryAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.createdDate" },
            { path: "results.updatedDate" },
            { path: "results.checkoutRecoveredDate" },
            { path: "results.activities.createdDate" },
            { path: "abandonedCheckouts.createdDate" },
            { path: "abandonedCheckouts.updatedDate" },
            { path: "abandonedCheckouts.checkoutRecoveredDate" },
            { path: "abandonedCheckouts.activities.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAbandonedCheckouts;
}
function searchAbandonedCheckouts(payload) {
  function __searchAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.SearchAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckouts.createdDate" },
            { path: "abandonedCheckouts.updatedDate" },
            { path: "abandonedCheckouts.checkoutRecoveredDate" },
            { path: "abandonedCheckouts.activities.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchAbandonedCheckouts;
}
function addAbandonedCheckoutActivity(payload) {
  function __addAbandonedCheckoutActivity({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.AddAbandonedCheckoutActivity",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkouts/{abandonedCheckoutId}/add-activity",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckout.createdDate" },
            { path: "abandonedCheckout.updatedDate" },
            { path: "abandonedCheckout.checkoutRecoveredDate" },
            { path: "abandonedCheckout.activities.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addAbandonedCheckoutActivity;
}
function redirectToCheckout(payload) {
  function __redirectToCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "GET",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.RedirectToCheckout",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}/redirect-to-checkout",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTBytesToSDKBytes,
          paths: [{ path: "body" }]
        }
      ])
    };
    return metadata;
  }
  return __redirectToCheckout;
}

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.meta.ts
function getAbandonedCheckout2() {
  const payload = { abandonedCheckoutId: ":abandonedCheckoutId" };
  const getRequestOptions = getAbandonedCheckout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/abandoned-checkout/{abandonedCheckoutId}",
    pathParams: { abandonedCheckoutId: "abandonedCheckoutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteAbandonedCheckout2() {
  const payload = { abandonedCheckoutId: ":abandonedCheckoutId" };
  const getRequestOptions = deleteAbandonedCheckout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/abandoned-checkout/{abandonedCheckoutId}",
    pathParams: { abandonedCheckoutId: "abandonedCheckoutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryAbandonedCheckouts2() {
  const payload = {};
  const getRequestOptions = queryAbandonedCheckouts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/abandoned-checkout/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function searchAbandonedCheckouts2() {
  const payload = {};
  const getRequestOptions = searchAbandonedCheckouts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/abandoned-checkout/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addAbandonedCheckoutActivity2() {
  const payload = { abandonedCheckoutId: ":abandonedCheckoutId" };
  const getRequestOptions = addAbandonedCheckoutActivity(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/abandoned-checkouts/{abandonedCheckoutId}/add-activity",
    pathParams: { abandonedCheckoutId: "abandonedCheckoutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function redirectToCheckout2() {
  const payload = { abandonedCheckoutId: ":abandonedCheckoutId" };
  const getRequestOptions = redirectToCheckout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/abandoned-checkout/{abandonedCheckoutId}/redirect-to-checkout",
    pathParams: { abandonedCheckoutId: "abandonedCheckoutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  addAbandonedCheckoutActivity2 as addAbandonedCheckoutActivity,
  deleteAbandonedCheckout2 as deleteAbandonedCheckout,
  getAbandonedCheckout2 as getAbandonedCheckout,
  queryAbandonedCheckouts2 as queryAbandonedCheckouts,
  redirectToCheckout2 as redirectToCheckout,
  searchAbandonedCheckouts2 as searchAbandonedCheckouts
};
