// @wix/auto_sdk_ecom_delivery-profile/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  addDeliveryCarrier: () => addDeliveryCarrier2,
  addDeliveryRegion: () => addDeliveryRegion2,
  createDeliveryProfile: () => createDeliveryProfile2,
  deleteDeliveryProfile: () => deleteDeliveryProfile2,
  getDeliveryProfile: () => getDeliveryProfile2,
  getDeliveryProfileByDeliveryRegionId: () => getDeliveryProfileByDeliveryRegionId2,
  listDeliveryCarriers: () => listDeliveryCarriers2,
  listInstalledDeliveryCarriers: () => listInstalledDeliveryCarriers2,
  queryDeliveryProfiles: () => queryDeliveryProfiles2,
  removeDeliveryCarrier: () => removeDeliveryCarrier2,
  removeDeliveryRegion: () => removeDeliveryRegion2,
  updateCarrierSettings: () => updateCarrierSettings2,
  updateDeliveryCarrier: () => updateDeliveryCarrier2,
  updateDeliveryProfile: () => updateDeliveryProfile2,
  updateDeliveryRegion: () => updateDeliveryRegion2,
  updateExtendedFields: () => updateExtendedFields2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-v1-delivery-profile-delivery-profile.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/list-delivery-carriers",
        destPath: "/v1/list-delivery-carriers"
      },
      {
        srcPath: "/ecom/v1/list-delivery-carrier-details",
        destPath: "/v1/list-delivery-carrier-details"
      },
      {
        srcPath: "/ecom/v1/delivery-profiles",
        destPath: "/v1/delivery-profiles"
      },
      {
        srcPath: "/ecom/v1/delivery-carriers",
        destPath: "/v1/delivery-carriers"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/delivery-profiles-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/delivery-profiles",
        destPath: "/v1/delivery-profiles"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/delivery-profiles",
        destPath: "/v1/delivery-profiles"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/delivery-profiles",
        destPath: "/v1/delivery-profiles"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_delivery-profile";
function createDeliveryProfile(payload) {
  function __createDeliveryProfile({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "deliveryProfile.createdDate" },
          { path: "deliveryProfile.updatedDate" },
          { path: "deliveryProfile.deliveryRegions.createdDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.CreateDeliveryProfile",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDeliveryProfile;
}
function getDeliveryProfile(payload) {
  function __getDeliveryProfile({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "GET",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.GetDeliveryProfile",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/{deliveryProfileId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDeliveryProfile;
}
function updateDeliveryProfile(payload) {
  function __updateDeliveryProfile({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "deliveryProfile.createdDate" },
          { path: "deliveryProfile.updatedDate" },
          { path: "deliveryProfile.deliveryRegions.createdDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "PATCH",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateDeliveryProfile",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/{deliveryProfile.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateDeliveryProfile;
}
function deleteDeliveryProfile(payload) {
  function __deleteDeliveryProfile({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "DELETE",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.DeleteDeliveryProfile",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/{deliveryProfileId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteDeliveryProfile;
}
function queryDeliveryProfiles(payload) {
  function __queryDeliveryProfiles({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "GET",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.QueryDeliveryProfiles",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/query",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfiles.createdDate" },
            { path: "deliveryProfiles.updatedDate" },
            { path: "deliveryProfiles.deliveryRegions.createdDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
            protoPath: "/v1/delivery-profiles/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDeliveryProfiles;
}
function addDeliveryRegion(payload) {
  function __addDeliveryRegion({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "deliveryRegion.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.AddDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/{deliveryProfileId}/delivery-region",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addDeliveryRegion;
}
function updateDeliveryRegion(payload) {
  function __updateDeliveryRegion({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "deliveryRegion.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "PATCH",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegion.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateDeliveryRegion;
}
function getDeliveryProfileByDeliveryRegionId(payload) {
  function __getDeliveryProfileByDeliveryRegionId({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "GET",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.GetDeliveryProfileByDeliveryRegionId",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/delivery-regions/{deliveryRegionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDeliveryProfileByDeliveryRegionId;
}
function removeDeliveryRegion(payload) {
  function __removeDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "DELETE",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.RemoveDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removeDeliveryRegion;
}
function addDeliveryCarrier(payload) {
  function __addDeliveryCarrier({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.AddDeliveryCarrier",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/add-delivery-carrier",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addDeliveryCarrier;
}
function removeDeliveryCarrier(payload) {
  function __removeDeliveryCarrier({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.RemoveDeliveryCarrier",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/remove-delivery-carrier",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removeDeliveryCarrier;
}
function updateDeliveryCarrier(payload) {
  function __updateDeliveryCarrier({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "PATCH",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateDeliveryCarrier",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/update-delivery-carrier",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateDeliveryCarrier;
}
function listInstalledDeliveryCarriers(payload) {
  function __listInstalledDeliveryCarriers({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "GET",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.ListInstalledDeliveryCarriers",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/installed-carriers",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listInstalledDeliveryCarriers;
}
function listDeliveryCarriers(payload) {
  function __listDeliveryCarriers({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.ListDeliveryCarriers",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/{deliveryProfileId}/delivery-carriers",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listDeliveryCarriers;
}
function updateCarrierSettings(payload) {
  function __updateCarrierSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateCarrierSettings",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-carriers/update-settings",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateCarrierSettings;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.delivery_profile",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
        protoPath: "/v1/delivery-profiles/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliveryProfile.createdDate" },
            { path: "deliveryProfile.updatedDate" },
            { path: "deliveryProfile.deliveryRegions.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}

// src/ecom-v1-delivery-profile-delivery-profile.meta.ts
function createDeliveryProfile2() {
  const payload = {};
  const getRequestOptions = createDeliveryProfile(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/delivery-profiles",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDeliveryProfile2() {
  const payload = { deliveryProfileId: ":deliveryProfileId" };
  const getRequestOptions = getDeliveryProfile(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/delivery-profiles/{deliveryProfileId}",
    pathParams: { deliveryProfileId: "deliveryProfileId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateDeliveryProfile2() {
  const payload = { deliveryProfile: { id: ":deliveryProfileId" } };
  const getRequestOptions = updateDeliveryProfile(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/delivery-profiles/{deliveryProfile.id}",
    pathParams: { deliveryProfileId: "deliveryProfileId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteDeliveryProfile2() {
  const payload = { deliveryProfileId: ":deliveryProfileId" };
  const getRequestOptions = deleteDeliveryProfile(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/delivery-profiles/{deliveryProfileId}",
    pathParams: { deliveryProfileId: "deliveryProfileId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDeliveryProfiles2() {
  const payload = {};
  const getRequestOptions = queryDeliveryProfiles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/delivery-profiles/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addDeliveryRegion2() {
  const payload = { deliveryProfileId: ":deliveryProfileId" };
  const getRequestOptions = addDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/delivery-profiles/{deliveryProfileId}/delivery-region",
    pathParams: { deliveryProfileId: "deliveryProfileId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateDeliveryRegion2() {
  const payload = {
    deliveryProfileId: ":deliveryProfileId",
    deliveryRegion: { id: ":deliveryRegionId" }
  };
  const getRequestOptions = updateDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegion.id}",
    pathParams: {
      deliveryProfileId: "deliveryProfileId",
      deliveryRegionId: "deliveryRegionId"
    },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDeliveryProfileByDeliveryRegionId2() {
  const payload = { deliveryRegionId: ":deliveryRegionId" };
  const getRequestOptions = getDeliveryProfileByDeliveryRegionId(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/delivery-profiles/delivery-regions/{deliveryRegionId}",
    pathParams: { deliveryRegionId: "deliveryRegionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeDeliveryRegion2() {
  const payload = {
    deliveryProfileId: ":deliveryProfileId",
    deliveryRegionId: ":deliveryRegionId"
  };
  const getRequestOptions = removeDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegionId}",
    pathParams: {
      deliveryProfileId: "deliveryProfileId",
      deliveryRegionId: "deliveryRegionId"
    },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addDeliveryCarrier2() {
  const payload = {};
  const getRequestOptions = addDeliveryCarrier(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/delivery-profiles/add-delivery-carrier",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeDeliveryCarrier2() {
  const payload = {};
  const getRequestOptions = removeDeliveryCarrier(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/delivery-profiles/remove-delivery-carrier",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateDeliveryCarrier2() {
  const payload = {};
  const getRequestOptions = updateDeliveryCarrier(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/delivery-profiles/update-delivery-carrier",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listInstalledDeliveryCarriers2() {
  const payload = {};
  const getRequestOptions = listInstalledDeliveryCarriers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/delivery-profiles/installed-carriers",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDeliveryCarriers2() {
  const payload = { deliveryProfileId: ":deliveryProfileId" };
  const getRequestOptions = listDeliveryCarriers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/delivery-profiles/{deliveryProfileId}/delivery-carriers",
    pathParams: { deliveryProfileId: "deliveryProfileId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCarrierSettings2() {
  const payload = {};
  const getRequestOptions = updateCarrierSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/delivery-carriers/update-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/delivery-profiles/{id}/update-extended-fields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  addDeliveryCarrier,
  addDeliveryRegion,
  createDeliveryProfile,
  deleteDeliveryProfile,
  getDeliveryProfile,
  getDeliveryProfileByDeliveryRegionId,
  listDeliveryCarriers,
  listInstalledDeliveryCarriers,
  queryDeliveryProfiles,
  removeDeliveryCarrier,
  removeDeliveryRegion,
  updateCarrierSettings,
  updateDeliveryCarrier,
  updateDeliveryProfile,
  updateDeliveryRegion,
  updateExtendedFields
});
