// @wix/error-handler/dist/cjs/components/ErrorToastContent/ErrorToastContent.js
"use strict";

exports.__esModule = true;
exports.ErrorToastContent = ErrorToastContent;
var _designSystem = require("@wix/design-system");
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _react = _interopRequireWildcard(require("react"));
var _jsxFileName = "/home/builduser/work/e2b2992289f90ea5/packages/error-handler/dist/cjs/components/ErrorToastContent/ErrorToastContent.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const PopoverContent = /*#__PURE__*/(0, _react.memo)(function PopoverContent(props) {
  const {
    requestId,
    errorHandler
  } = props;
  const {
    translations: {
      translate: t
    }
  } = errorHandler;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    gap: "SP2",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 31,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 32,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
    dataHook: "info-description",
    size: "small",
    skin: "standard",
    weight: "thin",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 33,
      columnNumber: 9
    }
  }, t('errorHandler.requestID.tooltip.description'))), /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    marginTop: 2,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 42,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.CopyClipboard, {
    value: requestId,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 43,
      columnNumber: 9
    }
  }, ({
    isCopied,
    copyToClipboard
  }) => /*#__PURE__*/_react.default.createElement(_designSystem.TextButton, {
    prefixIcon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.DuplicateSmall, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 46,
        columnNumber: 27
      }
    }),
    size: "small",
    onClick: () => {
      copyToClipboard();
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 45,
      columnNumber: 13
    }
  }, isCopied ? t('errorHandler.requestID.tooltip.cta.copied-success') : t('errorHandler.requestID.tooltip.cta')))));
});
const RequestInfo = /*#__PURE__*/(0, _react.memo)(function RequestInfo(props) {
  const {
    errorHandler,
    requestId,
    shown,
    setShown
  } = props;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    gap: 1,
    dataHook: "request-info",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 72,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
    dataHook: "info-request-id",
    size: "small",
    weight: "thin",
    light: true,
    secondary: true,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 73,
      columnNumber: 7
    }
  }, `Request ID: ${requestId}`), /*#__PURE__*/_react.default.createElement(_designSystem.Popover, {
    showArrow: true,
    animate: true,
    placement: "auto",
    shown: shown,
    onMouseEnter: () => setShown(true),
    onMouseLeave: () => setShown(false),
    appendTo: "parent",
    dataHook: "info-popover",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 82,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Popover.Element, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 92,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.TextButton, {
    onClick: () => setShown(true),
    as: "button",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 93,
      columnNumber: 11
    }
  }, /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.InfoCircleSmall, {
    fill: "var(--wsr-color-D40, #868aa5)",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 94,
      columnNumber: 13
    }
  }))), /*#__PURE__*/_react.default.createElement(_designSystem.Popover.Content, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 97,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    width: "350px",
    position: "relative",
    padding: "SP5",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 98,
      columnNumber: 11
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    position: "absolute",
    top: "6px",
    right: "12px",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 99,
      columnNumber: 13
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.CloseButton, {
    size: "medium",
    onClick: () => setShown(false),
    skin: "dark",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 100,
      columnNumber: 15
    }
  })), /*#__PURE__*/_react.default.createElement(PopoverContent, {
    errorHandler: errorHandler,
    requestId: requestId,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 107,
      columnNumber: 13
    }
  })))));
});
function ErrorToastContent(props) {
  const {
    dataHook,
    message,
    requestId,
    errorHandler
  } = props;
  const {
    translations
  } = errorHandler;
  const [shown, setShown] = (0, _react.useState)(false);
  const [ready, setReady] = (0, _react.useState)(translations._ready);
  (0, _react.useEffect)(() => {
    if (ready) {
      return;
    }
    translations.getInitPromise().then(() => {
      setReady(true);
    });
  }, []);
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    dataHook: dataHook,
    direction: "vertical",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 134,
      columnNumber: 5
    }
  }, message, requestId && ready && /*#__PURE__*/_react.default.createElement(RequestInfo, {
    errorHandler: errorHandler,
    requestId: requestId,
    shown: shown,
    setShown: setShown,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 138,
      columnNumber: 9
    }
  }));
}
