// @wix/auto_sdk_ecom_checkout-content/build/internal/es/index.mjs
// src/ecom-api-v1-checkout-content-provider-checkout-content.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/ecom-api-v1-checkout-content-provider-checkout-content.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomCheckoutContentSpiHostV1CheckoutContentSpiHostServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_checkout-content";
function getCheckoutContent(payload) {
  function __getCheckoutContent({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.api.v1.checkout_content_provider",
      method: "GET",
      methodFqn: "wix.ecom.checkout_content_spi_host.v1.CheckoutContentSPIHostService.GetCheckoutContent",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomCheckoutContentSpiHostV1CheckoutContentSpiHostServiceUrl(
        {
          protoPath: "/v1/checkout-content/{appId}/{componentId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCheckoutContent;
}

// src/ecom-api-v1-checkout-content-provider-checkout-content.universal.ts
async function getCheckoutContent2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    appId: identifiers?.appId,
    componentId: identifiers?.componentId,
    languageCode: options?.languageCode
  });
  const reqOpts = getCheckoutContent(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          appId: "$[0].appId",
          componentId: "$[0].componentId",
          languageCode: "$[1].languageCode"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecom-api-v1-checkout-content-provider-checkout-content.public.ts
function getCheckoutContent3(httpClient) {
  return (identifiers, options) => getCheckoutContent2(
    identifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/ecom-api-v1-checkout-content-provider-checkout-content.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getCheckoutContent4 = /* @__PURE__ */ createRESTModule(getCheckoutContent3);
export {
  getCheckoutContent4 as getCheckoutContent
};
