// @wix/auto_sdk_ecom_delivery-solutions/build/internal/es/index.mjs
// src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.http.ts
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomDeliveryprofileV1DeliveryRatesUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/delivery-rates",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_delivery-solutions";
function getDeliverySolutions(payload) {
  function __getDeliverySolutions({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "lineItems.physicalProperties.weight" },
          { path: "destination.geocode.latitude" },
          { path: "destination.geocode.longitude" },
          { path: "origin.geocode.latitude" },
          { path: "origin.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.gateways.ecom.v1.delivery_rates_gateway",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryRates.GetDeliverySolutions",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryRatesUrl({
        protoPath: "/v1/get-delivery-solutions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "deliverySolutions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "deliverySolutions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliverySolutions.logistics.deliveryTimeSlot.from" },
            { path: "deliverySolutions.logistics.deliveryTimeSlot.to" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDeliverySolutions;
}

// src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.universal.ts
import { transformSDKAddressToRESTAddress } from "@wix/sdk-runtime/transformations/address";
import { transformRESTAddressToSDKAddress } from "@wix/sdk-runtime/transformations/address";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SubscriptionFrequency = /* @__PURE__ */ ((SubscriptionFrequency2) => {
  SubscriptionFrequency2["UNDEFINED"] = "UNDEFINED";
  SubscriptionFrequency2["DAY"] = "DAY";
  SubscriptionFrequency2["WEEK"] = "WEEK";
  SubscriptionFrequency2["MONTH"] = "MONTH";
  SubscriptionFrequency2["YEAR"] = "YEAR";
  return SubscriptionFrequency2;
})(SubscriptionFrequency || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var WeightUnit = /* @__PURE__ */ ((WeightUnit2) => {
  WeightUnit2["UNSPECIFIED_WEIGHT_UNIT"] = "UNSPECIFIED_WEIGHT_UNIT";
  WeightUnit2["KG"] = "KG";
  WeightUnit2["LB"] = "LB";
  return WeightUnit2;
})(WeightUnit || {});
var PickupMethod = /* @__PURE__ */ ((PickupMethod2) => {
  PickupMethod2["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
  PickupMethod2["STORE_PICKUP"] = "STORE_PICKUP";
  PickupMethod2["PICKUP_POINT"] = "PICKUP_POINT";
  return PickupMethod2;
})(PickupMethod || {});
var ChargeType = /* @__PURE__ */ ((ChargeType2) => {
  ChargeType2["HANDLING_FEE"] = "HANDLING_FEE";
  return ChargeType2;
})(ChargeType || {});
var RuleType = /* @__PURE__ */ ((RuleType2) => {
  RuleType2["VALIDATION"] = "VALIDATION";
  RuleType2["OTHER"] = "OTHER";
  RuleType2["MAX"] = "MAX";
  RuleType2["MIN"] = "MIN";
  RuleType2["MAX_LENGTH"] = "MAX_LENGTH";
  RuleType2["MIN_LENGTH"] = "MIN_LENGTH";
  RuleType2["MAX_SIZE"] = "MAX_SIZE";
  RuleType2["MIN_SIZE"] = "MIN_SIZE";
  RuleType2["FORMAT"] = "FORMAT";
  RuleType2["DECIMAL_LTE"] = "DECIMAL_LTE";
  RuleType2["DECIMAL_GTE"] = "DECIMAL_GTE";
  RuleType2["DECIMAL_LT"] = "DECIMAL_LT";
  RuleType2["DECIMAL_GT"] = "DECIMAL_GT";
  RuleType2["DECIMAL_MAX_SCALE"] = "DECIMAL_MAX_SCALE";
  RuleType2["INVALID_ENUM_VALUE"] = "INVALID_ENUM_VALUE";
  RuleType2["REQUIRED_FIELD"] = "REQUIRED_FIELD";
  RuleType2["FIELD_NOT_ALLOWED"] = "FIELD_NOT_ALLOWED";
  RuleType2["ONE_OF_ALIGNMENT"] = "ONE_OF_ALIGNMENT";
  RuleType2["EXACT_LENGTH"] = "EXACT_LENGTH";
  RuleType2["EXACT_SIZE"] = "EXACT_SIZE";
  RuleType2["REQUIRED_ONE_OF_FIELD"] = "REQUIRED_ONE_OF_FIELD";
  return RuleType2;
})(RuleType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getDeliverySolutions2(lineItems, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      lineItems,
      destination: options?.destination,
      origin: options?.origin,
      currency: options?.currency,
      contactDetails: options?.contactDetails,
      weightUnit: options?.weightUnit,
      userPreferences: options?.userPreferences,
      purchaseFlowId: options?.purchaseFlowId,
      externalReferences: options?.externalReferences
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "destination" }, { path: "origin" }]
      }
    ]
  );
  const reqOpts = getDeliverySolutions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "deliverySolutions.logistics.pickupDetails.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          lineItems: "$[0]",
          destination: "$[1].destination",
          origin: "$[1].origin",
          currency: "$[1].currency",
          contactDetails: "$[1].contactDetails",
          weightUnit: "$[1].weightUnit",
          userPreferences: "$[1].userPreferences",
          purchaseFlowId: "$[1].purchaseFlowId",
          externalReferences: "$[1].externalReferences"
        },
        singleArgumentUnchanged: false
      },
      ["lineItems", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.public.ts
function getDeliverySolutions3(httpClient) {
  return (lineItems, options) => getDeliverySolutions2(
    lineItems,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getDeliverySolutions4 = /* @__PURE__ */ createRESTModule(getDeliverySolutions3);
export {
  ChargeType,
  PickupMethod,
  RuleType,
  SubdivisionType,
  SubscriptionFrequency,
  VatType,
  WebhookIdentityType,
  WeightUnit,
  getDeliverySolutions4 as getDeliverySolutions
};
