// @wix/auto_sdk_ecom_pickup-locations/build/internal/es/meta.mjs
// src/ecom-v1-pickup-location-pickup-locations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomV1PickupLocationsUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pickup-locations",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/pickup-locations",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_pickup-locations";
function createPickupLocation(payload) {
  function __createPickupLocation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pickupLocation.createdDate" },
          { path: "pickupLocation.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.CreatePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPickupLocation;
}
function getPickupLocation(payload) {
  function __getPickupLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "GET",
      methodFqn: "wix.ecom.v1.PickupLocations.GetPickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveWixEcomV1PickupLocationsUrl({
            protoPath: "/v1/pickup-locations/{pickupLocationId}",
            data: payload,
            host
          }),
          params: toURLSearchParams(payload)
        }
      ]
    };
    return metadata;
  }
  return __getPickupLocation;
}
function updatePickupLocation(payload) {
  function __updatePickupLocation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pickupLocation.createdDate" },
          { path: "pickupLocation.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "PATCH",
      methodFqn: "wix.ecom.v1.PickupLocations.UpdatePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocation.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePickupLocation;
}
function deletePickupLocation(payload) {
  function __deletePickupLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "DELETE",
      methodFqn: "wix.ecom.v1.PickupLocations.DeletePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deletePickupLocation;
}
function queryPickupLocation(payload) {
  function __queryPickupLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.QueryPickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocations.createdDate" },
            { path: "pickupLocations.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPickupLocation;
}
function addDeliveryRegion(payload) {
  function __addDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.AddDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocationId}/add-delivery-region",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addDeliveryRegion;
}
function removeDeliveryRegion(payload) {
  function __removeDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.RemoveDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocationId}/remove-delivery-region",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removeDeliveryRegion;
}
function bulkCreatePickupLocation(payload) {
  function __bulkCreatePickupLocation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pickupLocations.createdDate" },
          { path: "pickupLocations.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.BulkCreatePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/bulk/pickup-locations/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocations.createdDate" },
            { path: "pickupLocations.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreatePickupLocation;
}
function bulkUpdatePickupLocation(payload) {
  function __bulkUpdatePickupLocation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pickupLocations.createdDate" },
          { path: "pickupLocations.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "PATCH",
      methodFqn: "wix.ecom.v1.PickupLocations.BulkUpdatePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/bulk/pickup-locations/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocations.createdDate" },
            { path: "pickupLocations.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdatePickupLocation;
}
function bulkDeletePickupLocation(payload) {
  function __bulkDeletePickupLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "DELETE",
      methodFqn: "wix.ecom.v1.PickupLocations.BulkDeletePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/bulk/pickup-locations/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeletePickupLocation;
}

// src/ecom-v1-pickup-location-pickup-locations.meta.ts
function createPickupLocation2() {
  const payload = {};
  const getRequestOptions = createPickupLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pickup-locations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPickupLocation2() {
  const payload = { pickupLocationId: ":pickupLocationId" };
  const getRequestOptions = getPickupLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/pickup-locations/{pickupLocationId}",
    pathParams: { pickupLocationId: "pickupLocationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePickupLocation2() {
  const payload = { pickupLocation: { id: ":pickupLocationId" } };
  const getRequestOptions = updatePickupLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/pickup-locations/{pickupLocation.id}",
    pathParams: { pickupLocationId: "pickupLocationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deletePickupLocation2() {
  const payload = { pickupLocationId: ":pickupLocationId" };
  const getRequestOptions = deletePickupLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/pickup-locations/{pickupLocationId}",
    pathParams: { pickupLocationId: "pickupLocationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPickupLocation2() {
  const payload = {};
  const getRequestOptions = queryPickupLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pickup-locations/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addDeliveryRegion2() {
  const payload = { pickupLocationId: ":pickupLocationId" };
  const getRequestOptions = addDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pickup-locations/{pickupLocationId}/add-delivery-region",
    pathParams: { pickupLocationId: "pickupLocationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeDeliveryRegion2() {
  const payload = { pickupLocationId: ":pickupLocationId" };
  const getRequestOptions = removeDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pickup-locations/{pickupLocationId}/remove-delivery-region",
    pathParams: { pickupLocationId: "pickupLocationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreatePickupLocation2() {
  const payload = {};
  const getRequestOptions = bulkCreatePickupLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/pickup-locations/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePickupLocation2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePickupLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/bulk/pickup-locations/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeletePickupLocation2() {
  const payload = {};
  const getRequestOptions = bulkDeletePickupLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/bulk/pickup-locations/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  addDeliveryRegion2 as addDeliveryRegion,
  bulkCreatePickupLocation2 as bulkCreatePickupLocation,
  bulkDeletePickupLocation2 as bulkDeletePickupLocation,
  bulkUpdatePickupLocation2 as bulkUpdatePickupLocation,
  createPickupLocation2 as createPickupLocation,
  deletePickupLocation2 as deletePickupLocation,
  getPickupLocation2 as getPickupLocation,
  queryPickupLocation2 as queryPickupLocation,
  removeDeliveryRegion2 as removeDeliveryRegion,
  updatePickupLocation2 as updatePickupLocation
};
