// @wix/ecom_app-extensions/build/cjs/site.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// site.ts
var site_exports = {};
__export(site_exports, {
  plugins: () => editor_pages_plugin_props_exports
});
module.exports = __toCommonJS(site_exports);

// src/extensions/editor-pages-plugin-props.ts
var editor_pages_plugin_props_exports = {};
__export(editor_pages_plugin_props_exports, {
  CategoryPage: () => bda15dc1_816d_4ff3_8dcb_1172d5343cce_exports,
  Checkout: () => fd5970_8072_c276_1246_058b79e70c1a_exports,
  GridGallery: () => afb094_84f9_739f_44fd_78d036adb028_exports,
  ProductPage: () => a94f09_2766_3c40_4a32_8edb5acdd8bc_exports,
  SearchModalGallery: () => c9_d856_49b9_af14_216071924e3b_exports,
  SearchResultsPageGallery: () => abcd87fe_c51f_4538_848d_2902a2f50d2d_exports,
  Shop: () => bba0_253e_a800_a235_88821cf3f8a4_exports
});

// src/extensions/editor-page-plugins/13afb094-84f9-739f-44fd-78d036adb028.ts
var afb094_84f9_739f_44fd_78d036adb028_exports = {};
__export(afb094_84f9_739f_44fd_78d036adb028_exports, {
  SlotIds: () => SlotIds
});
var SlotIds = {
  galleryFiltersTop: "a81dea42-6bdc-4cd9-b8e0-995751d926a9",
  galleryFiltersBottom: "bb2f1ffa-3dd8-434f-bd9a-faa1fbbfa0f9",
  galleryProductsTop: "4c5fb938-90bc-420d-8766-d8b428dd00ee",
  galleryProductsBottom: "4a10dfa2-e2aa-47ae-9629-ca86c1d04751"
};

// src/extensions/editor-page-plugins/bda15dc1-816d-4ff3-8dcb-1172d5343cce.ts
var bda15dc1_816d_4ff3_8dcb_1172d5343cce_exports = {};
__export(bda15dc1_816d_4ff3_8dcb_1172d5343cce_exports, {
  SlotIds: () => SlotIds2
});
var SlotIds2 = {
  galleryFiltersTop: "a81dea42-6bdc-4cd9-b8e0-995751d926a9",
  galleryFiltersBottom: "bb2f1ffa-3dd8-434f-bd9a-faa1fbbfa0f9",
  galleryProductsTop: "4c5fb938-90bc-420d-8766-d8b428dd00ee",
  galleryProductsBottom: "4a10dfa2-e2aa-47ae-9629-ca86c1d04751",
  categoryPageHeroBottom: "e5f2b946-9932-4b38-9165-8a6054fad913",
  categoryPageListTop: "281f21dd-27f1-4466-b1fb-1af332f0cea8",
  categoryPageHeroTop: "7bff6166-e633-48b8-ac3a-f6b9660d39dc"
};

// src/extensions/editor-page-plugins/13a94f09-2766-3c40-4a32-8edb5acdd8bc.ts
var a94f09_2766_3c40_4a32_8edb5acdd8bc_exports = {};
__export(a94f09_2766_3c40_4a32_8edb5acdd8bc_exports, {
  SlotIds: () => SlotIds3
});
var SlotIds3 = {
  checkoutHeader: "bb058d1f-b70f-4df9-9da0-0523322a38a1",
  productPageAdditionalButtons: "9d31fcc2-53ee-4547-b837-878f5394ed7e",
  productPageDetails5: "1db8f870-879b-46b8-88df-f232e2621eb2",
  productPageDetails7: "15c88f6e-fbd2-418b-a389-00bcd1937dc1",
  productPageDetails2: "905e9719-9672-437f-85a0-1975337ba08c",
  productPageDetails9: "a49a45fe-2f34-4eae-a8f5-d971f822e645",
  productPageMedia1: "f74d0746-dd85-4a4e-b1c5-8207b29cc191",
  productPageDetails6: "6feecc92-8459-4698-ad4b-b085a74407d8",
  productPageDetails4: "cec2ee92-7e12-42a1-a2f5-fe02f30e688a",
  productPageTop: "2832b6e7-1bb3-4b08-b361-5e77a13621dc",
  productPageDetails1: "8e497c6c-3ed9-4995-8ee9-e9f939f7a355",
  productPageDetails3: "b9dff4d5-192d-4574-bf14-baf6e2734f3a",
  productPageDetails8: "edce2d4e-51c9-4e75-b7db-090da80ff455",
  productPageBottom: "ad454e1e-aa18-46a6-ae81-5df549178c7a"
};

// src/extensions/editor-page-plugins/14fd5970-8072-c276-1246-058b79e70c1a.ts
var fd5970_8072_c276_1246_058b79e70c1a_exports = {};
__export(fd5970_8072_c276_1246_058b79e70c1a_exports, {
  SlotIds: () => SlotIds4
});
var SlotIds4 = {
  checkoutPoliciesAfter1: "465be850-f1e6-4e80-b125-a62e56005f05",
  checkoutSummaryLineItemsAfter2: "f63efbab-2b28-4565-af96-d198040d2303",
  checkoutSummaryAfter: "c9614893-83dd-4481-8a86-62ce3c696fb0",
  checkoutTop: "210b7bf5-8532-4f42-9f5f-00204534b403",
  checkoutSummaryTotalsBreakdownBefore: "5e7ea8d8-adec-4a12-b28b-d07fbb055442",
  checkoutStepsBefore: "16cf4440-fa01-47ff-973e-051bb91c25d9",
  checkoutSummaryBefore: "969d874f-bdba-4429-a626-ff801d712c6a",
  checkoutSummaryLineItemsAfter: "6918440e-160c-4b55-87b6-2af2e83b3174",
  checkoutHeader: "bb058d1f-b70f-4df9-9da0-0523322a38a1"
};

// src/extensions/editor-page-plugins/abcd87fe-c51f-4538-848d-2902a2f50d2d.ts
var abcd87fe_c51f_4538_848d_2902a2f50d2d_exports = {};
__export(abcd87fe_c51f_4538_848d_2902a2f50d2d_exports, {
  SlotIds: () => SlotIds5
});
var SlotIds5 = {
  galleryFiltersTop: "a81dea42-6bdc-4cd9-b8e0-995751d926a9",
  galleryFiltersBottom: "bb2f1ffa-3dd8-434f-bd9a-faa1fbbfa0f9",
  galleryProductsTop: "4c5fb938-90bc-420d-8766-d8b428dd00ee",
  galleryProductsBottom: "4a10dfa2-e2aa-47ae-9629-ca86c1d04751"
};

// src/extensions/editor-page-plugins/1380bba0-253e-a800-a235-88821cf3f8a4.ts
var bba0_253e_a800_a235_88821cf3f8a4_exports = {};
__export(bba0_253e_a800_a235_88821cf3f8a4_exports, {
  SlotIds: () => SlotIds6
});
var SlotIds6 = {
  galleryFiltersTop: "a81dea42-6bdc-4cd9-b8e0-995751d926a9",
  galleryFiltersBottom: "bb2f1ffa-3dd8-434f-bd9a-faa1fbbfa0f9",
  galleryProductsTop: "4c5fb938-90bc-420d-8766-d8b428dd00ee",
  galleryProductsBottom: "4a10dfa2-e2aa-47ae-9629-ca86c1d04751"
};

// src/extensions/editor-page-plugins/244576c9-d856-49b9-af14-216071924e3b.ts
var c9_d856_49b9_af14_216071924e3b_exports = {};
__export(c9_d856_49b9_af14_216071924e3b_exports, {
  SlotIds: () => SlotIds7
});
var SlotIds7 = {
  galleryFiltersTop: "a81dea42-6bdc-4cd9-b8e0-995751d926a9",
  galleryFiltersBottom: "bb2f1ffa-3dd8-434f-bd9a-faa1fbbfa0f9",
  galleryProductsTop: "4c5fb938-90bc-420d-8766-d8b428dd00ee",
  galleryProductsBottom: "4a10dfa2-e2aa-47ae-9629-ca86c1d04751"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  plugins
});
