// @wix/auto_sdk_ecom_gift-vouchers/build/internal/es/meta.mjs
// src/ecom-v1-gift-card-gift-vouchers.http.ts
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/gift-cards-spi-host",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_gift-vouchers";
function getGiftCard(payload) {
  function __getGiftCard({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.gift_card",
      method: "POST",
      methodFqn: "wix.ecom.gift_cards_spi_host.v1.GiftCardsSpiHostService.GetGiftCard",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl({
        protoPath: "/v1/gift-cards/{code}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getGiftCard;
}
function redeemGiftCard(payload) {
  function __redeemGiftCard({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.gift_card",
      method: "POST",
      methodFqn: "wix.ecom.gift_cards_spi_host.v1.GiftCardsSpiHostService.RedeemGiftCard",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl({
        protoPath: "/v1/gift-cards/redeem",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __redeemGiftCard;
}
function voidTransaction(payload) {
  function __voidTransaction({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.gift_card",
      method: "POST",
      methodFqn: "wix.ecom.gift_cards_spi_host.v1.GiftCardsSpiHostService.VoidTransaction",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl({
        protoPath: "/v1/gift-cards/void",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __voidTransaction;
}

// src/ecom-v1-gift-card-gift-vouchers.meta.ts
function getGiftCard2() {
  const payload = { code: ":code" };
  const getRequestOptions = getGiftCard(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/gift-cards/{code}",
    pathParams: { code: "code" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function redeemGiftCard2() {
  const payload = {};
  const getRequestOptions = redeemGiftCard(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/gift-cards/redeem",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function voidTransaction2() {
  const payload = {};
  const getRequestOptions = voidTransaction(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/gift-cards/void",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getGiftCard2 as getGiftCard,
  redeemGiftCard2 as redeemGiftCard,
  voidTransaction2 as voidTransaction
};
