// @wix/auto_sdk_ecom_recommendations/build/internal/es/meta.mjs
// src/ecom-recommendations-v1-recommendation-recommendations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixEcomRecommendationsV1RecommendationsServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/v1/recommendations",
        destPath: "/v1/recommendations"
      },
      {
        srcPath: "/ecom/v1/recommendations",
        destPath: "/v1/recommendations"
      }
    ],
    _: [
      {
        srcPath: "/v1/recommendations",
        destPath: "/v1/recommendations"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/v1/recommendations",
        destPath: "/v1/recommendations"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_recommendations";
function listAvailableAlgorithms(payload) {
  function __listAvailableAlgorithms({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.recommendations.v1.recommendation",
      method: "GET",
      methodFqn: "com.wix.ecom.recommendations.v1.RecommendationsService.ListAvailableAlgorithms",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomRecommendationsV1RecommendationsServiceUrl({
        protoPath: "/v1/recommendations/algorithms",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listAvailableAlgorithms;
}
function getRecommendation(payload) {
  function __getRecommendation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.recommendations.v1.recommendation",
      method: "POST",
      methodFqn: "com.wix.ecom.recommendations.v1.RecommendationsService.GetRecommendation",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomRecommendationsV1RecommendationsServiceUrl({
        protoPath: "/v1/recommendations/get",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getRecommendation;
}

// src/ecom-recommendations-v1-recommendation-recommendations.meta.ts
function listAvailableAlgorithms2() {
  const payload = {};
  const getRequestOptions = listAvailableAlgorithms(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/recommendations/algorithms",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRecommendation2() {
  const payload = {};
  const getRequestOptions = getRecommendation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/recommendations/get",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getRecommendation2 as getRecommendation,
  listAvailableAlgorithms2 as listAvailableAlgorithms
};
