// @wix/error-handler/dist/esm/testkit/createMockHttpError.js
import { createHttpError } from '@wix/fe-essentials/http-client';
import { getReasonPhrase } from 'http-status-codes';
export function createMockHttpError(response, overrides) {
  var _overrides$response;
  const config = {
    ...(overrides == null ? void 0 : overrides.config)
  };
  return createHttpError({
    name: 'MockHttpError',
    message: response.message ?? 'MockHttpError',
    response: {
      status: response.status,
      statusText: getReasonPhrase(response.status),
      data: {
        message: response.message
      },
      headers: {
        'x-wix-request-id': response.requestId,
        ...(overrides == null || (_overrides$response = overrides.response) == null ? void 0 : _overrides$response.headers)
      },
      config,
      ...(overrides == null ? void 0 : overrides.response)
    },
    config,
    toJSON() {
      return this;
    },
    __CANCEL__: false,
    ...overrides
  });
}
export function createMockHttpApplicationError(applicationError, response, overrides) {
  var _overrides$response2;
  const config = {
    ...(overrides == null ? void 0 : overrides.config)
  };
  return createHttpError({
    name: 'MockHttpApplicationError',
    message: response.message ?? 'MockHttpApplicationError',
    response: {
      status: response.status,
      statusText: getReasonPhrase(response.status),
      data: {
        message: response.message,
        details: {
          applicationError
        }
      },
      headers: {
        'x-wix-request-id': response.requestId,
        ...(overrides == null || (_overrides$response2 = overrides.response) == null ? void 0 : _overrides$response2.headers)
      },
      config,
      ...(overrides == null ? void 0 : overrides.response)
    },
    config,
    toJSON() {
      return this;
    },
    __CANCEL__: false,
    ...overrides
  });
}
export function createMockHttpValidationError(validationError, response, overrides) {
  var _overrides$response3;
  const config = {
    ...(overrides == null ? void 0 : overrides.config)
  };
  return createHttpError({
    name: 'MockHttpValidationError',
    message: response.message ?? 'MockHttpValidationError',
    response: {
      status: response.status,
      statusText: getReasonPhrase(response.status),
      data: {
        message: response.message,
        details: {
          validationError
        }
      },
      headers: {
        'x-wix-request-id': response.requestId,
        ...(overrides == null || (_overrides$response3 = overrides.response) == null ? void 0 : _overrides$response3.headers)
      },
      config,
      ...(overrides == null ? void 0 : overrides.response)
    },
    config,
    toJSON() {
      return this;
    },
    __CANCEL__: false,
    ...overrides
  });
}
export function createMockNetworkError(type) {
  return new Error(type);
}
