// @wix/auto_sdk_ecom_currencies/build/internal/es/meta.mjs
// src/ecommerce-currency-converter-v1-currency-rate-currencies.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/currency-converter",
        destPath: ""
      }
    ],
    "cronulla-jobs._base_domain_": [
      {
        srcPath: "/currency-converter-bo",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "currency-converter.wixapps.net": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/currency_converter",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_currencies";
function listCurrencies(payload) {
  function __listCurrencies({ host }) {
    const metadata = {
      entityFqdn: "wix.ecommerce.currency_converter.v1.currency_rate",
      method: "GET",
      methodFqn: "com.wixpress.currency.converter.api.v1.CurrencyConverter.ListCurrencies",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
        protoPath: "/v1/currencies",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listCurrencies;
}
function convertCurrency(payload) {
  function __convertCurrency({ host }) {
    const metadata = {
      entityFqdn: "wix.ecommerce.currency_converter.v1.currency_rate",
      method: "POST",
      methodFqn: "com.wixpress.currency.converter.api.v1.CurrencyConverter.ConvertCurrency",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
        protoPath: "/v1/currencies/amounts/{from}/convert/{to}",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rateTimestamp" }]
        }
      ])
    };
    return metadata;
  }
  return __convertCurrency;
}
function conversionRate(payload) {
  function __conversionRate({ host }) {
    const metadata = {
      entityFqdn: "wix.ecommerce.currency_converter.v1.currency_rate",
      method: "GET",
      methodFqn: "com.wixpress.currency.converter.api.v1.CurrencyConverter.ConversionRate",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
        protoPath: "/v1/currencies/rate/{from}/convert/{to}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rateTimestamp" }]
        }
      ])
    };
    return metadata;
  }
  return __conversionRate;
}

// src/ecommerce-currency-converter-v1-currency-rate-currencies.meta.ts
function listCurrencies2() {
  const payload = {};
  const getRequestOptions = listCurrencies(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/currencies",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function convertCurrency2() {
  const payload = { from: ":from", to: ":to" };
  const getRequestOptions = convertCurrency(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/currencies/amounts/{from}/convert/{to}",
    pathParams: { from: "from", to: "to" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getConversionRate() {
  const payload = { from: ":from", to: ":to" };
  const getRequestOptions = conversionRate(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/currencies/rate/{from}/convert/{to}",
    pathParams: { from: "from", to: "to" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  convertCurrency2 as convertCurrency,
  getConversionRate,
  listCurrencies2 as listCurrencies
};
