// @wix/error-handler-core/@wix/bi-logger-os-data/dist/esm/testkit/node-testkit.js
/* tslint:disable */
/* eslint:disable */
import * as BiNodeTestkit from '@wix/wix-bi-node-testkit';
import { TestkitBase } from '@wix/wix-testkit-base';
import { setTimeout } from 'timers';
import { EventsFilter, validateSingleEvent } from './common';
export class WixBiSchemaNodeTestkit extends TestkitBase {
    constructor(logDir) {
        super();
        this.allFieldsAreMandatory = false;
        this.util = {
            /**
             * @description This function returns all sent events directly from 'biLoggerTestkit', without typechecking.
             */
            getRawEvents: () => {
                return this._biTestkit.events;
            },
            /**
             * This function will assert that all sent bi events have correct types in afterEach() phase of tests
             * @param timeout
             */
            assertAllEventsCorrectTypes: (timeout = 100) => {
                afterEach((done) => {
                    setTimeout(() => {
                        try {
                            this._biTestkit.events.forEach(this.validate);
                            done();
                        }
                        catch (e) {
                            done(e);
                        }
                    }, timeout);
                });
            },
            /**
             * This function will mark all fields as mandatory in the next assertion
             * @param timeout
             */
            makeAllFieldsMandatory: (value) => {
                this.allFieldsAreMandatory = value;
            },
        };
        this._biTestkit = BiNodeTestkit.interceptor(logDir);
        this.validate = this.validate.bind(this);
    }
    doStart() {
        return this._biTestkit.start();
    }
    doStop() {
        return this._biTestkit.stop();
    }
    validate(event) {
        return validateSingleEvent(event, this.allFieldsAreMandatory);
    }
    get wixPatternsSummaryBarIsShownSrc144Evid1() {
        return new EventsFilter(this.util.getRawEvents, 144, 1, this.validate);
    }
    get wixPatternsPageInteractiveSrc144Evid1000() {
        return new EventsFilter(this.util.getRawEvents, 144, 1000, this.validate);
    }
    get wixPatternsFullPageLoadCompletedSrc144Evid1001() {
        return new EventsFilter(this.util.getRawEvents, 144, 1001, this.validate);
    }
    get pickerModalUsability() {
        return new EventsFilter(this.util.getRawEvents, 144, 103, this.validate);
    }
    get pickerPickModalCtaButtonClickOrImmediateCta() {
        return new EventsFilter(this.util.getRawEvents, 144, 104, this.validate);
    }
    get pickerPickerOpenButton() {
        return new EventsFilter(this.util.getRawEvents, 144, 105, this.validate);
    }
    get newItemCreationStart() {
        return new EventsFilter(this.util.getRawEvents, 144, 106, this.validate);
    }
    get newItemCreationEnd() {
        return new EventsFilter(this.util.getRawEvents, 144, 107, this.validate);
    }
    get pikachuCategoriesDropdownButtonClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 108, this.validate);
    }
    get pikachuItemCategorySelected() {
        return new EventsFilter(this.util.getRawEvents, 144, 109, this.validate);
    }
    get loadStart() {
        return new EventsFilter(this.util.getRawEvents, 144, 110, this.validate);
    }
    get loadEnd() {
        return new EventsFilter(this.util.getRawEvents, 144, 111, this.validate);
    }
    get loadMore() {
        return new EventsFilter(this.util.getRawEvents, 144, 112, this.validate);
    }
    get componentDismissed() {
        return new EventsFilter(this.util.getRawEvents, 144, 113, this.validate);
    }
    get cairoItemSelectionToggled() {
        return new EventsFilter(this.util.getRawEvents, 144, 114, this.validate);
    }
    get cairoCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 115, this.validate);
    }
    get cairoSearchResults() {
        return new EventsFilter(this.util.getRawEvents, 144, 116, this.validate);
    }
    get cairoFilterToggled() {
        return new EventsFilter(this.util.getRawEvents, 144, 117, this.validate);
    }
    get cairoAllFiltersCleared() {
        return new EventsFilter(this.util.getRawEvents, 144, 118, this.validate);
    }
    get cairoFiltersPanelUsed() {
        return new EventsFilter(this.util.getRawEvents, 144, 119, this.validate);
    }
    get cairoComponentSorted() {
        return new EventsFilter(this.util.getRawEvents, 144, 120, this.validate);
    }
    get cairoTabSelected() {
        return new EventsFilter(this.util.getRawEvents, 144, 121, this.validate);
    }
    get cairoViewSaved() {
        return new EventsFilter(this.util.getRawEvents, 144, 122, this.validate);
    }
    get cairoViewSelected() {
        return new EventsFilter(this.util.getRawEvents, 144, 123, this.validate);
    }
    get wixPatternsViewSelectedLoadEndSrc144Evid1234() {
        return new EventsFilter(this.util.getRawEvents, 144, 1234, this.validate);
    }
    get cairoViewDeleted() {
        return new EventsFilter(this.util.getRawEvents, 144, 124, this.validate);
    }
    get cairoExportCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 125, this.validate);
    }
    get cairoItemToggledInExportModal() {
        return new EventsFilter(this.util.getRawEvents, 144, 126, this.validate);
    }
    get cairoExportModalCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 127, this.validate);
    }
    get cairoExportProcessStart() {
        return new EventsFilter(this.util.getRawEvents, 144, 128, this.validate);
    }
    get cairoUserAttemptsToCancelTheExport() {
        return new EventsFilter(this.util.getRawEvents, 144, 129, this.validate);
    }
    get cairoCancelExportModalCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 130, this.validate);
    }
    get cairoExportProcessEnd() {
        return new EventsFilter(this.util.getRawEvents, 144, 131, this.validate);
    }
    get cairoFileDownloadToastDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 144, 132, this.validate);
    }
    get cairoFileDownloadToastDismissed() {
        return new EventsFilter(this.util.getRawEvents, 144, 133, this.validate);
    }
    get cairoFileDownloadToastDownloadButtonClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 134, this.validate);
    }
    get cairoExportFailedModalCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 135, this.validate);
    }
    get cairoTableTopNotificationDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 144, 136, this.validate);
    }
    get cairoTableTopNotificationDismissed() {
        return new EventsFilter(this.util.getRawEvents, 144, 137, this.validate);
    }
    get cairoDragItem() {
        return new EventsFilter(this.util.getRawEvents, 144, 138, this.validate);
    }
    get cairoDragItemEndDrop() {
        return new EventsFilter(this.util.getRawEvents, 144, 139, this.validate);
    }
    get cairoDragActionUnsuccessfulUpdateInServer() {
        return new EventsFilter(this.util.getRawEvents, 144, 140, this.validate);
    }
    get cairoTryAgainClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 141, this.validate);
    }
    get cairoManageViewClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 142, this.validate);
    }
    get cairoManageViewActionSelected() {
        return new EventsFilter(this.util.getRawEvents, 144, 143, this.validate);
    }
    get cairoUndoClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 144, this.validate);
    }
    get cairoViewsUnsuccessfulUpdateInServer() {
        return new EventsFilter(this.util.getRawEvents, 144, 145, this.validate);
    }
    get cairoSortPanelUsed() {
        return new EventsFilter(this.util.getRawEvents, 144, 146, this.validate);
    }
    get cairoReorderingSortColumns() {
        return new EventsFilter(this.util.getRawEvents, 144, 147, this.validate);
    }
    get cairoSortUnsuccessfulUpdateInServer() {
        return new EventsFilter(this.util.getRawEvents, 144, 148, this.validate);
    }
    get cairoApiRequestDataLoadEnd() {
        return new EventsFilter(this.util.getRawEvents, 144, 149, this.validate);
    }
    get cairoPageCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 150, this.validate);
    }
    get cairoSearchViewInViews() {
        return new EventsFilter(this.util.getRawEvents, 144, 151, this.validate);
    }
    get cairoPageBackOrBreadcrumbClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 152, this.validate);
    }
    get cairoSectionCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 153, this.validate);
    }
    get cairoTpaSCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 154, this.validate);
    }
    get cairoDataExAddCustomField() {
        return new EventsFilter(this.util.getRawEvents, 144, 155, this.validate);
    }
    get cairoDataExCustomFieldEndProcess() {
        return new EventsFilter(this.util.getRawEvents, 144, 156, this.validate);
    }
    get cairoDataExCustomFieldAdvancesSettings() {
        return new EventsFilter(this.util.getRawEvents, 144, 157, this.validate);
    }
    get cairoLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 144, 158, this.validate);
    }
    get cairoApplyButtonClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 159, this.validate);
    }
    get cairoChangesBeforeApply() {
        return new EventsFilter(this.util.getRawEvents, 144, 160, this.validate);
    }
    get cairoDiscardChangesBeforeApply() {
        return new EventsFilter(this.util.getRawEvents, 144, 161, this.validate);
    }
    get cairoModalCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 162, this.validate);
    }
    get cairoDataExtensionUnsuccessfulUpdateInServer() {
        return new EventsFilter(this.util.getRawEvents, 144, 163, this.validate);
    }
    get cairoErrorInLoadingAComponent() {
        return new EventsFilter(this.util.getRawEvents, 144, 164, this.validate);
    }
    get cairoNestedTableCollapseAndExpand() {
        return new EventsFilter(this.util.getRawEvents, 144, 165, this.validate);
    }
    get cairoTagsModalStartLoad() {
        return new EventsFilter(this.util.getRawEvents, 144, 166, this.validate);
    }
    get cairoTagsModalEndProcess() {
        return new EventsFilter(this.util.getRawEvents, 144, 167, this.validate);
    }
    get cairoTagsAddDeleteTag() {
        return new EventsFilter(this.util.getRawEvents, 144, 168, this.validate);
    }
    get cairoTagsUnsuccessfulUpdateInServer() {
        return new EventsFilter(this.util.getRawEvents, 144, 169, this.validate);
    }
    get cairoSearchForTags() {
        return new EventsFilter(this.util.getRawEvents, 144, 170, this.validate);
    }
    get cairoTagsWidgetLoaded() {
        return new EventsFilter(this.util.getRawEvents, 144, 171, this.validate);
    }
    get cairoTagsWidgetClickOnCta() {
        return new EventsFilter(this.util.getRawEvents, 144, 172, this.validate);
    }
    get cairoTagsAssignTag() {
        return new EventsFilter(this.util.getRawEvents, 144, 173, this.validate);
    }
    get cairoPageDiscardChangesBeforeSave() {
        return new EventsFilter(this.util.getRawEvents, 144, 174, this.validate);
    }
    get cairoAssignTagsToEntities() {
        return new EventsFilter(this.util.getRawEvents, 144, 175, this.validate);
    }
    get cairoReloadErrorState() {
        return new EventsFilter(this.util.getRawEvents, 144, 176, this.validate);
    }
    get cairoCustomizeColumnsInSidePanel() {
        return new EventsFilter(this.util.getRawEvents, 144, 177, this.validate);
    }
    get wixPatternsNestedTableCollapseAndExpandAllSrc144Evid178() {
        return new EventsFilter(this.util.getRawEvents, 144, 178, this.validate);
    }
    get wixPatternsSectionHeaderPrimaryActionClicked() {
        return new EventsFilter(this.util.getRawEvents, 144, 2, this.validate);
    }
    get wixPatternsAppWasInstalledSrc144Evid200() {
        return new EventsFilter(this.util.getRawEvents, 144, 200, this.validate);
    }
    get wixPatternsItemCreatedWithAiSrc144Evid3() {
        return new EventsFilter(this.util.getRawEvents, 144, 3, this.validate);
    }
    get wixPatternsClickOnViewDropdownSrc144Evid4() {
        return new EventsFilter(this.util.getRawEvents, 144, 4, this.validate);
    }
    get wixPatternsMappingPagesSrc144Evid5() {
        return new EventsFilter(this.util.getRawEvents, 144, 5, this.validate);
    }
    get wixPatternsSelectAProviderSrc144Evid6() {
        return new EventsFilter(this.util.getRawEvents, 144, 6, this.validate);
    }
    get httpClientErrorMappingSrc176Evid100() {
        return new EventsFilter(this.util.getRawEvents, 176, 100, this.validate);
    }
    get getResolvedErrorSrc176Evid101() {
        return new EventsFilter(this.util.getRawEvents, 176, 101, this.validate);
    }
    get showErrorSrc176Evid102() {
        return new EventsFilter(this.util.getRawEvents, 176, 102, this.validate);
    }
    get retryErrorSrc176Evid103() {
        return new EventsFilter(this.util.getRawEvents, 176, 103, this.validate);
    }
    get httpClientErrorSrc176Evid404() {
        return new EventsFilter(this.util.getRawEvents, 176, 404, this.validate);
    }
    get pingMarketingConsentUnsubscribeClickSrc19Evid10001() {
        return new EventsFilter(this.util.getRawEvents, 19, 10001, this.validate);
    }
    get unsubscribeEmailPageView() {
        return new EventsFilter(this.util.getRawEvents, 19, 1006, this.validate);
    }
    get unsubscribeEmailPageClick() {
        return new EventsFilter(this.util.getRawEvents, 19, 1007, this.validate);
    }
    get unsubscribeEmailPersonalizePreferencesClick() {
        return new EventsFilter(this.util.getRawEvents, 19, 1008, this.validate);
    }
    get pushNotificationNotificationSentToAssure() {
        return new EventsFilter(this.util.getRawEvents, 19, 1121, this.validate);
    }
    get pushNotificationNotificationSentToDevice() {
        return new EventsFilter(this.util.getRawEvents, 19, 1122, this.validate);
    }
    get homeNotificationCreatedNew() {
        return new EventsFilter(this.util.getRawEvents, 19, 1125, this.validate);
    }
    get notificationsResolveAlertNotificationSrc19Evid1128() {
        return new EventsFilter(this.util.getRawEvents, 19, 1128, this.validate);
    }
    get pingNotificationCreationRequest() {
        return new EventsFilter(this.util.getRawEvents, 19, 1129, this.validate);
    }
    get pingNotificationSend() {
        return new EventsFilter(this.util.getRawEvents, 19, 1130, this.validate);
    }
    get pingNotificationFiltered() {
        return new EventsFilter(this.util.getRawEvents, 19, 1133, this.validate);
    }
    get androidNotificationReceivedSrc19Evid1134() {
        return new EventsFilter(this.util.getRawEvents, 19, 1134, this.validate);
    }
    get androidNotificationReceived() {
        return new EventsFilter(this.util.getRawEvents, 19, 1135, this.validate);
    }
    get pingReviewSendingList() {
        return new EventsFilter(this.util.getRawEvents, 19, 1136, this.validate);
    }
    get pingNotificationFilteredBeforeCreatedStage() {
        return new EventsFilter(this.util.getRawEvents, 19, 1137, this.validate);
    }
    get dummyNameSrc19Evid1256700() {
        return new EventsFilter(this.util.getRawEvents, 19, 1256700, this.validate);
    }
    get mobileAppSettingsChangeStatusOfNotificationCategoryInOsNotificationsSettings() {
        return new EventsFilter(this.util.getRawEvents, 19, 1333, this.validate);
    }
    get contactIsUnsubscribedFromEmailsSms() {
        return new EventsFilter(this.util.getRawEvents, 19, 20000, this.validate);
    }
    get smsStatusDeliveredToPingByTwilio() {
        return new EventsFilter(this.util.getRawEvents, 19, 40000, this.validate);
    }
    get errorEvent() {
        return new EventsFilter(this.util.getRawEvents, 5, 10, this.validate);
    }
    get dashboardDeleteAndRestoreSiteOpenPopup() {
        return new EventsFilter(this.util.getRawEvents, 5, 100, this.validate);
    }
    get pageClassifierSrc5Evid10000() {
        return new EventsFilter(this.util.getRawEvents, 5, 10000, this.validate);
    }
    get dashboardDeleteAndRestoreSiteRestoreSiteClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 101, this.validate);
    }
    get formBuilderFormSettingsContactListChooseSavedFieldsHoverYourFormFields() {
        return new EventsFilter(this.util.getRawEvents, 5, 1010, this.validate);
    }
    get formBuilderFormSettingsContactListChooseSavedFieldsHoverFieldsInContactList() {
        return new EventsFilter(this.util.getRawEvents, 5, 1011, this.validate);
    }
    get formBuilderFormSettingsContactListChooseSavedFieldsClickOnDropdownField() {
        return new EventsFilter(this.util.getRawEvents, 5, 1012, this.validate);
    }
    get formBuilderSubmitButtonSettingsUpdateValueOfSubmitButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 1013, this.validate);
    }
    get formBuilderFormSubmitSettingsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1014, this.validate);
    }
    get formBuilderFormSettingsDisplayErrorPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1015, this.validate);
    }
    get formBuilderFormSettingsErrorPageTryAgainClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1016, this.validate);
    }
    get formBuilderFormSettingsCreateEmailMarketing() {
        return new EventsFilter(this.util.getRawEvents, 5, 1017, this.validate);
    }
    get formBuilderFormSettingsSubscriberDoubleOptIn() {
        return new EventsFilter(this.util.getRawEvents, 5, 1018, this.validate);
    }
    get formBuilderFormSettingsSubscriberDoubleOptInLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1019, this.validate);
    }
    get dashboardSiteActivityCenterClickDeleteItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 102, this.validate);
    }
    get formBuilderFormSettingsSubscriberDoubleOptInEnable() {
        return new EventsFilter(this.util.getRawEvents, 5, 1020, this.validate);
    }
    get formBuilderFormSettingsSubscriberDoubleOptInSubmitMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1021, this.validate);
    }
    get formBuilderFormSettingsInboxOptInOutWhoToNotify() {
        return new EventsFilter(this.util.getRawEvents, 5, 1022, this.validate);
    }
    get formBuilderFormSettingsInboxOptInOutUpdateEmail() {
        return new EventsFilter(this.util.getRawEvents, 5, 1023, this.validate);
    }
    get formBuilderFormSettingsInboxOptInOutLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1024, this.validate);
    }
    get formBuilderFormSettingsInboxOptInOutAreYouSure() {
        return new EventsFilter(this.util.getRawEvents, 5, 1025, this.validate);
    }
    get formBuilderFormSettingsInboxOptInOutFindOutMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1026, this.validate);
    }
    get formBuilderFormSettingsEmailNotificationAddContributer() {
        return new EventsFilter(this.util.getRawEvents, 5, 1027, this.validate);
    }
    get formBuilderFormSettingsEmailNotificationAddNewEmail() {
        return new EventsFilter(this.util.getRawEvents, 5, 1028, this.validate);
    }
    get formBuilderFormSettingsEmailNotificationWantToNotifyDonTShowThisAgain() {
        return new EventsFilter(this.util.getRawEvents, 5, 1029, this.validate);
    }
    get dashboardSiteActivityCenterClickUndoDeletedItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 103, this.validate);
    }
    get workflowsSettingsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1030, this.validate);
    }
    get workflowsSettingsShowArchive() {
        return new EventsFilter(this.util.getRawEvents, 5, 1031, this.validate);
    }
    get workflowsSettingsHideArchive() {
        return new EventsFilter(this.util.getRawEvents, 5, 1032, this.validate);
    }
    get workflowsSettingsSetAutomations() {
        return new EventsFilter(this.util.getRawEvents, 5, 1033, this.validate);
    }
    get workflowsSettingsDeleteWorkflowClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1034, this.validate);
    }
    get workflowsSettingsDeleteWorkflowConfirmation() {
        return new EventsFilter(this.util.getRawEvents, 5, 1035, this.validate);
    }
    get workflowsSettingsAddNewWorkflow() {
        return new EventsFilter(this.util.getRawEvents, 5, 1036, this.validate);
    }
    get workflowsBoardUpdateBoardNameIconClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1037, this.validate);
    }
    get workflowsBoardUpdateBoardNameComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 1038, this.validate);
    }
    get workflowsBoardDropdownIconClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1039, this.validate);
    }
    get dashboardSiteActivityCenterClickOnArchiveItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 104, this.validate);
    }
    get workflowsAddNewPhaseClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1040, this.validate);
    }
    get workflowsAddNewPhaseSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 1041, this.validate);
    }
    get workflowsAddNewCardClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1042, this.validate);
    }
    get workflowsAddNewCardSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 1043, this.validate);
    }
    get workflowsPhaseSettingsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1044, this.validate);
    }
    get workflowsPhaseSettingsCreateAutomation() {
        return new EventsFilter(this.util.getRawEvents, 5, 1045, this.validate);
    }
    get workflowsPhaseSettingsRenameStepClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1046, this.validate);
    }
    get workflowsPhaseSettingsRenameStepComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 1047, this.validate);
    }
    get workflowsPhaseSettingsDeleteStepClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1048, this.validate);
    }
    get workflowsPhaseSettingsDeleteStepConfirm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1049, this.validate);
    }
    get dashboardUserClickedOnContactSNameFromUserActivity() {
        return new EventsFilter(this.util.getRawEvents, 5, 105, this.validate);
    }
    get workflowsCardSettingsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1050, this.validate);
    }
    get workflowsCardSettingsCardActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 1051, this.validate);
    }
    get workflowsCardSettingsEditCardNameComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 1052, this.validate);
    }
    get workflowsBoardSelectABoard() {
        return new EventsFilter(this.util.getRawEvents, 5, 1053, this.validate);
    }
    get workflowsSettingsLearnMoreClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1054, this.validate);
    }
    get workflowsIntroStep1WorkSmarterWithWorkflows() {
        return new EventsFilter(this.util.getRawEvents, 5, 1055, this.validate);
    }
    get workflowsIntroStep2HowItWorks() {
        return new EventsFilter(this.util.getRawEvents, 5, 1056, this.validate);
    }
    get workflowsIntroStep3MoveCardsAsYouGo() {
        return new EventsFilter(this.util.getRawEvents, 5, 1057, this.validate);
    }
    get workflowsIntroStep4SaveTimeWithAutomations() {
        return new EventsFilter(this.util.getRawEvents, 5, 1058, this.validate);
    }
    get workflowsIntroStep5CompleteSatisfaction() {
        return new EventsFilter(this.util.getRawEvents, 5, 1059, this.validate);
    }
    get clickOnSidebarSHeader() {
        return new EventsFilter(this.util.getRawEvents, 5, 106, this.validate);
    }
    get workflowsIntroStep6ThatSIt() {
        return new EventsFilter(this.util.getRawEvents, 5, 1060, this.validate);
    }
    get dummyNameSrc5Evid10600() {
        return new EventsFilter(this.util.getRawEvents, 5, 10600, this.validate);
    }
    get workflowsGoToBoardFromContactPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1061, this.validate);
    }
    get workflowsConnectContactToWorkflowFromContactPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1062, this.validate);
    }
    get workflowsClickUpdateContactInWfFromContactPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1063, this.validate);
    }
    get workflowsClickOnCompletedArchivedFlowsInContactPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1064, this.validate);
    }
    get workflowsErrorCardCannotBeRemovedInContactPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1065, this.validate);
    }
    get workflowsChangePhaseInContactPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1066, this.validate);
    }
    get workflowsClickOnSaveChangesInContactPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1067, this.validate);
    }
    get workflowsMoveCard() {
        return new EventsFilter(this.util.getRawEvents, 5, 1068, this.validate);
    }
    get workflowsEmptyStatePageLoaded() {
        return new EventsFilter(this.util.getRawEvents, 5, 1069, this.validate);
    }
    get sidebarState107() {
        return new EventsFilter(this.util.getRawEvents, 5, 107, this.validate);
    }
    get workflowsEmptyStateClickOnGetStarted() {
        return new EventsFilter(this.util.getRawEvents, 5, 1070, this.validate);
    }
    get workflowsTemplatesModalIsPresented() {
        return new EventsFilter(this.util.getRawEvents, 5, 1071, this.validate);
    }
    get workflowsClickOnUseTemplateOrStartFromScratchOnTemplatesModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 1072, this.validate);
    }
    get workflowsClickOnDismissTemplatesModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 1073, this.validate);
    }
    get cookiesBannerUouUserClickOnVisitorDataOption() {
        return new EventsFilter(this.util.getRawEvents, 5, 1074, this.validate);
    }
    get cookiesBannerUouUserHoverOverTooltip() {
        return new EventsFilter(this.util.getRawEvents, 5, 1075, this.validate);
    }
    get cookiesBannerUouCookieConsentBannerGetStartedClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1076, this.validate);
    }
    get cookiesBannerUouDisplayCookieBannerOnSiteCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 1077, this.validate);
    }
    get cookiesBannerUouUserClickedOnAdvancedOption() {
        return new EventsFilter(this.util.getRawEvents, 5, 1078, this.validate);
    }
    get cookiesBannerUouUserClickedOnTheGeneralXButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 1079, this.validate);
    }
    get mobileDashboardCreateNewSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 108, this.validate);
    }
    get cookiesBannerUouSavePublishClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1080, this.validate);
    }
    get cookiesBannerUouUserWithPopUpCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 1081, this.validate);
    }
    get cookiesBannerUouAdvancedOptionsCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 1082, this.validate);
    }
    get cookiesBannerUouSetUpAPrivacyPolicyClicks() {
        return new EventsFilter(this.util.getRawEvents, 5, 1083, this.validate);
    }
    get cookiesBannerUouAddADoNotSellDateLinkClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1084, this.validate);
    }
    get cookiesBannerUouUouClickOnDoNotSellMyDataLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 1085, this.validate);
    }
    get cookiesBannerUouUouDisableDataTransfersPopUpIsDisplay() {
        return new EventsFilter(this.util.getRawEvents, 5, 1086, this.validate);
    }
    get cookiesBannerUouUouWithDisableDataTransfersPopUp() {
        return new EventsFilter(this.util.getRawEvents, 5, 1087, this.validate);
    }
    get cookiesBannerUouUserConsentLogCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 1088, this.validate);
    }
    get cookiesBannerUouExportConsentLogCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 1089, this.validate);
    }
    get cookiesBannerUouAPopUpIsDisplay() {
        return new EventsFilter(this.util.getRawEvents, 5, 1090, this.validate);
    }
    get cookiesBannerUouSearchMembersById() {
        return new EventsFilter(this.util.getRawEvents, 5, 1091, this.validate);
    }
    get cookiesBannerUouReviewRequirementsAlert() {
        return new EventsFilter(this.util.getRawEvents, 5, 1092, this.validate);
    }
    get formBuilderFormAddNewFieldInnerTabsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1093, this.validate);
    }
    get formBuilderEditorXAddASubscribeCheckbox() {
        return new EventsFilter(this.util.getRawEvents, 5, 1094, this.validate);
    }
    get formBuilderMigratedFormContactFormToWixFormSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 1095, this.validate);
    }
    get formBuilderConditionalFormAddNewRule() {
        return new EventsFilter(this.util.getRawEvents, 5, 1096, this.validate);
    }
    get formBuilderConditionalFormLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1097, this.validate);
    }
    get formBuilderConditionalFormIfClickOnDropdown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1098, this.validate);
    }
    get formBuilderConditionalFormIfChooseFromDropdown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1099, this.validate);
    }
    get accountManagerStartLoadingAccountManager() {
        return new EventsFilter(this.util.getRawEvents, 5, 110, this.validate);
    }
    get formBuilderConditionalFormIfFilledNotFilled() {
        return new EventsFilter(this.util.getRawEvents, 5, 1100, this.validate);
    }
    get formBuilderConditionalFormIfFilledSelectionOption() {
        return new EventsFilter(this.util.getRawEvents, 5, 1101, this.validate);
    }
    get formBuilderConditionalFormThenClickOnDropdown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1102, this.validate);
    }
    get formBuilderConditionalFormThenChooseFromDropdown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1103, this.validate);
    }
    get formBuilderConditionalFormIsClickOnDropdown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1104, this.validate);
    }
    get formBuilderConditionalFormIsChooseFromDropdown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1105, this.validate);
    }
    get formBuilderConditionalFormSaveRule() {
        return new EventsFilter(this.util.getRawEvents, 5, 1106, this.validate);
    }
    get formBuilderConditionalFormMoreActionClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1107, this.validate);
    }
    get formBuilderConditionalFormMoreActionChoose() {
        return new EventsFilter(this.util.getRawEvents, 5, 1108, this.validate);
    }
    get formBuilderConditionalFormContradictingRuleWarning() {
        return new EventsFilter(this.util.getRawEvents, 5, 1109, this.validate);
    }
    get accountManagerFinishLoadingAccountManager() {
        return new EventsFilter(this.util.getRawEvents, 5, 111, this.validate);
    }
    get formBuilderConditionalFormSaveRuleError() {
        return new EventsFilter(this.util.getRawEvents, 5, 1110, this.validate);
    }
    get formBuilderMigratedFormContactFormToWixFormPhoneNumberIssue() {
        return new EventsFilter(this.util.getRawEvents, 5, 1111, this.validate);
    }
    get formBuilderConditionalFormViewRuleFromFieldSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1112, this.validate);
    }
    get formBuilderConditionalFormLearnMoreFromTooltipFieldSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1113, this.validate);
    }
    get crmHomeIntegrationsMarketingConnect() {
        return new EventsFilter(this.util.getRawEvents, 5, 1114, this.validate);
    }
    get crmHomeIntegrationsFinanceViewAll() {
        return new EventsFilter(this.util.getRawEvents, 5, 1115, this.validate);
    }
    get crmHomeIntegrationsFinanceConnect() {
        return new EventsFilter(this.util.getRawEvents, 5, 1116, this.validate);
    }
    get accountSettingsDisableSocialAccountConfirmationModalLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1119, this.validate);
    }
    get accountManagerPageComponentStartLoadingAccountManager() {
        return new EventsFilter(this.util.getRawEvents, 5, 112, this.validate);
    }
    get accountSettingsDisableSocialAccountClickDisableInConfirmationModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 1120, this.validate);
    }
    get crmHomeGoToMarketingTools() {
        return new EventsFilter(this.util.getRawEvents, 5, 1122, this.validate);
    }
    get formBuilderPaymentFormMoveStateInFunnel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1123, this.validate);
    }
    get formBuilderPaymentFormCloseFunnel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1124, this.validate);
    }
    get formBuilderPaymentFormQuestionMarkFunnel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1125, this.validate);
    }
    get formBuilderPaymentFormDemoSiteFunnel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1126, this.validate);
    }
    get formBuilderPaymentFormChoosePaymentFormType() {
        return new EventsFilter(this.util.getRawEvents, 5, 1127, this.validate);
    }
    get formBuilderPaymentFormEnterItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 1128, this.validate);
    }
    get formBuilderPaymentFormEnterPrice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1129, this.validate);
    }
    get accountManagerPageComponentFinishLoadingAccountManager() {
        return new EventsFilter(this.util.getRawEvents, 5, 113, this.validate);
    }
    get dummyNameSrc5Evid1130() {
        return new EventsFilter(this.util.getRawEvents, 5, 1130, this.validate);
    }
    get dummyNameSrc5Evid1131() {
        return new EventsFilter(this.util.getRawEvents, 5, 1131, this.validate);
    }
    get notificationInvalidTokens() {
        return new EventsFilter(this.util.getRawEvents, 5, 1132, this.validate);
    }
    get formBuilderPaymentFormHoverPrice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1133, this.validate);
    }
    get formBuilderPaymentFormDeleteItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 1134, this.validate);
    }
    get formBuilderPaymentFormAddAnotherItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 1135, this.validate);
    }
    get formBuilderPaymentFormHoverDeleteItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 1136, this.validate);
    }
    get formBuilderPaymentFormReachLimitOfItems() {
        return new EventsFilter(this.util.getRawEvents, 5, 1137, this.validate);
    }
    get formBuilderPaymentFormTellUsMoreItems() {
        return new EventsFilter(this.util.getRawEvents, 5, 1138, this.validate);
    }
    get formBuilderPaymentFormEnterMinMaxPrice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1139, this.validate);
    }
    get accountManagerClickOnSidebarItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 114, this.validate);
    }
    get formBuilderPaymentFormConnectPaymentMethod() {
        return new EventsFilter(this.util.getRawEvents, 5, 1140, this.validate);
    }
    get formBuilderPaymentFormLearnMoreStep4() {
        return new EventsFilter(this.util.getRawEvents, 5, 1141, this.validate);
    }
    get formBuilderPaymentFormDashboardContinueGreenBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 1142, this.validate);
    }
    get formBuilderPaymentFormDashboardCloseGreenBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 1143, this.validate);
    }
    get formBuilderPaymentFormChangePaymentMethod() {
        return new EventsFilter(this.util.getRawEvents, 5, 1145, this.validate);
    }
    get formBuilderPaymentFormDiscardFunnel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1146, this.validate);
    }
    get formBuilderPaymentFormPaymentSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1147, this.validate);
    }
    get formBuilderPaymentFormOpenPaymentOverview() {
        return new EventsFilter(this.util.getRawEvents, 5, 1148, this.validate);
    }
    get formBuilderPaymentFormGeneralSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1149, this.validate);
    }
    get accountManagerClickOnBackInSidebarItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 115, this.validate);
    }
    get formBuilderPaymentFormSettingsAddAnotherItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 1150, this.validate);
    }
    get formBuilderPaymentFormSettingsItemMoreActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 1151, this.validate);
    }
    get formBuilderPaymentFormSettingsItemMoreActionsChoose() {
        return new EventsFilter(this.util.getRawEvents, 5, 1152, this.validate);
    }
    get formBuilderPaymentFormSettingsConnectPaymentWhenPaymentNotConnected() {
        return new EventsFilter(this.util.getRawEvents, 5, 1153, this.validate);
    }
    get formBuilderPaymentFormSettingsRestoreField() {
        return new EventsFilter(this.util.getRawEvents, 5, 1154, this.validate);
    }
    get formBuilderPaymentFormSettingsStopAcceptingPayments() {
        return new EventsFilter(this.util.getRawEvents, 5, 1155, this.validate);
    }
    get settingsLobbySectionIsSeen() {
        return new EventsFilter(this.util.getRawEvents, 5, 1156, this.validate);
    }
    get settingsLobbyClickOnItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 1158, this.validate);
    }
    get formBuilderSubmissionLimitReasonForLimit() {
        return new EventsFilter(this.util.getRawEvents, 5, 1159, this.validate);
    }
    get foldersMarkUnmarkSelectAllButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 116, this.validate);
    }
    get formBuilderSubmissionLimitReasonForLimitInfoHover() {
        return new EventsFilter(this.util.getRawEvents, 5, 1160, this.validate);
    }
    get reportNetwork() {
        return new EventsFilter(this.util.getRawEvents, 5, 11601, this.validate);
    }
    get formBuilderSubmissionLimitLimitByNumberTotalNumber() {
        return new EventsFilter(this.util.getRawEvents, 5, 1161, this.validate);
    }
    get formBuilderSubmissionLimitLimitByNumberEditMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1162, this.validate);
    }
    get formBuilderSubmissionLimitLimitByNumberNumberInfoHover() {
        return new EventsFilter(this.util.getRawEvents, 5, 1163, this.validate);
    }
    get formBuilderSubmissionLimitLimitByNumberMessageInfoHover() {
        return new EventsFilter(this.util.getRawEvents, 5, 1164, this.validate);
    }
    get formBuilderSubmissionLimitLimitByDateEnteredDate() {
        return new EventsFilter(this.util.getRawEvents, 5, 1165, this.validate);
    }
    get formBuilderSubmissionLimitLimitByDateChangeTimeFormat() {
        return new EventsFilter(this.util.getRawEvents, 5, 1166, this.validate);
    }
    get formBuilderSubmissionLimitLimitByDateChangeTimeFormatInvalid() {
        return new EventsFilter(this.util.getRawEvents, 5, 1167, this.validate);
    }
    get formBuilderSubmissionLimitLimitByDateChangeDefaultTime() {
        return new EventsFilter(this.util.getRawEvents, 5, 1168, this.validate);
    }
    get formBuilderSubmissionLimitLimitByDateChangeMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1169, this.validate);
    }
    get foldersClickOnChangeViewMode() {
        return new EventsFilter(this.util.getRawEvents, 5, 117, this.validate);
    }
    get formBuilderSubmissionLimitLimitByDateTimezoneSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1170, this.validate);
    }
    get formBuilderSubmissionLimitRestoreSuccessMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1171, this.validate);
    }
    get viewInDashboard() {
        return new EventsFilter(this.util.getRawEvents, 5, 1172, this.validate);
    }
    get formBuilderAutofillMemberEmailsEnableToggle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1173, this.validate);
    }
    get formBuilderAutofillMemberEmailsMembersAreaLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1174, this.validate);
    }
    get formBuilderAutofillMemberEmailsBrowserAutocompleteLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1175, this.validate);
    }
    get formBuilderAutofillMemberEmailsMainTooltipHover() {
        return new EventsFilter(this.util.getRawEvents, 5, 1176, this.validate);
    }
    get formBuilderAutofillMemberEmailsLetMembersEdit() {
        return new EventsFilter(this.util.getRawEvents, 5, 1177, this.validate);
    }
    get formBuilderAutofillMemberEmailsDropdownEmail() {
        return new EventsFilter(this.util.getRawEvents, 5, 1178, this.validate);
    }
    get formBuilderAutofillMemberEmailsChooseDifferentEmail() {
        return new EventsFilter(this.util.getRawEvents, 5, 1179, this.validate);
    }
    get foldersTooltipShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 118, this.validate);
    }
    get formBuilderAutofillMemberEmailsAddAnEmailField() {
        return new EventsFilter(this.util.getRawEvents, 5, 1180, this.validate);
    }
    get veloIdeModalIsOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 11801, this.validate);
    }
    get veloIdeCodeHasBeenUpdated() {
        return new EventsFilter(this.util.getRawEvents, 5, 11802, this.validate);
    }
    get veloIdeClickOnMoreActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 11803, this.validate);
    }
    get veloIdeClickOnATab() {
        return new EventsFilter(this.util.getRawEvents, 5, 11804, this.validate);
    }
    get veloIdeModalEndOfProcess() {
        return new EventsFilter(this.util.getRawEvents, 5, 11805, this.validate);
    }
    get veloIdeModalClickOnLinks() {
        return new EventsFilter(this.util.getRawEvents, 5, 11806, this.validate);
    }
    get veloIdeModalUnsuccessfulSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 11807, this.validate);
    }
    get formBuilderAutofillMemberEmailsAddedMembersArea() {
        return new EventsFilter(this.util.getRawEvents, 5, 1181, this.validate);
    }
    get formBuilderSubmissionLimitUouSeenLimitMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1182, this.validate);
    }
    get sidebarModalOfFteToTheNewSidebarIsLoadedWithSpecificCluster() {
        return new EventsFilter(this.util.getRawEvents, 5, 1185, this.validate);
    }
    get dashboardSiteSettingsLocationsModalPageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 1189, this.validate);
    }
    get foldersClickOnShowMoreShoeLess() {
        return new EventsFilter(this.util.getRawEvents, 5, 119, this.validate);
    }
    get dashboardUpgradeLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 12, this.validate);
    }
    get ascendPagePageNavigationClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 120, this.validate);
    }
    get workspaceSelectorSelectWorkspaceModalView() {
        return new EventsFilter(this.util.getRawEvents, 5, 12002, this.validate);
    }
    get workspaceSelectorSelectWorkspaceModalCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 12003, this.validate);
    }
    get workspaceSelectorSelectWorkspaceModalClose() {
        return new EventsFilter(this.util.getRawEvents, 5, 12004, this.validate);
    }
    get switchWorkspaceModalSelectWorkspaceModalView() {
        return new EventsFilter(this.util.getRawEvents, 5, 12010, this.validate);
    }
    get switchWorkspaceModalSelectWorkspaceModalCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 12011, this.validate);
    }
    get switchWorkspaceModalWorkspaceModalClose() {
        return new EventsFilter(this.util.getRawEvents, 5, 12012, this.validate);
    }
    get formBuilderBizMgrFormsBizTipsClickOnTheTip() {
        return new EventsFilter(this.util.getRawEvents, 5, 1208, this.validate);
    }
    get invoicesAddAProductVariantOpenModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 1209, this.validate);
    }
    get ascendPageAscendVideoClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 121, this.validate);
    }
    get invoicesAddAProductVariantAddToInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1210, this.validate);
    }
    get foldersHoverOverThePermissionIcon1212() {
        return new EventsFilter(this.util.getRawEvents, 5, 1212, this.validate);
    }
    get invoicesCreateOrderNewInvoiceClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1213, this.validate);
    }
    get dashboardFinishLoadingLoadingTimeFromStartModule() {
        return new EventsFilter(this.util.getRawEvents, 5, 1215, this.validate);
    }
    get sitesListPageFoldersLoaded1216() {
        return new EventsFilter(this.util.getRawEvents, 5, 1216, this.validate);
    }
    get deleteLocationModalIsOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 1217, this.validate);
    }
    get successfullyDeletedLocation() {
        return new EventsFilter(this.util.getRawEvents, 5, 1218, this.validate);
    }
    get foldersClickOnCreateNewFolder() {
        return new EventsFilter(this.util.getRawEvents, 5, 122, this.validate);
    }
    get locationModalFloatingNotification() {
        return new EventsFilter(this.util.getRawEvents, 5, 1225, this.validate);
    }
    get foldersClickOnCreateNewFolderInDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 123, this.validate);
    }
    get apesSaveEditedNotification() {
        return new EventsFilter(this.util.getRawEvents, 5, 1234, this.validate);
    }
    get apesSaveEditedNotificationSrc5Evid1235() {
        return new EventsFilter(this.util.getRawEvents, 5, 1235, this.validate);
    }
    get apesEditNotificationSelectAudience() {
        return new EventsFilter(this.util.getRawEvents, 5, 1238, this.validate);
    }
    get foldersClickOnMoveToFolderFromDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 124, this.validate);
    }
    get suggestionsFeedbackClickOnNavigationButtonInPopup() {
        return new EventsFilter(this.util.getRawEvents, 5, 1240, this.validate);
    }
    get formBuilderBizMgrClickOnMoreActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 1243, this.validate);
    }
    get formBuilderBizMgrClickOnMoreActionsOfASpecificForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1246, this.validate);
    }
    get formBuilderBizMgrClickOnViewForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1247, this.validate);
    }
    get formBuilderBizMgrClickOnEditForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1248, this.validate);
    }
    get formBuilderBizMgrClickOnDeleteForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1249, this.validate);
    }
    get foldersClickOnFolderSQuickActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 125, this.validate);
    }
    get customerNotificationsSettingsClickInsidePageWithoutEmailsSections() {
        return new EventsFilter(this.util.getRawEvents, 5, 1250, this.validate);
    }
    get customerNotificationsSettingsClickOnHideUnhide() {
        return new EventsFilter(this.util.getRawEvents, 5, 1251, this.validate);
    }
    get customerNotificationsSettingsPreviewEditClickOnToggle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1252, this.validate);
    }
    get customerNotificationsSettingsPageLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1253, this.validate);
    }
    get bizManagerShowWarningMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1254, this.validate);
    }
    get formBuilderBizMgrClickOnGoToEditorToPublishYourSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 1256, this.validate);
    }
    get formBuilderBizMgrClickOnGoToEditorToCreateSubmissionTable() {
        return new EventsFilter(this.util.getRawEvents, 5, 1257, this.validate);
    }
    get copyOfFormBuilderBizMgrClickOnGoToEditorToRemoveForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1258, this.validate);
    }
    get customerNotificationsSettingsPreviewEditShowTooltip() {
        return new EventsFilter(this.util.getRawEvents, 5, 1259, this.validate);
    }
    get foldersApproveActionFromFolderSQuickActionPopup() {
        return new EventsFilter(this.util.getRawEvents, 5, 126, this.validate);
    }
    get formBuilderBizMgrEmptyStateClickOnWatchVideo() {
        return new EventsFilter(this.util.getRawEvents, 5, 1263, this.validate);
    }
    get formBuilderBizMgrClickOnLoadMoreForms() {
        return new EventsFilter(this.util.getRawEvents, 5, 1264, this.validate);
    }
    get formBuilderBizMgrMoreActionsViewRemovedForms() {
        return new EventsFilter(this.util.getRawEvents, 5, 1265, this.validate);
    }
    get formBuilderBizMgrMoreActionsLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1266, this.validate);
    }
    get formBuilderBizMgrMoreActionsSendFeedback() {
        return new EventsFilter(this.util.getRawEvents, 5, 1267, this.validate);
    }
    get sidebarTooltipOfModalOfFteToTheNewSidebarIsClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 1268, this.validate);
    }
    get foldersSortListView() {
        return new EventsFilter(this.util.getRawEvents, 5, 127, this.validate);
    }
    get verticalsEmailNotificationSentSrc5Evid1270() {
        return new EventsFilter(this.util.getRawEvents, 5, 1270, this.validate);
    }
    get verticalsEmailNotificationSent() {
        return new EventsFilter(this.util.getRawEvents, 5, 1271, this.validate);
    }
    get verticalsEmailNotificationClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 1272, this.validate);
    }
    get sidebarTooltipOfModalOfFteToTheNewSidebarIsShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1274, this.validate);
    }
    get sidebarModalOfFteToTheNewSidebarIsClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 1276, this.validate);
    }
    get apesLoadSystemNotificationsModalEnd() {
        return new EventsFilter(this.util.getRawEvents, 5, 1277, this.validate);
    }
    get accountSettingsCloseYourWixAccountEnterPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1278, this.validate);
    }
    get accountSettingsCloseYourWixAccountCloseAccountRequestSubmited() {
        return new EventsFilter(this.util.getRawEvents, 5, 1279, this.validate);
    }
    get foldersMarkASite() {
        return new EventsFilter(this.util.getRawEvents, 5, 128, this.validate);
    }
    get formBuilderComplexPhoneSettingsShowCountryCodeDropdownToggle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1280, this.validate);
    }
    get formBuilderComplexPhoneSettingsShowFieldTitleToggle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1281, this.validate);
    }
    get formBuilderComplexPhoneSettingsEnterFieldTitle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1282, this.validate);
    }
    get formBuilderComplexPhoneSettingsShowInitialTextOfThePhoneFieldSelection() {
        return new EventsFilter(this.util.getRawEvents, 5, 1283, this.validate);
    }
    get formBuilderComplexPhoneSettingsPlaceHolderTextOfThePhoneField() {
        return new EventsFilter(this.util.getRawEvents, 5, 1284, this.validate);
    }
    get formBuilderComplexPhoneSettingsIsFieldRequired() {
        return new EventsFilter(this.util.getRawEvents, 5, 1285, this.validate);
    }
    get formBuilderComplexPhoneSettingsCountryCodeEnterFieldTitle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1286, this.validate);
    }
    get formBuilderComplexPhoneSettingsCountryCodeIncludeCodesFrom() {
        return new EventsFilter(this.util.getRawEvents, 5, 1287, this.validate);
    }
    get formBuilderComplexPhoneSettingsCountryCodeShowInitialTextSelection() {
        return new EventsFilter(this.util.getRawEvents, 5, 1289, this.validate);
    }
    get myAccountSelectSiteFromSiteListPopup() {
        return new EventsFilter(this.util.getRawEvents, 5, 129, this.validate);
    }
    get formBuilderComplexPhoneSettingsCountryCodeShowInitialTextValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 1290, this.validate);
    }
    get formBuilderComplexPhoneSettingsSelectNumberFormat() {
        return new EventsFilter(this.util.getRawEvents, 5, 1291, this.validate);
    }
    get formBuilderComplexPhoneSettingsEnteredNumberFormat() {
        return new EventsFilter(this.util.getRawEvents, 5, 1292, this.validate);
    }
    get formBuilderComplexPhoneSettingsCountryCodeClickOnManageCodeList() {
        return new EventsFilter(this.util.getRawEvents, 5, 1295, this.validate);
    }
    get formBuilderComplexPhoneEditFieldClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1299, this.validate);
    }
    get dashboardSearch() {
        return new EventsFilter(this.util.getRawEvents, 5, 13, this.validate);
    }
    get foldersClickOnFolder() {
        return new EventsFilter(this.util.getRawEvents, 5, 130, this.validate);
    }
    get crmHomeScrollDown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1300, this.validate);
    }
    get financialServicesSettingsChangeTab() {
        return new EventsFilter(this.util.getRawEvents, 5, 1301, this.validate);
    }
    get financialServicesSettingsInvoicesClickCard() {
        return new EventsFilter(this.util.getRawEvents, 5, 1302, this.validate);
    }
    get financialServicesSettingsInvoicesEditNumbering() {
        return new EventsFilter(this.util.getRawEvents, 5, 1303, this.validate);
    }
    get financialServicesSettingsInvoicesEditLogo() {
        return new EventsFilter(this.util.getRawEvents, 5, 1304, this.validate);
    }
    get financialServicesSettingsInvoicesBusinessInfoEditThem() {
        return new EventsFilter(this.util.getRawEvents, 5, 1305, this.validate);
    }
    get financialServicesSettingsInvoicesBusinessInfoEditField() {
        return new EventsFilter(this.util.getRawEvents, 5, 1306, this.validate);
    }
    get financialServicesSettingsInvoicesDefaultsNotesEditNotes() {
        return new EventsFilter(this.util.getRawEvents, 5, 1307, this.validate);
    }
    get financialServicesSettingsInvoicesDefaultsNotesLegalTerms() {
        return new EventsFilter(this.util.getRawEvents, 5, 1308, this.validate);
    }
    get financialServicesSettingsInvoicesDefaultsNotesPreviewInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1309, this.validate);
    }
    get ascendPpPageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 131, this.validate);
    }
    get financialServicesSettingsPriceQuotesBusinessDetailsSiteSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1310, this.validate);
    }
    get financialServicesSettingsPriceQuotesBusinessDetailsGoToSiteSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1311, this.validate);
    }
    get financialServicesSettingsPriceQuotesEditNumbering() {
        return new EventsFilter(this.util.getRawEvents, 5, 1312, this.validate);
    }
    get financialServicesSettingsInvoicesSettingsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1313, this.validate);
    }
    get crmHomeGeneralVideo() {
        return new EventsFilter(this.util.getRawEvents, 5, 1314, this.validate);
    }
    get invoicesItemsListChooseAProduct() {
        return new EventsFilter(this.util.getRawEvents, 5, 1315, this.validate);
    }
    get invoicesItemsListAddNewProduct() {
        return new EventsFilter(this.util.getRawEvents, 5, 1316, this.validate);
    }
    get invoicesItemsListLinkInTooltipProducts() {
        return new EventsFilter(this.util.getRawEvents, 5, 1317, this.validate);
    }
    get ascendPpClickOnShowMoreLess() {
        return new EventsFilter(this.util.getRawEvents, 5, 132, this.validate);
    }
    get invoicesWidgetMoreAction() {
        return new EventsFilter(this.util.getRawEvents, 5, 1320, this.validate);
    }
    get invoicesWidgetMoreActionChooseAnAction() {
        return new EventsFilter(this.util.getRawEvents, 5, 1321, this.validate);
    }
    get invoicesWidgetResendViaEmail() {
        return new EventsFilter(this.util.getRawEvents, 5, 1322, this.validate);
    }
    get invoicesWidgetChangeTab() {
        return new EventsFilter(this.util.getRawEvents, 5, 1323, this.validate);
    }
    get invoicesWidgetCreateNew() {
        return new EventsFilter(this.util.getRawEvents, 5, 1324, this.validate);
    }
    get invoicesWidgetPreviewInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1325, this.validate);
    }
    get invoicesWidgetSeeAllInvoices() {
        return new EventsFilter(this.util.getRawEvents, 5, 1326, this.validate);
    }
    get invoicesWidgetReportsViewReport() {
        return new EventsFilter(this.util.getRawEvents, 5, 1327, this.validate);
    }
    get invoicesWidgetReportsTimePicker() {
        return new EventsFilter(this.util.getRawEvents, 5, 1328, this.validate);
    }
    get invoicesEditModesEditInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1329, this.validate);
    }
    get ascendPpScrollToEnd() {
        return new EventsFilter(this.util.getRawEvents, 5, 133, this.validate);
    }
    get invoicesEditModesEditCustomerInfoModalChoose() {
        return new EventsFilter(this.util.getRawEvents, 5, 1330, this.validate);
    }
    get invoicesEditModesNegativePriceToast() {
        return new EventsFilter(this.util.getRawEvents, 5, 1331, this.validate);
    }
    get clickOnWixStudioLogoSrc5Evid1332() {
        return new EventsFilter(this.util.getRawEvents, 5, 1332, this.validate);
    }
    get crmDiscoverCrmTools() {
        return new EventsFilter(this.util.getRawEvents, 5, 1333, this.validate);
    }
    get formBuilderComplexPhoneLayoutsDesignSelected() {
        return new EventsFilter(this.util.getRawEvents, 5, 1334, this.validate);
    }
    get formBuilderComplexAddressShowTheseSubfields() {
        return new EventsFilter(this.util.getRawEvents, 5, 1337, this.validate);
    }
    get formBuilderComplexAddressIncludeCountriesFrom() {
        return new EventsFilter(this.util.getRawEvents, 5, 1338, this.validate);
    }
    get formBuilderComplexAddressShowSubfieldTitles() {
        return new EventsFilter(this.util.getRawEvents, 5, 1339, this.validate);
    }
    get ascendPpClickOnPackageInPp() {
        return new EventsFilter(this.util.getRawEvents, 5, 134, this.validate);
    }
    get formBuilderComplexAddressSubfieldTitles() {
        return new EventsFilter(this.util.getRawEvents, 5, 1340, this.validate);
    }
    get formBuilderComplexAddressSubfieldPlaceholder() {
        return new EventsFilter(this.util.getRawEvents, 5, 1341, this.validate);
    }
    get formBuilderComplexAddressGeneralSettingRequired() {
        return new EventsFilter(this.util.getRawEvents, 5, 1342, this.validate);
    }
    get formBuilderComplexAddressSubfieldTitlesEnterText() {
        return new EventsFilter(this.util.getRawEvents, 5, 1343, this.validate);
    }
    get formBuilderComplexAddressSubfieldPlaceholdersEnterText() {
        return new EventsFilter(this.util.getRawEvents, 5, 1344, this.validate);
    }
    get formBuilderComplexAddressSubfieldPlaceholdersDefaultCountryChoose() {
        return new EventsFilter(this.util.getRawEvents, 5, 1345, this.validate);
    }
    get formBuilderComplexAddressManageCountriesList() {
        return new EventsFilter(this.util.getRawEvents, 5, 1346, this.validate);
    }
    get formBuilderComplexAddressManageCountriesListFinish() {
        return new EventsFilter(this.util.getRawEvents, 5, 1347, this.validate);
    }
    get formBuilderComplexAddressSelectRequiredSubfieldOpen() {
        return new EventsFilter(this.util.getRawEvents, 5, 1348, this.validate);
    }
    get formBuilderComplexAddressSelectRequiredSubfieldChoose() {
        return new EventsFilter(this.util.getRawEvents, 5, 1349, this.validate);
    }
    get ascendPpCyclePageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 135, this.validate);
    }
    get formBuilderAutocompleteAddressFieldShowFieldTitleToggle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1353, this.validate);
    }
    get formBuilderAutocompleteAddressFieldFieldTitle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1354, this.validate);
    }
    get formBuilderAutocompleteAddressFieldPlaceholderText() {
        return new EventsFilter(this.util.getRawEvents, 5, 1356, this.validate);
    }
    get formBuilderAutocompleteAddressFieldRequired() {
        return new EventsFilter(this.util.getRawEvents, 5, 1357, this.validate);
    }
    get formBuilderAutocompleteAddressFieldCountrySearch() {
        return new EventsFilter(this.util.getRawEvents, 5, 1358, this.validate);
    }
    get formBuilderAutocompleteAddressFieldCountrySearchOption() {
        return new EventsFilter(this.util.getRawEvents, 5, 1359, this.validate);
    }
    get ascendPpClickOnCycleInPp() {
        return new EventsFilter(this.util.getRawEvents, 5, 136, this.validate);
    }
    get formBuilderAutocompleteAddressFieldAddressLayoutChanged() {
        return new EventsFilter(this.util.getRawEvents, 5, 1360, this.validate);
    }
    get formBuilderAutocompleteAddressFieldViewerChooseAnAddress() {
        return new EventsFilter(this.util.getRawEvents, 5, 1363, this.validate);
    }
    get ascendPpClickOnNextInCyclePage() {
        return new EventsFilter(this.util.getRawEvents, 5, 137, this.validate);
    }
    get customerNotificationsSettingsPreviewEditChooseAudience() {
        return new EventsFilter(this.util.getRawEvents, 5, 1375, this.validate);
    }
    get formBuilderComplexPhoneCountryCodeFieldDeleted() {
        return new EventsFilter(this.util.getRawEvents, 5, 1376, this.validate);
    }
    get dashboardSiteSelectorIsOpen() {
        return new EventsFilter(this.util.getRawEvents, 5, 1377, this.validate);
    }
    get dashboardSiteSelectorIsLoadedWithoutScreenshots() {
        return new EventsFilter(this.util.getRawEvents, 5, 1378, this.validate);
    }
    get dashboardSiteSelectorErrorLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 1379, this.validate);
    }
    get ascendPpPurchasePageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 138, this.validate);
    }
    get dashboardSiteSelectorClickOnSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 1380, this.validate);
    }
    get dashboardClickInSiteSelector() {
        return new EventsFilter(this.util.getRawEvents, 5, 1381, this.validate);
    }
    get formBuilderComplexAddressEditFieldClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1382, this.validate);
    }
    get crmHomeInboxViewMessages() {
        return new EventsFilter(this.util.getRawEvents, 5, 1383, this.validate);
    }
    get crmHomeTopCarouselClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1384, this.validate);
    }
    get crmHomeTopCarouselMainCtaClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1385, this.validate);
    }
    get crmHomeTopCarouselLoadStart() {
        return new EventsFilter(this.util.getRawEvents, 5, 1386, this.validate);
    }
    get crmHomeTopCarouselLoadEnd() {
        return new EventsFilter(this.util.getRawEvents, 5, 1387, this.validate);
    }
    get formBuilderReportOnSubmissionAsSpam() {
        return new EventsFilter(this.util.getRawEvents, 5, 1389, this.validate);
    }
    get ascendPpPurchasePageChangeCard() {
        return new EventsFilter(this.util.getRawEvents, 5, 139, this.validate);
    }
    get ascendSelectCancelReason() {
        return new EventsFilter(this.util.getRawEvents, 5, 1390, this.validate);
    }
    get ascendCancelWindowApproveCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1391, this.validate);
    }
    get bmApiCallReceivedSrc5Evid1392() {
        return new EventsFilter(this.util.getRawEvents, 5, 1392, this.validate);
    }
    get ascendPpPurchasePageEnterPromo() {
        return new EventsFilter(this.util.getRawEvents, 5, 140, this.validate);
    }
    get rolesAndPermissionsSetUpTeamClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1400, this.validate);
    }
    get teamManagementTeammatesPageLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1401, this.validate);
    }
    get teamManagementFilterByStatus() {
        return new EventsFilter(this.util.getRawEvents, 5, 1402, this.validate);
    }
    get teamManagementFilterByRole() {
        return new EventsFilter(this.util.getRawEvents, 5, 1403, this.validate);
    }
    get teamManagementViewTeamClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 1404, this.validate);
    }
    get teamManagementChangeRoleOpenDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 1405, this.validate);
    }
    get teamManagementChangeRoleUpdate() {
        return new EventsFilter(this.util.getRawEvents, 5, 1406, this.validate);
    }
    get teamManagementRemoveTeammateOpenDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 1407, this.validate);
    }
    get teamManagementRemoveTeammateRemoveSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 1408, this.validate);
    }
    get teamManagementInvitePeopleClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1409, this.validate);
    }
    get ascendPpPurchasePageSubmitPurchase() {
        return new EventsFilter(this.util.getRawEvents, 5, 141, this.validate);
    }
    get teamManagementInvitePeoplePageLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1410, this.validate);
    }
    get teamManagementInvitePeopleSendInvite() {
        return new EventsFilter(this.util.getRawEvents, 5, 1411, this.validate);
    }
    get teamManagementInvitePeopleResendInvite() {
        return new EventsFilter(this.util.getRawEvents, 5, 1412, this.validate);
    }
    get teamManagementInvitePeopleGetInviteLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 1413, this.validate);
    }
    get teamManagementManageRolesClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1414, this.validate);
    }
    get teamManagementManageRolesPageLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1415, this.validate);
    }
    get teamManagementManageRolesViewRole() {
        return new EventsFilter(this.util.getRawEvents, 5, 1416, this.validate);
    }
    get teamManagementManageRolesRolePageLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1417, this.validate);
    }
    get teamManagementManageRolesDuplicateAndEditRole() {
        return new EventsFilter(this.util.getRawEvents, 5, 1418, this.validate);
    }
    get teamManagementManageRolesEditCustomRole() {
        return new EventsFilter(this.util.getRawEvents, 5, 1419, this.validate);
    }
    get teamManagementManageRolesUpdateRole() {
        return new EventsFilter(this.util.getRawEvents, 5, 1420, this.validate);
    }
    get teamManagementManageRolesCreateNewRoleClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1421, this.validate);
    }
    get teamManagementManageRolesSaveNewRole() {
        return new EventsFilter(this.util.getRawEvents, 5, 1422, this.validate);
    }
    get teamManagementManageRolesDeleteCustomRoleClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1423, this.validate);
    }
    get teamManagementManageRolesDeleteCustomRoleSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 1424, this.validate);
    }
    get teamManagementManageRolesCannotDeleteRole() {
        return new EventsFilter(this.util.getRawEvents, 5, 1425, this.validate);
    }
    get teamManagementLearnMoreClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1426, this.validate);
    }
    get teamManagementEditTeamProfileClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1427, this.validate);
    }
    get teamManagementSaveTeamProfile() {
        return new EventsFilter(this.util.getRawEvents, 5, 1428, this.validate);
    }
    get teamManagementJoinTheTeamPageLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1429, this.validate);
    }
    get teamManagementJoinTheTeamClickContinue() {
        return new EventsFilter(this.util.getRawEvents, 5, 1430, this.validate);
    }
    get teamManagementJoinTheTeamClickOnLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1431, this.validate);
    }
    get rolesAndPermissionsClickOnUpgradeButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 1432, this.validate);
    }
    get rolesAndPermissionsInvitePeopleError() {
        return new EventsFilter(this.util.getRawEvents, 5, 1433, this.validate);
    }
    get rolesAndPermissionsUpgradePopUpLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1434, this.validate);
    }
    get rolesAndPermissionsAcceptInviteFromMobilePageLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1435, this.validate);
    }
    get rolesAndPermissionsAcceptInviteFromMobileClickContinueToSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 1436, this.validate);
    }
    get rolesAndPermissionsAcceptInviteFromMobileClickManageSiteOnTheWixApp() {
        return new EventsFilter(this.util.getRawEvents, 5, 1437, this.validate);
    }
    get rolesAndPermissionsInvitePeopleAnswerOnInvitePurpose() {
        return new EventsFilter(this.util.getRawEvents, 5, 1438, this.validate);
    }
    get rolesAndPermissionsInvitePeopleSkipOnInvitePurposeQuestion() {
        return new EventsFilter(this.util.getRawEvents, 5, 1439, this.validate);
    }
    get rolesAndPermissionsSitesAndFoldersSelectSitesAndFoldersModalLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1440, this.validate);
    }
    get rolesAndPermissionsSitesAndFoldersClickOnSelectInSfModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 1441, this.validate);
    }
    get rolesAndPermissionsSitesAndFoldersClickOnSelectMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1442, this.validate);
    }
    get rolesAndPermissionsSitesAndFoldersClickOnShowMoreSites() {
        return new EventsFilter(this.util.getRawEvents, 5, 1443, this.validate);
    }
    get rolesAndPermissionsSitesAndFoldersRemoveSiteFolderClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1444, this.validate);
    }
    get rolesAndPermissionsSitesAndFoldersClickOnAssignMoreRoles() {
        return new EventsFilter(this.util.getRawEvents, 5, 1445, this.validate);
    }
    get rolesAndPermissionsSitesAndFoldersSiteRpWithTeamModalLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1446, this.validate);
    }
    get rolesAndPermissionsSitesAndFoldersChooseActionInSiteRpWithTeamModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 1447, this.validate);
    }
    get rolesAndPermissionsPeoplePermissionsComponentLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1448, this.validate);
    }
    get rolesAndPermissionsPeoplePermissionsComponentClickOnManage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1449, this.validate);
    }
    get rolesAndPermissionsTeamManagementSearchText() {
        return new EventsFilter(this.util.getRawEvents, 5, 1450, this.validate);
    }
    get headerLoaded() {
        return new EventsFilter(this.util.getRawEvents, 5, 1452, this.validate);
    }
    get formBuilderCaptchaBoxModalLoadedClient() {
        return new EventsFilter(this.util.getRawEvents, 5, 1453, this.validate);
    }
    get formBuilderCaptchaBoxModalClosedClient() {
        return new EventsFilter(this.util.getRawEvents, 5, 1454, this.validate);
    }
    get customerNotificationsSettingsPreviewEditLoadingTime() {
        return new EventsFilter(this.util.getRawEvents, 5, 1455, this.validate);
    }
    get settingsLobbyHoverOverAnItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 1457, this.validate);
    }
    get bizmgrComponentRequestToLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1458, this.validate);
    }
    get bizmgrComponentLoadingStart() {
        return new EventsFilter(this.util.getRawEvents, 5, 1459, this.validate);
    }
    get bizmgrComponentLoadingFinish() {
        return new EventsFilter(this.util.getRawEvents, 5, 1460, this.validate);
    }
    get bizmgrComponentLoadingWithAnError() {
        return new EventsFilter(this.util.getRawEvents, 5, 1461, this.validate);
    }
    get bizmgrComponentComponentWasShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1462, this.validate);
    }
    get dashboardFinishLoadingBasicDashboardWidgets() {
        return new EventsFilter(this.util.getRawEvents, 5, 1463, this.validate);
    }
    get invoicesSetBusinessInfoModalClickOnAddContinue() {
        return new EventsFilter(this.util.getRawEvents, 5, 1464, this.validate);
    }
    get invoicesSetBusinessInfoModalClickOnCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1465, this.validate);
    }
    get headerFullyLoaded() {
        return new EventsFilter(this.util.getRawEvents, 5, 1467, this.validate);
    }
    get dashboardClickOnItemFromHeader() {
        return new EventsFilter(this.util.getRawEvents, 5, 147, this.validate);
    }
    get rolesAndPermissionsInvitePeopleAddRemoveAnotherRoleOnDiffSiteClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1472, this.validate);
    }
    get nyaExposureEvent() {
        return new EventsFilter(this.util.getRawEvents, 5, 148, this.validate);
    }
    get generalExposureEvent() {
        return new EventsFilter(this.util.getRawEvents, 5, 149, this.validate);
    }
    get formBuilderBizManagerChooseAForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1493, this.validate);
    }
    get formBuilderBackofficeFormChooseATemplate() {
        return new EventsFilter(this.util.getRawEvents, 5, 1494, this.validate);
    }
    get formBuilderBackofficeFormChooseElement() {
        return new EventsFilter(this.util.getRawEvents, 5, 1496, this.validate);
    }
    get formBuilderBackofficeFormDeleteElement() {
        return new EventsFilter(this.util.getRawEvents, 5, 1497, this.validate);
    }
    get formBuilderBackofficeFormCloneElement() {
        return new EventsFilter(this.util.getRawEvents, 5, 1498, this.validate);
    }
    get dashboardSiteNavigation() {
        return new EventsFilter(this.util.getRawEvents, 5, 15, this.validate);
    }
    get dashboardMySitesActionSelectionPageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 150, this.validate);
    }
    get formBuilderBackofficeFormSaveForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1501, this.validate);
    }
    get formBuilderBackofficeFormShareForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1504, this.validate);
    }
    get formBuilderBackofficeFormPreviewForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1505, this.validate);
    }
    get formBuilderBackofficeFormFormSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1506, this.validate);
    }
    get formBuilderBackofficeFormEditTitleDescription() {
        return new EventsFilter(this.util.getRawEvents, 5, 1507, this.validate);
    }
    get dashboardMySitesSiteSelectionPageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 151, this.validate);
    }
    get crmHomeSuggestionsCarouselClickCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 1511, this.validate);
    }
    get crmHomeSuggestionsCarouselCloseBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 1512, this.validate);
    }
    get crmHomeSuggestionsCarouselClickOnCarousel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1513, this.validate);
    }
    get dashboardMySitesSiteSelectionPageNumberView() {
        return new EventsFilter(this.util.getRawEvents, 5, 152, this.validate);
    }
    get dashboardMySitesClickPromote() {
        return new EventsFilter(this.util.getRawEvents, 5, 153, this.validate);
    }
    get formBuilderBackofficeFormSubmissionClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1531, this.validate);
    }
    get formBuilderBackofficeFormSubmissionSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 1532, this.validate);
    }
    get formBuilderBackofficeFormSubmissionFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 1533, this.validate);
    }
    get formBuilderBackofficeFormReorderElements() {
        return new EventsFilter(this.util.getRawEvents, 5, 1534, this.validate);
    }
    get formBuilderBackofficeFormViewSubmissionTableFromSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1535, this.validate);
    }
    get formBuilderBackofficeFormAddEmailInSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1536, this.validate);
    }
    get formBuilderBackofficeFormSettingsToggleEmailType() {
        return new EventsFilter(this.util.getRawEvents, 5, 1539, this.validate);
    }
    get dashboardMySitesWelcomeMessageForProsDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 154, this.validate);
    }
    get quotesCreateInvoiceEnterItemDescription() {
        return new EventsFilter(this.util.getRawEvents, 5, 1540, this.validate);
    }
    get quotesCreateInvoiceClickOnSeeMoreItems() {
        return new EventsFilter(this.util.getRawEvents, 5, 1541, this.validate);
    }
    get quotesCreateInvoiceRearrangeItems() {
        return new EventsFilter(this.util.getRawEvents, 5, 1542, this.validate);
    }
    get quotesCreateInvoiceClickOnAddTax() {
        return new EventsFilter(this.util.getRawEvents, 5, 1543, this.validate);
    }
    get quotesCreateInvoiceTaxSelected() {
        return new EventsFilter(this.util.getRawEvents, 5, 1544, this.validate);
    }
    get quotesCreateInvoiceClickOnInvoiceSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 1545, this.validate);
    }
    get quotesCreateInvoiceAddRemoveDiscount() {
        return new EventsFilter(this.util.getRawEvents, 5, 1546, this.validate);
    }
    get quotesCreateInvoiceLeavingCreateInvoicePagePopUpAppears() {
        return new EventsFilter(this.util.getRawEvents, 5, 1547, this.validate);
    }
    get quotesCreateInvoiceLeavingCreateInvoicePagePopUpResolution() {
        return new EventsFilter(this.util.getRawEvents, 5, 1548, this.validate);
    }
    get quotesPreviewInvoiceClickInPreviewPageMenu() {
        return new EventsFilter(this.util.getRawEvents, 5, 1549, this.validate);
    }
    get dashboardMySitesWelcomeMessageForProsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 155, this.validate);
    }
    get quotesCreateInvoiceUserClickOnSendInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1550, this.validate);
    }
    get quotesCreateInvoiceSendInvoiceModalResolution() {
        return new EventsFilter(this.util.getRawEvents, 5, 1551, this.validate);
    }
    get quotesCreateInvoiceSendInvoiceModalFieldChange() {
        return new EventsFilter(this.util.getRawEvents, 5, 1552, this.validate);
    }
    get quotesCreateInvoiceAdditionalInfoFieldChange() {
        return new EventsFilter(this.util.getRawEvents, 5, 1553, this.validate);
    }
    get quotesCreateInvoiceChangingEmailOfAContactForPaidInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1554, this.validate);
    }
    get quotesCreateInvoiceAdditionalInfoHoverOnTooltipShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1555, this.validate);
    }
    get notificationClickOnAccountNotification() {
        return new EventsFilter(this.util.getRawEvents, 5, 1556, this.validate);
    }
    get formBuilderBackofficeFormSettingsContactSync() {
        return new EventsFilter(this.util.getRawEvents, 5, 1557, this.validate);
    }
    get dashboardAccountSettingsRenderedWithShowProBoxValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 156, this.validate);
    }
    get formBuilderBackofficeFormRemoveDescription() {
        return new EventsFilter(this.util.getRawEvents, 5, 1560, this.validate);
    }
    get invoicesNewRecurrentInvoiceClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1564, this.validate);
    }
    get invoicesChooseADueDateRecurringInvoices() {
        return new EventsFilter(this.util.getRawEvents, 5, 1565, this.validate);
    }
    get invoicesRecurringSelectABillingCycle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1566, this.validate);
    }
    get invoicesRecurringSelectNumberOfOccurrences() {
        return new EventsFilter(this.util.getRawEvents, 5, 1567, this.validate);
    }
    get invoicesRecurringSelectCustomNumberOfOccurrences() {
        return new EventsFilter(this.util.getRawEvents, 5, 1568, this.validate);
    }
    get invoicesRecurringIsAutoCharge() {
        return new EventsFilter(this.util.getRawEvents, 5, 1569, this.validate);
    }
    get dashboardMySitesSiteHomeAlertMessageFromPremiumDisplay() {
        return new EventsFilter(this.util.getRawEvents, 5, 157, this.validate);
    }
    get invoicesRecurringClickOnGetStarted() {
        return new EventsFilter(this.util.getRawEvents, 5, 1570, this.validate);
    }
    get invoicesRecurringClickOnHowDoRecurringInvoicesWork() {
        return new EventsFilter(this.util.getRawEvents, 5, 1571, this.validate);
    }
    get invoicesRecurringSetPaymentTermsPageClickOnNext() {
        return new EventsFilter(this.util.getRawEvents, 5, 1572, this.validate);
    }
    get invoicesRecurringSetPaymentTermsPageClickOnCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 1573, this.validate);
    }
    get invoicesRecurringClickOnBackToastInNewRecurringInvoicesCreationPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 1574, this.validate);
    }
    get invoicesRecurringClickOnConnectNowForAutoCharging() {
        return new EventsFilter(this.util.getRawEvents, 5, 1575, this.validate);
    }
    get invoicesRecurringClickOnLearnMoreAboutAutomaticCharging() {
        return new EventsFilter(this.util.getRawEvents, 5, 1576, this.validate);
    }
    get invoicesRecurringFilterInvoices() {
        return new EventsFilter(this.util.getRawEvents, 5, 1577, this.validate);
    }
    get invoicesRecurringTableActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 1578, this.validate);
    }
    get invoicesRecurringSearchInvoices() {
        return new EventsFilter(this.util.getRawEvents, 5, 1579, this.validate);
    }
    get dashboardMySitesSiteHomeAlertMessageFromPremiumClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 158, this.validate);
    }
    get invoicesRecurringCancelSubscription() {
        return new EventsFilter(this.util.getRawEvents, 5, 1580, this.validate);
    }
    get invoicesRecurringClickOnNewRecurringInvoiceSrc5Evid1581() {
        return new EventsFilter(this.util.getRawEvents, 5, 1581, this.validate);
    }
    get invoicesRecurringHoverOnNextWhenNotConnectedToPaymentMethod() {
        return new EventsFilter(this.util.getRawEvents, 5, 1582, this.validate);
    }
    get invoicesRecurringBackToRecurringInvoiceFromConnectPayment() {
        return new EventsFilter(this.util.getRawEvents, 5, 1583, this.validate);
    }
    get cookieBannerThemeClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 1584, this.validate);
    }
    get formBuilderFormSettingsNotificationsClickOnButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 1586, this.validate);
    }
    get invoicesRecurringClickOnNewRecurringInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1587, this.validate);
    }
    get formBuilderFormSettingsNotificationsToggle() {
        return new EventsFilter(this.util.getRawEvents, 5, 1588, this.validate);
    }
    get mySitesShowTooltip() {
        return new EventsFilter(this.util.getRawEvents, 5, 159, this.validate);
    }
    get inAppTipsGenericExposureEvent() {
        return new EventsFilter(this.util.getRawEvents, 5, 1590, this.validate);
    }
    get invoicesOwnerappClickOnCreateNewInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1592, this.validate);
    }
    get invoicesOwnerappClickOnInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1593, this.validate);
    }
    get invoicesOwnerappClickOnViewAllInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 1594, this.validate);
    }
    get invoicesOwnerappClickOnTabs() {
        return new EventsFilter(this.util.getRawEvents, 5, 1595, this.validate);
    }
    get dashboardSiteSelectorStartLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 1596, this.validate);
    }
    get dashboardSiteSelectorSearch() {
        return new EventsFilter(this.util.getRawEvents, 5, 1597, this.validate);
    }
    get dashboardSiteSelectorSearchResults() {
        return new EventsFilter(this.util.getRawEvents, 5, 1598, this.validate);
    }
    get dashboardSiteSelectorUserNotification() {
        return new EventsFilter(this.util.getRawEvents, 5, 1599, this.validate);
    }
    get dashboardSiteQuickActionClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 16, this.validate);
    }
    get mySitesTooltipClose() {
        return new EventsFilter(this.util.getRawEvents, 5, 160, this.validate);
    }
    get headerStartLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 1601, this.validate);
    }
    get invoicesRecurringMessagePresentedWhenUserReachesMaxNumberOfRecurringInvoices() {
        return new EventsFilter(this.util.getRawEvents, 5, 1602, this.validate);
    }
    get formBuilderWixDataCheck() {
        return new EventsFilter(this.util.getRawEvents, 5, 1607, this.validate);
    }
    get dashboardSiteSelectorMoreSitesLoadedSuccessfully() {
        return new EventsFilter(this.util.getRawEvents, 5, 1608, this.validate);
    }
    get mySitesHoverSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 161, this.validate);
    }
    get formBuilderBackofficeFormPostSaveForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1611, this.validate);
    }
    get formBuilderBackOfficeFormClickEditForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1618, this.validate);
    }
    get searchSearchInDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 162, this.validate);
    }
    get formBuilderBackofficeFormShareOpenForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 1621, this.validate);
    }
    get formBuilderBackofficeFormShareCopyLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 1622, this.validate);
    }
    get rolesAndPermissionsInvitePeopleClickOnGetYourMailbox() {
        return new EventsFilter(this.util.getRawEvents, 5, 1625, this.validate);
    }
    get rolesAndPermissionsMarkAsAClient() {
        return new EventsFilter(this.util.getRawEvents, 5, 1626, this.validate);
    }
    get dashboardDuplicateSiteActionWindowNavigationClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1627, this.validate);
    }
    get formBuilderBackofficeFormCloseNote() {
        return new EventsFilter(this.util.getRawEvents, 5, 1628, this.validate);
    }
    get searchScrollingInResultView() {
        return new EventsFilter(this.util.getRawEvents, 5, 163, this.validate);
    }
    get formBuilderBackofficeFormClickOnLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 1631, this.validate);
    }
    get invoicesRecurringSelectPremiumPlanIncludedOnlyServices() {
        return new EventsFilter(this.util.getRawEvents, 5, 1632, this.validate);
    }
    get invoicesRecurringAddPremiumPlanClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1633, this.validate);
    }
    get invoicesRecurringPremiumPlanSelected() {
        return new EventsFilter(this.util.getRawEvents, 5, 1634, this.validate);
    }
    get invoicesRecurringErrorMessageForInvoiceAmountLowerThenPremium() {
        return new EventsFilter(this.util.getRawEvents, 5, 1635, this.validate);
    }
    get invoicesRecurringTableActionsCancelModalClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1636, this.validate);
    }
    get searchFocusedOnASearchResult() {
        return new EventsFilter(this.util.getRawEvents, 5, 164, this.validate);
    }
    get cookieBannerCustomColorInaccessible() {
        return new EventsFilter(this.util.getRawEvents, 5, 1644, this.validate);
    }
    get clickOnSearchEverywhereInSettingsLobbysSearchSrc5Evid1646() {
        return new EventsFilter(this.util.getRawEvents, 5, 1646, this.validate);
    }
    get gizaOverlayPageModalCtaClicked1648() {
        return new EventsFilter(this.util.getRawEvents, 5, 1648, this.validate);
    }
    get bizManagerSidebarItemSelectedHighlighted() {
        return new EventsFilter(this.util.getRawEvents, 5, 1649, this.validate);
    }
    get searchClickOnSearchResult() {
        return new EventsFilter(this.util.getRawEvents, 5, 165, this.validate);
    }
    get cookieBannerUouRestoreDefaultText() {
        return new EventsFilter(this.util.getRawEvents, 5, 1650, this.validate);
    }
    get invoicesRecurringClientSelected() {
        return new EventsFilter(this.util.getRawEvents, 5, 1651, this.validate);
    }
    get invoicesRecurringSiteOfClientSelected() {
        return new EventsFilter(this.util.getRawEvents, 5, 1652, this.validate);
    }
    get formBuilderBizMgrFromTheAdditionalActionMenuClickAction() {
        return new EventsFilter(this.util.getRawEvents, 5, 1654, this.validate);
    }
    get formBuilderBizMgrFormDeletion() {
        return new EventsFilter(this.util.getRawEvents, 5, 1656, this.validate);
    }
    get chatSettingsPcfOfflineOnlineFormChevron() {
        return new EventsFilter(this.util.getRawEvents, 5, 1658, this.validate);
    }
    get searchCloseSearchDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 166, this.validate);
    }
    get siteActionsDropDownListShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1662, this.validate);
    }
    get teamManagementSwitchToStudioPageLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 1664, this.validate);
    }
    get formBuilderBackofficeFormUserChangeLtrRtl() {
        return new EventsFilter(this.util.getRawEvents, 5, 1665, this.validate);
    }
    get formBuilderBackofficeFormAutoAddressRadioButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 1666, this.validate);
    }
    get cookieBannerCustomFontAndSizeClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 1669, this.validate);
    }
    get clickOnSeeMoreButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 167, this.validate);
    }
    get bizManagerSetupWidgetSectionCompleted() {
        return new EventsFilter(this.util.getRawEvents, 5, 1671, this.validate);
    }
    get clickOnShowAllButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 168, this.validate);
    }
    get ssoSettingsChangeActivationStatus() {
        return new EventsFilter(this.util.getRawEvents, 5, 1684, this.validate);
    }
    get ssoSettingsIsMandatoryToggleStatus() {
        return new EventsFilter(this.util.getRawEvents, 5, 1685, this.validate);
    }
    get mySitesUnableToMoveSiteToTrashMessageShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1686, this.validate);
    }
    get financialSettingsGoToInvoicesPriceQuotes() {
        return new EventsFilter(this.util.getRawEvents, 5, 1688, this.validate);
    }
    get invoicesFirstTimeExpirienceGetStarted() {
        return new EventsFilter(this.util.getRawEvents, 5, 1689, this.validate);
    }
    get invoicesFirstTimeExpirienceLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1690, this.validate);
    }
    get invoicesDepositRequestDeposit() {
        return new EventsFilter(this.util.getRawEvents, 5, 1692, this.validate);
    }
    get invoicesDepositChooseFlatOrPercentageDeposit() {
        return new EventsFilter(this.util.getRawEvents, 5, 1693, this.validate);
    }
    get invoicesDepositLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1694, this.validate);
    }
    get invoicesConnectPaymentLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1695, this.validate);
    }
    get invoicesDepositDropDeposit() {
        return new EventsFilter(this.util.getRawEvents, 5, 1697, this.validate);
    }
    get pingPingWidgetLoadedInEditNotificationDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 1698, this.validate);
    }
    get pingSaveInEditNotificationDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 1699, this.validate);
    }
    get dashboardSiteAppPromo() {
        return new EventsFilter(this.util.getRawEvents, 5, 17, this.validate);
    }
    get teamManagementClickToLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 170, this.validate);
    }
    get locationsLoadedOnBusinessInfoPageSrc5Evid1700() {
        return new EventsFilter(this.util.getRawEvents, 5, 1700, this.validate);
    }
    get apesUpdateTooltip() {
        return new EventsFilter(this.util.getRawEvents, 5, 1701, this.validate);
    }
    get apesVersioningPreviewNewEmail() {
        return new EventsFilter(this.util.getRawEvents, 5, 1702, this.validate);
    }
    get apesVersioningUpdateNewVersion() {
        return new EventsFilter(this.util.getRawEvents, 5, 1703, this.validate);
    }
    get apesVersioningUpdateNewVersionError() {
        return new EventsFilter(this.util.getRawEvents, 5, 1704, this.validate);
    }
    get apesVersioningSaveAnUpdateError() {
        return new EventsFilter(this.util.getRawEvents, 5, 1705, this.validate);
    }
    get dashboardAfterSiteSelectorStartedLoadingAndBeforeFinished() {
        return new EventsFilter(this.util.getRawEvents, 5, 1706, this.validate);
    }
    get formBuilderEmailSettingsInSchemaButExperimentClosed() {
        return new EventsFilter(this.util.getRawEvents, 5, 1708, this.validate);
    }
    get siteSelectorSiteChosen() {
        return new EventsFilter(this.util.getRawEvents, 5, 1709, this.validate);
    }
    get teamManagementEditTeamProfileOpenDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 171, this.validate);
    }
    get mySitesClickOnUrl() {
        return new EventsFilter(this.util.getRawEvents, 5, 1713, this.validate);
    }
    get dashboardSiteCategoryShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1714, this.validate);
    }
    get dashboardSiteSelectorClickForDebuggingPurposes() {
        return new EventsFilter(this.util.getRawEvents, 5, 1715, this.validate);
    }
    get dashboardSiteSelectorLoadedSuccessfullyForDebuggingPurposes() {
        return new EventsFilter(this.util.getRawEvents, 5, 1716, this.validate);
    }
    get dashboardLabelingPointInSiteSelectorForDebuggingPurposes() {
        return new EventsFilter(this.util.getRawEvents, 5, 1719, this.validate);
    }
    get teamManagementPickLogo() {
        return new EventsFilter(this.util.getRawEvents, 5, 172, this.validate);
    }
    get formBuilderPdfAttachmentsClickOnCheckbox() {
        return new EventsFilter(this.util.getRawEvents, 5, 1721, this.validate);
    }
    get bmACallForTokenRefreshIsFailing() {
        return new EventsFilter(this.util.getRawEvents, 5, 1722, this.validate);
    }
    get wixStudioExplanatoryModalForEditorXMigrationOpenedSrc5Evid1723() {
        return new EventsFilter(this.util.getRawEvents, 5, 1723, this.validate);
    }
    get wixStudioExplanatoryModalForEditorXMigrationCtaClickedSrc5Evid1724() {
        return new EventsFilter(this.util.getRawEvents, 5, 1724, this.validate);
    }
    get transferSiteSiteCanTBeTransferredPopUpView() {
        return new EventsFilter(this.util.getRawEvents, 5, 1725, this.validate);
    }
    get cookiesBannerSettingsWindowChange() {
        return new EventsFilter(this.util.getRawEvents, 5, 1727, this.validate);
    }
    get finishFocusWidgetSection() {
        return new EventsFilter(this.util.getRawEvents, 5, 1728, this.validate);
    }
    get teamManagementSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 173, this.validate);
    }
    get focusWidgetClickViewAll() {
        return new EventsFilter(this.util.getRawEvents, 5, 1730, this.validate);
    }
    get mySitesWhenDialogPopsUp() {
        return new EventsFilter(this.util.getRawEvents, 5, 1732, this.validate);
    }
    get focusWidgetItemViewedInList() {
        return new EventsFilter(this.util.getRawEvents, 5, 1733, this.validate);
    }
    get mySitesClickOnButtonInDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 1735, this.validate);
    }
    get dashboardFinishLoadingAllWidgets() {
        return new EventsFilter(this.util.getRawEvents, 5, 1736, this.validate);
    }
    get teamManagementAddMembersOpenDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 174, this.validate);
    }
    get sapir1() {
        return new EventsFilter(this.util.getRawEvents, 5, 1748, this.validate);
    }
    get teamManagementAddMembersSendInvite() {
        return new EventsFilter(this.util.getRawEvents, 5, 175, this.validate);
    }
    get formBuilderBizMgrSafShareOptions() {
        return new EventsFilter(this.util.getRawEvents, 5, 1751, this.validate);
    }
    get formFormsInContactPanelViewSubmission() {
        return new EventsFilter(this.util.getRawEvents, 5, 1753, this.validate);
    }
    get formFormsInContactPanelViewSubmissionsList() {
        return new EventsFilter(this.util.getRawEvents, 5, 1754, this.validate);
    }
    get formFormsInContactPanelDownloadFileFormSubmission() {
        return new EventsFilter(this.util.getRawEvents, 5, 1755, this.validate);
    }
    get formBuilderPaymentFormClickDoneAfterEnterItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 1757, this.validate);
    }
    get formBuilderPaymentFormHoverNextButtonAdi() {
        return new EventsFilter(this.util.getRawEvents, 5, 1759, this.validate);
    }
    get teamManagementAddMembersInviteClickSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 176, this.validate);
    }
    get headerClickOnItemInOneOfTheHeaderDialogsDropDownMenus() {
        return new EventsFilter(this.util.getRawEvents, 5, 1768, this.validate);
    }
    get teamManagementAddMembersInviteClickFail() {
        return new EventsFilter(this.util.getRawEvents, 5, 177, this.validate);
    }
    get renderingPlatformRedirect() {
        return new EventsFilter(this.util.getRawEvents, 5, 1770, this.validate);
    }
    get mySitesUnableToMoveSitesToTrashModalIsShownOnBulkMoveToTrashOrSingleSiteMoveToTrash() {
        return new EventsFilter(this.util.getRawEvents, 5, 1771, this.validate);
    }
    get teamManagementDeleteTeamMemberOpenDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 178, this.validate);
    }
    get teamManagementDeleteTeamMemberDeleteClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 179, this.validate);
    }
    get formBuilderOneAppModalShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 1790, this.validate);
    }
    get formBuilderOneAppModalAction() {
        return new EventsFilter(this.util.getRawEvents, 5, 1791, this.validate);
    }
    get formBuilderOneAppFormSelectorEmptyState() {
        return new EventsFilter(this.util.getRawEvents, 5, 1792, this.validate);
    }
    get formBuilderFormSectionLoadedInSpacesApp() {
        return new EventsFilter(this.util.getRawEvents, 5, 1794, this.validate);
    }
    get formBuilderOneAppLearnMoreClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1796, this.validate);
    }
    get formBuilderOneApp3DotsSwipe() {
        return new EventsFilter(this.util.getRawEvents, 5, 1797, this.validate);
    }
    get dashboardSiteNewsViewAll() {
        return new EventsFilter(this.util.getRawEvents, 5, 18, this.validate);
    }
    get teamManagementDeleteTeamMemberDeleteClickSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 180, this.validate);
    }
    get formBuilderOneAppClickInsideSwipeAction() {
        return new EventsFilter(this.util.getRawEvents, 5, 1800, this.validate);
    }
    get formBuilderOneAppWidgetClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 1805, this.validate);
    }
    get formBuilderOneAppFormsListPageOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 1806, this.validate);
    }
    get formBuilderOneAppSubmissionsListPageOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 1807, this.validate);
    }
    get teamManagementDeleteTeamMemberDeleteClickFail() {
        return new EventsFilter(this.util.getRawEvents, 5, 181, this.validate);
    }
    get focusWidgetHoverOverProgressBarToShowPopover() {
        return new EventsFilter(this.util.getRawEvents, 5, 1812, this.validate);
    }
    get formBuilderOneAppFormsResultsPageOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 1814, this.validate);
    }
    get formBuilderOneAppFormsResultsPageActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 1815, this.validate);
    }
    get teamManagementGetInviteLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 182, this.validate);
    }
    get teamManagementEditRoleOpenDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 183, this.validate);
    }
    get teamManagementSwitchToStudioClickToSwitch() {
        return new EventsFilter(this.util.getRawEvents, 5, 1838, this.validate);
    }
    get teamManagementSwitchToStudioLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 1839, this.validate);
    }
    get teamManagementEditRoleUpdateClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 184, this.validate);
    }
    get searchToDeleteSrc5Evid1847() {
        return new EventsFilter(this.util.getRawEvents, 5, 1847, this.validate);
    }
    get teamManagementResendInviteClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 185, this.validate);
    }
    get leavingWithoutSavingModalOpenedSrc5Evid1858() {
        return new EventsFilter(this.util.getRawEvents, 5, 1858, this.validate);
    }
    get dashboardContributorsClickOnMangeRolesAndPermissions() {
        return new EventsFilter(this.util.getRawEvents, 5, 187, this.validate);
    }
    get rolesAndPermissionsManageRolesEditRoleSrc5Evid1871() {
        return new EventsFilter(this.util.getRawEvents, 5, 1871, this.validate);
    }
    get rolesAndPermissionsInviteTeammatesSendInviteClickSrc5Evid1874() {
        return new EventsFilter(this.util.getRawEvents, 5, 1874, this.validate);
    }
    get restoreSiteModalOpenedSrc5Evid1875() {
        return new EventsFilter(this.util.getRawEvents, 5, 1875, this.validate);
    }
    get restoreSiteModalCtaClickedSrc5Evid1876() {
        return new EventsFilter(this.util.getRawEvents, 5, 1876, this.validate);
    }
    get leavingWithoutSavingModalCtaClickedSrc5Evid1878() {
        return new EventsFilter(this.util.getRawEvents, 5, 1878, this.validate);
    }
    get ascendHoverOnUpgradeAscend() {
        return new EventsFilter(this.util.getRawEvents, 5, 188, this.validate);
    }
    get ascendClickOnUpgradeAscend() {
        return new EventsFilter(this.util.getRawEvents, 5, 189, this.validate);
    }
    get microActionAdditionalInfo() {
        return new EventsFilter(this.util.getRawEvents, 5, 1895, this.validate);
    }
    get dashboardSiteNewsClickItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 19, this.validate);
    }
    get ascendPpClickOnMoveToWixPremiumPp() {
        return new EventsFilter(this.util.getRawEvents, 5, 190, this.validate);
    }
    get newFormsFormListTabOpenedOldFormsForms() {
        return new EventsFilter(this.util.getRawEvents, 5, 1901, this.validate);
    }
    get websitePageChecklistInteractWithStepInAList() {
        return new EventsFilter(this.util.getRawEvents, 5, 1902, this.validate);
    }
    get ascendPpThankYouPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 191, this.validate);
    }
    get privacyCookiesCookieConsentBannerCardCtaClickSrc5Evid1927() {
        return new EventsFilter(this.util.getRawEvents, 5, 1927, this.validate);
    }
    get invoicesMoreActionsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 193, this.validate);
    }
    get invoicesMoreActionsChooseOption() {
        return new EventsFilter(this.util.getRawEvents, 5, 194, this.validate);
    }
    get mobileClickOnCreateNewSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 196, this.validate);
    }
    get dashboardFetchDealerLayoutFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 1999, this.validate);
    }
    get dashboardUserSeeFeed() {
        return new EventsFilter(this.util.getRawEvents, 5, 20, this.validate);
    }
    get mySitesClickInUnableToMoveSitesToTrashDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 2000, this.validate);
    }
    get samplingPageFeatures() {
        return new EventsFilter(this.util.getRawEvents, 5, 20000, this.validate);
    }
    get clickingOnQuickActionButtonThreeDots() {
        return new EventsFilter(this.util.getRawEvents, 5, 2001, this.validate);
    }
    get tooltipForOwnerApp() {
        return new EventsFilter(this.util.getRawEvents, 5, 2002, this.validate);
    }
    get tooltipForOwnerAppClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 2003, this.validate);
    }
    get downloadModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 2004, this.validate);
    }
    get downloadModalClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 2005, this.validate);
    }
    get downloadModalSuccessMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 2006, this.validate);
    }
    get downloadModalErrorMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 2007, this.validate);
    }
    get automationMainTableUserSearchFocusOut() {
        return new EventsFilter(this.util.getRawEvents, 5, 2008, this.validate);
    }
    get searchClickedToOpenSearch() {
        return new EventsFilter(this.util.getRawEvents, 5, 202, this.validate);
    }
    get osClickOnFavorites() {
        return new EventsFilter(this.util.getRawEvents, 5, 203, this.validate);
    }
    get osFavoritesPopupShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 204, this.validate);
    }
    get dashboardSiteNewsFilter() {
        return new EventsFilter(this.util.getRawEvents, 5, 21, this.validate);
    }
    get osFavoritesPopupCtaClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 210, this.validate);
    }
    get osFavoritesPagesEdit() {
        return new EventsFilter(this.util.getRawEvents, 5, 215, this.validate);
    }
    get dashboardSettingsUpdate() {
        return new EventsFilter(this.util.getRawEvents, 5, 22, this.validate);
    }
    get gizaSdkApiCallReceivedSrc5Evid2222() {
        return new EventsFilter(this.util.getRawEvents, 5, 2222, this.validate);
    }
    get dashboardTpaUpgradeClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 23, this.validate);
    }
    get automationsPriceQuotesInvoicesSendFeedbackForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 232, this.validate);
    }
    get automationsPriceQuotesInvoicesSubmitFeedbackForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 233, this.validate);
    }
    get dashboardSiteSelectorLoaded() {
        return new EventsFilter(this.util.getRawEvents, 5, 24, this.validate);
    }
    get mobileClickOnChangeLng() {
        return new EventsFilter(this.util.getRawEvents, 5, 244, this.validate);
    }
    get quotesPrintInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 245, this.validate);
    }
    get quotesPayInvoiceClickSrc5Evid246() {
        return new EventsFilter(this.util.getRawEvents, 5, 246, this.validate);
    }
    get dashboardSiteSelectorButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 25, this.validate);
    }
    get homeClickNotificationCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 250, this.validate);
    }
    get waiverClientAppWaiverStatusDropdown() {
        return new EventsFilter(this.util.getRawEvents, 5, 2501, this.validate);
    }
    get waiverClientAppWaiverStatusChangedManually() {
        return new EventsFilter(this.util.getRawEvents, 5, 2502, this.validate);
    }
    get waiverClientAppSendWaiverToClient() {
        return new EventsFilter(this.util.getRawEvents, 5, 2503, this.validate);
    }
    get homeMarkNotificationUnread() {
        return new EventsFilter(this.util.getRawEvents, 5, 251, this.validate);
    }
    get homeMarkNotificationRead() {
        return new EventsFilter(this.util.getRawEvents, 5, 252, this.validate);
    }
    get homeMarkAllUnreadNotifications() {
        return new EventsFilter(this.util.getRawEvents, 5, 253, this.validate);
    }
    get homeNotificationBadgeShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 254, this.validate);
    }
    get homeAllowBlockBrowserNotificationClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 255, this.validate);
    }
    get notificationsClickOnAllSitesTab() {
        return new EventsFilter(this.util.getRawEvents, 5, 256, this.validate);
    }
    get dashboardEnterEditorx() {
        return new EventsFilter(this.util.getRawEvents, 5, 257, this.validate);
    }
    get dashboardQuickActionsDragNDrop() {
        return new EventsFilter(this.util.getRawEvents, 5, 26, this.validate);
    }
    get triggersServerTpaActivationInitiatedCompleted() {
        return new EventsFilter(this.util.getRawEvents, 5, 260, this.validate);
    }
    get triggersServerTpaActivationInitiatedFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 261, this.validate);
    }
    get triggersServerTaskScheduled() {
        return new EventsFilter(this.util.getRawEvents, 5, 262, this.validate);
    }
    get triggersServerTpaActivationCompletedSuccessfully() {
        return new EventsFilter(this.util.getRawEvents, 5, 263, this.validate);
    }
    get triggersServerTpaActivationCompleteFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 264, this.validate);
    }
    get triggersServerTriggerScheduleCancelled() {
        return new EventsFilter(this.util.getRawEvents, 5, 266, this.validate);
    }
    get notificationNotificationAutomationCreated() {
        return new EventsFilter(this.util.getRawEvents, 5, 269, this.validate);
    }
    get dashboardClickOnSideBar() {
        return new EventsFilter(this.util.getRawEvents, 5, 27, this.validate);
    }
    get invoicesClickOnSendInvoiceButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 270, this.validate);
    }
    get notificationShowNotification() {
        return new EventsFilter(this.util.getRawEvents, 5, 271, this.validate);
    }
    get notificationClickOnNotification() {
        return new EventsFilter(this.util.getRawEvents, 5, 272, this.validate);
    }
    get notificationsEnableDesktopNotificationsFirstTime() {
        return new EventsFilter(this.util.getRawEvents, 5, 273, this.validate);
    }
    get notificationsPushNotificationToggleIsDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 274, this.validate);
    }
    get notificationsEnableToTurnOnOffAllSiteNotificationsExperts() {
        return new EventsFilter(this.util.getRawEvents, 5, 275, this.validate);
    }
    get notificationsEnableToTurnOnOffNotificationsPerChannel() {
        return new EventsFilter(this.util.getRawEvents, 5, 276, this.validate);
    }
    get notificationsShowMobileDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 277, this.validate);
    }
    get notificationsDisplayPopupForUserBlockedDesktopNotifications() {
        return new EventsFilter(this.util.getRawEvents, 5, 278, this.validate);
    }
    get notificationCloseAHighPriorityNotifications() {
        return new EventsFilter(this.util.getRawEvents, 5, 279, this.validate);
    }
    get dashboardShowMorePopularShortcutsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 28, this.validate);
    }
    get notificationsDisplayDesktopNotificationsBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 280, this.validate);
    }
    get notificationsClickOnDesktopNotificationsBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 281, this.validate);
    }
    get notificationsResolveAlertNotification() {
        return new EventsFilter(this.util.getRawEvents, 5, 282, this.validate);
    }
    get invoicesUserSendInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 283, this.validate);
    }
    get bizManagerSetupWidgetOmniPhaseCompletedInDashboard() {
        return new EventsFilter(this.util.getRawEvents, 5, 284, this.validate);
    }
    get automationsDaComponentShowPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 285, this.validate);
    }
    get automationsDaComponentManageAutomations() {
        return new EventsFilter(this.util.getRawEvents, 5, 286, this.validate);
    }
    get automationsDaComponentCreateYourOwn() {
        return new EventsFilter(this.util.getRawEvents, 5, 287, this.validate);
    }
    get automationsDaComponentRecommendationClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 288, this.validate);
    }
    get automationsDaComponentExploreMoreClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 289, this.validate);
    }
    get dashboardLessPopularShortcutsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 29, this.validate);
    }
    get clickOnSearchInHelpCenterInSearchSrc5Evid292() {
        return new EventsFilter(this.util.getRawEvents, 5, 292, this.validate);
    }
    get errorPageClick295() {
        return new EventsFilter(this.util.getRawEvents, 5, 295, this.validate);
    }
    get accountDialogClickItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 296, this.validate);
    }
    get errorPage297() {
        return new EventsFilter(this.util.getRawEvents, 5, 297, this.validate);
    }
    get dashboardTourUserSeesTourBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 30, this.validate);
    }
    get dashboardStartLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 300, this.validate);
    }
    get dashboardEntryFeeding() {
        return new EventsFilter(this.util.getRawEvents, 5, 3000, this.validate);
    }
    get clickOnGoToEditorFromBmSrc5Evid3001() {
        return new EventsFilter(this.util.getRawEvents, 5, 3001, this.validate);
    }
    get homeCloseDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 301, this.validate);
    }
    get dashboardFinishLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 302, this.validate);
    }
    get homeOpenDialogClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 303, this.validate);
    }
    get dialogFirstLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 304, this.validate);
    }
    get homeMultipleSitesCheck() {
        return new EventsFilter(this.util.getRawEvents, 5, 305, this.validate);
    }
    get homeBizMgrEligible() {
        return new EventsFilter(this.util.getRawEvents, 5, 306, this.validate);
    }
    get homeClickOnCreateAnythingItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 307, this.validate);
    }
    get myAccountClickOnExpertDashboard() {
        return new EventsFilter(this.util.getRawEvents, 5, 308, this.validate);
    }
    get dashboardDashboardHomeIsFullyLoaded() {
        return new EventsFilter(this.util.getRawEvents, 5, 309, this.validate);
    }
    get dashboardTourCloseTour() {
        return new EventsFilter(this.util.getRawEvents, 5, 31, this.validate);
    }
    get dashboardSiteSettingsPageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 310, this.validate);
    }
    get dashboardSiteSettingsExperienceDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 311, this.validate);
    }
    get dashboardSiteSettingsChangeValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 312, this.validate);
    }
    get dashboardSiteSettingsChangeValueSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 313, this.validate);
    }
    get dashboardSiteSettingsChangeValueFailure() {
        return new EventsFilter(this.util.getRawEvents, 5, 314, this.validate);
    }
    get dashboardSiteSettingsClickLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 315, this.validate);
    }
    get dashboardSiteSettingsDeleteValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 316, this.validate);
    }
    get dashboardSiteSettingsDeleteValueSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 317, this.validate);
    }
    get dashboardSiteSettingsDeleteValueFailure() {
        return new EventsFilter(this.util.getRawEvents, 5, 318, this.validate);
    }
    get dashboardSiteSettingsSelectValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 319, this.validate);
    }
    get dashboardTourNextStep() {
        return new EventsFilter(this.util.getRawEvents, 5, 32, this.validate);
    }
    get dashboardSiteSettingsClosingComponent() {
        return new EventsFilter(this.util.getRawEvents, 5, 320, this.validate);
    }
    get dashboardSiteSettingsDomainSslOption() {
        return new EventsFilter(this.util.getRawEvents, 5, 321, this.validate);
    }
    get dashboardSiteSettingsDomainSslPopupCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 322, this.validate);
    }
    get dashboardSiteSettingsSeoClickEnterAddressManually() {
        return new EventsFilter(this.util.getRawEvents, 5, 323, this.validate);
    }
    get dashboardSiteSettingsSeoCloseEnterAddressManuallyPopUp() {
        return new EventsFilter(this.util.getRawEvents, 5, 324, this.validate);
    }
    get dashboardSiteSettingsSeoClickOnGetStarted() {
        return new EventsFilter(this.util.getRawEvents, 5, 325, this.validate);
    }
    get dashboardSiteSettingsOverviewUnpublishSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 326, this.validate);
    }
    get dashboardSiteSettingsOverviewUnpublishPopupCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 327, this.validate);
    }
    get renameSiteOpenRenameDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 328, this.validate);
    }
    get dashboardChangeSiteDisplayNameSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 329, this.validate);
    }
    get dashboardUserSeeFeedMoreThen3Sec() {
        return new EventsFilter(this.util.getRawEvents, 5, 33, this.validate);
    }
    get dashboardChangeUrlSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 330, this.validate);
    }
    get needsYourActionNyaWidgetShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 331, this.validate);
    }
    get needsYourActionItemIsShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 332, this.validate);
    }
    get needsYourActionClickOnItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 333, this.validate);
    }
    get settingsLobbyItemExposedSrc5Evid3333() {
        return new EventsFilter(this.util.getRawEvents, 5, 3333, this.validate);
    }
    get needsYourActionClickOnUndoHidingAnItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 334, this.validate);
    }
    get needsYourActionResolvedNyaItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 335, this.validate);
    }
    get hooverNewReleasesIcon() {
        return new EventsFilter(this.util.getRawEvents, 5, 336, this.validate);
    }
    get tooltipNotificationPopsUp() {
        return new EventsFilter(this.util.getRawEvents, 5, 337, this.validate);
    }
    get newReleasesApiRequest() {
        return new EventsFilter(this.util.getRawEvents, 5, 338, this.validate);
    }
    get closeTooltipPopUp() {
        return new EventsFilter(this.util.getRawEvents, 5, 339, this.validate);
    }
    get badgeNotification() {
        return new EventsFilter(this.util.getRawEvents, 5, 340, this.validate);
    }
    get viewFeatureCard() {
        return new EventsFilter(this.util.getRawEvents, 5, 341, this.validate);
    }
    get cardCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 342, this.validate);
    }
    get viewedAllUpdates() {
        return new EventsFilter(this.util.getRawEvents, 5, 343, this.validate);
    }
    get settingsOverviewPageClickOnItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 345, this.validate);
    }
    get iconRender() {
        return new EventsFilter(this.util.getRawEvents, 5, 347, this.validate);
    }
    get quotesBulkActionsPrint() {
        return new EventsFilter(this.util.getRawEvents, 5, 348, this.validate);
    }
    get setupWidgetItems() {
        return new EventsFilter(this.util.getRawEvents, 5, 349, this.validate);
    }
    get dashboardShowSideBar() {
        return new EventsFilter(this.util.getRawEvents, 5, 35, this.validate);
    }
    get rolesAddNewRoleClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 350, this.validate);
    }
    get searchQuery() {
        return new EventsFilter(this.util.getRawEvents, 5, 3500, this.validate);
    }
    get rolesAddRoleCloseWindow() {
        return new EventsFilter(this.util.getRawEvents, 5, 351, this.validate);
    }
    get rolesAddRoleInviteClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 352, this.validate);
    }
    get rolesAddRoleInviteClickSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 353, this.validate);
    }
    get crmHomeButtonClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 354, this.validate);
    }
    get rolesAddRoleInviteClickFail() {
        return new EventsFilter(this.util.getRawEvents, 5, 355, this.validate);
    }
    get rolesLoginPageLoadStart() {
        return new EventsFilter(this.util.getRawEvents, 5, 356, this.validate);
    }
    get rolesLoginPageLoadComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 357, this.validate);
    }
    get rolesAcceptFail() {
        return new EventsFilter(this.util.getRawEvents, 5, 358, this.validate);
    }
    get rolesResendInviteClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 359, this.validate);
    }
    get dashboardCloseBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 36, this.validate);
    }
    get rolesDeleteRoleClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 360, this.validate);
    }
    get rolesDeleteRoleCloseWindow() {
        return new EventsFilter(this.util.getRawEvents, 5, 361, this.validate);
    }
    get rolesDeleteRoleDeleteClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 362, this.validate);
    }
    get rolesDeleteRoleDeleteClickSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 363, this.validate);
    }
    get rolesDeleteRoleDeleteClickFail() {
        return new EventsFilter(this.util.getRawEvents, 5, 364, this.validate);
    }
    get rolesAcceptInviteFromMobilePageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 365, this.validate);
    }
    get rolesEditRoleClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 366, this.validate);
    }
    get rolesEditRoleUpdateClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 367, this.validate);
    }
    get rolesGetInviteLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 368, this.validate);
    }
    get analyticsKpiIsShownInAnalyticsWidget() {
        return new EventsFilter(this.util.getRawEvents, 5, 369, this.validate);
    }
    get dashboardNewsUserClickReadMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 37, this.validate);
    }
    get exposureForAnalyticsTests() {
        return new EventsFilter(this.util.getRawEvents, 5, 3701, this.validate);
    }
    get analyticsTooltip() {
        return new EventsFilter(this.util.getRawEvents, 5, 371, this.validate);
    }
    get analyticsClickOnWidgetButtons() {
        return new EventsFilter(this.util.getRawEvents, 5, 372, this.validate);
    }
    get analyticsClickOnKpi() {
        return new EventsFilter(this.util.getRawEvents, 5, 373, this.validate);
    }
    get analyticsChooseKpiFromCustomizationList() {
        return new EventsFilter(this.util.getRawEvents, 5, 374, this.validate);
    }
    get analyticsClickOnButtonFromCustomizationList() {
        return new EventsFilter(this.util.getRawEvents, 5, 375, this.validate);
    }
    get analyticsChangeKpisTimePeriod() {
        return new EventsFilter(this.util.getRawEvents, 5, 376, this.validate);
    }
    get analyticsHoverAKpi() {
        return new EventsFilter(this.util.getRawEvents, 5, 377, this.validate);
    }
    get notificationsClickOnAChannelSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 379, this.validate);
    }
    get dashboardNewsClickViewAll() {
        return new EventsFilter(this.util.getRawEvents, 5, 38, this.validate);
    }
    get notificationsClickOnChannelLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 380, this.validate);
    }
    get quotesClickDueDate() {
        return new EventsFilter(this.util.getRawEvents, 5, 383, this.validate);
    }
    get quotesChooseDueDate() {
        return new EventsFilter(this.util.getRawEvents, 5, 384, this.validate);
    }
    get quotesClickIssueDate() {
        return new EventsFilter(this.util.getRawEvents, 5, 385, this.validate);
    }
    get quotesChooseIssueDate() {
        return new EventsFilter(this.util.getRawEvents, 5, 386, this.validate);
    }
    get notificationGroupNotificationIsDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 387, this.validate);
    }
    get quotesViewPayments() {
        return new EventsFilter(this.util.getRawEvents, 5, 388, this.validate);
    }
    get quotesSubscriptionViewPayments() {
        return new EventsFilter(this.util.getRawEvents, 5, 389, this.validate);
    }
    get dashboardProsSiteActionsOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 39, this.validate);
    }
    get quotesBulkActionsDownloadPdf() {
        return new EventsFilter(this.util.getRawEvents, 5, 397, this.validate);
    }
    get quotesBulkActionsExportCsv() {
        return new EventsFilter(this.util.getRawEvents, 5, 398, this.validate);
    }
    get quotesDuplicateInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 399, this.validate);
    }
    get dashboardClickOnMySites() {
        return new EventsFilter(this.util.getRawEvents, 5, 40, this.validate);
    }
    get quotesTabPageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 400, this.validate);
    }
    get quotesInvoicesOnboardingPageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 401, this.validate);
    }
    get quotesGetPaymentClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 402, this.validate);
    }
    get quotesNewInvoiceClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 403, this.validate);
    }
    get quotesEmailInvoicePage() {
        return new EventsFilter(this.util.getRawEvents, 5, 404, this.validate);
    }
    get quotesEmailInvoicePageCancelClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 405, this.validate);
    }
    get quotesViewInvoicePageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 406, this.validate);
    }
    get quotesChangeValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 407, this.validate);
    }
    get quotesCreateInvoiceSaveDraftClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 408, this.validate);
    }
    get quotesCreateInvoiceSaveDraftComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 409, this.validate);
    }
    get dashboardClickToggleSitesListInNavbar() {
        return new EventsFilter(this.util.getRawEvents, 5, 41, this.validate);
    }
    get quotesCreateInvoiceSaveDraftFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 410, this.validate);
    }
    get submitGdprForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 4101, this.validate);
    }
    get submitGdprFormSrc5Evid4102() {
        return new EventsFilter(this.util.getRawEvents, 5, 4102, this.validate);
    }
    get quotesCreateInvoiceCancelClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 411, this.validate);
    }
    get quotesCreateInvoiceAddLine() {
        return new EventsFilter(this.util.getRawEvents, 5, 412, this.validate);
    }
    get quotesCreateInvoiceDeleteLine() {
        return new EventsFilter(this.util.getRawEvents, 5, 413, this.validate);
    }
    get quotesNewProductClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 414, this.validate);
    }
    get quotesNewProductSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 415, this.validate);
    }
    get quotesNewProductCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 416, this.validate);
    }
    get quotesEditProductClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 417, this.validate);
    }
    get quotesDeleteProductClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 418, this.validate);
    }
    get quotesDeleteProductPageOk() {
        return new EventsFilter(this.util.getRawEvents, 5, 419, this.validate);
    }
    get dashboardSiteQuickActionDialogApprove() {
        return new EventsFilter(this.util.getRawEvents, 5, 42, this.validate);
    }
    get quotesDeleteProductPageCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 420, this.validate);
    }
    get cookiesBannerBannerDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 4200, this.validate);
    }
    get cookiesBannerPolicyApplied() {
        return new EventsFilter(this.util.getRawEvents, 5, 4201, this.validate);
    }
    get quotesEditInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 421, this.validate);
    }
    get quotesPreviewInvoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 422, this.validate);
    }
    get quotesPrintInvoiceSrc5Evid423() {
        return new EventsFilter(this.util.getRawEvents, 5, 423, this.validate);
    }
    get quotesAddPaymentClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 424, this.validate);
    }
    get quotesAddPaymentPageOk() {
        return new EventsFilter(this.util.getRawEvents, 5, 425, this.validate);
    }
    get quotesAddPaymentPageCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 426, this.validate);
    }
    get quotesVoidPageOk() {
        return new EventsFilter(this.util.getRawEvents, 5, 428, this.validate);
    }
    get quotesVoidPageCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 429, this.validate);
    }
    get dashboardSiteQuickActionDialogClose() {
        return new EventsFilter(this.util.getRawEvents, 5, 43, this.validate);
    }
    get quotesDeleteInvoiceClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 430, this.validate);
    }
    get quotesDeleteInvoicePageOk() {
        return new EventsFilter(this.util.getRawEvents, 5, 431, this.validate);
    }
    get quotesDeleteInvoicePageCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 432, this.validate);
    }
    get quotesSettingsNewTaxClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 433, this.validate);
    }
    get quotesSettingsNewTaxPageSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 434, this.validate);
    }
    get quotesSettingsNewTaxPageCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 435, this.validate);
    }
    get quotesEditTaxClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 436, this.validate);
    }
    get quotesDeleteTaxClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 437, this.validate);
    }
    get quotesDeleteTaxPageOk() {
        return new EventsFilter(this.util.getRawEvents, 5, 438, this.validate);
    }
    get quotesDeleteTaxPageCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 439, this.validate);
    }
    get myAccountClickOnHelp() {
        return new EventsFilter(this.util.getRawEvents, 5, 44, this.validate);
    }
    get quotesShowPaymentPopupPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 440, this.validate);
    }
    get quotesConnectPaymentClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 441, this.validate);
    }
    get quotesCreateAnAccountOfPaymentProvider() {
        return new EventsFilter(this.util.getRawEvents, 5, 442, this.validate);
    }
    get quotesNonValidPaypalAccountPopup() {
        return new EventsFilter(this.util.getRawEvents, 5, 443, this.validate);
    }
    get quotesNonValidPaypalAccountPopupClose() {
        return new EventsFilter(this.util.getRawEvents, 5, 444, this.validate);
    }
    get quotesNonValidPaypalAccountPopupClickLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 445, this.validate);
    }
    get quotesSortInvoices() {
        return new EventsFilter(this.util.getRawEvents, 5, 446, this.validate);
    }
    get quotesFilterInvoices() {
        return new EventsFilter(this.util.getRawEvents, 5, 447, this.validate);
    }
    get quotesSearchInvoices() {
        return new EventsFilter(this.util.getRawEvents, 5, 448, this.validate);
    }
    get quotesDownloadPdf() {
        return new EventsFilter(this.util.getRawEvents, 5, 449, this.validate);
    }
    get dashboardAccountSettingsImageAction() {
        return new EventsFilter(this.util.getRawEvents, 5, 45, this.validate);
    }
    get quotesResendFromPreviewPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 450, this.validate);
    }
    get quotesRunReportClickGenerateReportClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 451, this.validate);
    }
    get quotesPrintReportClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 452, this.validate);
    }
    get quotesExportReportClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 453, this.validate);
    }
    get quotesPageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 454, this.validate);
    }
    get quotesManagePaymentsBannerDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 455, this.validate);
    }
    get quotesDismissManagePaymentsBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 456, this.validate);
    }
    get quotesClickCtaInManagePaymentsBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 457, this.validate);
    }
    get quotesClickViewOrderBooking() {
        return new EventsFilter(this.util.getRawEvents, 5, 458, this.validate);
    }
    get quotesInvoicesAutomaticallyCreateAndSendInvoices() {
        return new EventsFilter(this.util.getRawEvents, 5, 459, this.validate);
    }
    get dashboardAccountSettingsImageActionSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 46, this.validate);
    }
    get quotesAddCustomerClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 460, this.validate);
    }
    get quotesChooseACustomer() {
        return new EventsFilter(this.util.getRawEvents, 5, 461, this.validate);
    }
    get quotesRemoveCustomer() {
        return new EventsFilter(this.util.getRawEvents, 5, 462, this.validate);
    }
    get quotesDueDateUpdating() {
        return new EventsFilter(this.util.getRawEvents, 5, 463, this.validate);
    }
    get quotesDateOfIssueUpdating() {
        return new EventsFilter(this.util.getRawEvents, 5, 464, this.validate);
    }
    get quotesAddDiscount() {
        return new EventsFilter(this.util.getRawEvents, 5, 465, this.validate);
    }
    get quotesChooseAProduct() {
        return new EventsFilter(this.util.getRawEvents, 5, 466, this.validate);
    }
    get quotesRemoveProduct() {
        return new EventsFilter(this.util.getRawEvents, 5, 467, this.validate);
    }
    get quotesEditProductOrService() {
        return new EventsFilter(this.util.getRawEvents, 5, 468, this.validate);
    }
    get quotesSelectTax() {
        return new EventsFilter(this.util.getRawEvents, 5, 469, this.validate);
    }
    get quotesAddingBusinessLogo() {
        return new EventsFilter(this.util.getRawEvents, 5, 470, this.validate);
    }
    get quotesAddingBusinessName() {
        return new EventsFilter(this.util.getRawEvents, 5, 471, this.validate);
    }
    get quotesClickSaveOnAddBusinessDetailsPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 472, this.validate);
    }
    get quotesBoardingPageClickClose() {
        return new EventsFilter(this.util.getRawEvents, 5, 473, this.validate);
    }
    get quotesGetTheWixAppBannerDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 474, this.validate);
    }
    get quotesDismissGetWixAppBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 475, this.validate);
    }
    get quotesClickCtaInGetWixAppBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 476, this.validate);
    }
    get quotesPrintInvoiceSrc5Evid477() {
        return new EventsFilter(this.util.getRawEvents, 5, 477, this.validate);
    }
    get quotesPayInvoiceClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 478, this.validate);
    }
    get quotesCloseManagePlanBillingTooltip() {
        return new EventsFilter(this.util.getRawEvents, 5, 479, this.validate);
    }
    get quotesCreateInvoiceToggleRecurring() {
        return new EventsFilter(this.util.getRawEvents, 5, 480, this.validate);
    }
    get quotesPrintFromPreviewPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 481, this.validate);
    }
    get quotesPreviewFromInsideTheInvoiceCreation() {
        return new EventsFilter(this.util.getRawEvents, 5, 482, this.validate);
    }
    get quotesCancelFromInsideTheInvoiceCreation() {
        return new EventsFilter(this.util.getRawEvents, 5, 483, this.validate);
    }
    get quotesPreviewFromTable() {
        return new EventsFilter(this.util.getRawEvents, 5, 484, this.validate);
    }
    get mySitesEditSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 486, this.validate);
    }
    get sidebarCategoryItemsShown488() {
        return new EventsFilter(this.util.getRawEvents, 5, 488, this.validate);
    }
    get crmHomeCreateButtonClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 489, this.validate);
    }
    get crmHomeCreateButtonChooseOption() {
        return new EventsFilter(this.util.getRawEvents, 5, 490, this.validate);
    }
    get crmHomeSecondaryCtaClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 491, this.validate);
    }
    get crmHomeCarouselClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 492, this.validate);
    }
    get crmHomeManageYourFinancesChoose() {
        return new EventsFilter(this.util.getRawEvents, 5, 493, this.validate);
    }
    get crmHomeCustomerManagementToolsChoose() {
        return new EventsFilter(this.util.getRawEvents, 5, 494, this.validate);
    }
    get quotesSettingsChangeTabs() {
        return new EventsFilter(this.util.getRawEvents, 5, 497, this.validate);
    }
    get quotesSettingsSuccessToast() {
        return new EventsFilter(this.util.getRawEvents, 5, 498, this.validate);
    }
    get contactEnrichmentTemporaryEvent() {
        return new EventsFilter(this.util.getRawEvents, 5, 499, this.validate);
    }
    get helpWidgetClickOnSeeAllResults() {
        return new EventsFilter(this.util.getRawEvents, 5, 5, this.validate);
    }
    get dashboardClickOnConnectDomain() {
        return new EventsFilter(this.util.getRawEvents, 5, 50, this.validate);
    }
    get dashboardClickOnSelectPremiumPlan() {
        return new EventsFilter(this.util.getRawEvents, 5, 51, this.validate);
    }
    get dashboardClickBuyDomainFirst() {
        return new EventsFilter(this.util.getRawEvents, 5, 52, this.validate);
    }
    get dashboardCloseConnectDomainPopUp() {
        return new EventsFilter(this.util.getRawEvents, 5, 53, this.validate);
    }
    get dashboardBannerLoadingInit() {
        return new EventsFilter(this.util.getRawEvents, 5, 54, this.validate);
    }
    get engageEmailSettingsPageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 540, this.validate);
    }
    get engageEmailSettingsUpdateValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 541, this.validate);
    }
    get engageEmailSettingsSaveChangesClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 542, this.validate);
    }
    get engageEmailSettingsCancelBack() {
        return new EventsFilter(this.util.getRawEvents, 5, 544, this.validate);
    }
    get engageEmailSettingsUnsavedChangesDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 546, this.validate);
    }
    get dashboardBannerLoadingComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 55, this.validate);
    }
    get sideBarItemClick550() {
        return new EventsFilter(this.util.getRawEvents, 5, 550, this.validate);
    }
    get bizManagerBizInfoFieldValueChangedNotSaved() {
        return new EventsFilter(this.util.getRawEvents, 5, 551, this.validate);
    }
    get bizManagerBizInfoClickOnSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 552, this.validate);
    }
    get bizManagerBizInfoSaveSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 553, this.validate);
    }
    get bizManagerBizInfoClickOnDiscardButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 554, this.validate);
    }
    get bizManagerBizInfoApproveDiscard() {
        return new EventsFilter(this.util.getRawEvents, 5, 555, this.validate);
    }
    get bizManagerBizInfoCancelDiscard() {
        return new EventsFilter(this.util.getRawEvents, 5, 556, this.validate);
    }
    get bizManagerBizInfoNavigateAwayDialogIsShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 557, this.validate);
    }
    get bizManagerBizInfoNavigateAwayStayInPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 558, this.validate);
    }
    get bizManagerBizInfoClickOnButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 559, this.validate);
    }
    get siteHistoryViewerNotLoaded() {
        return new EventsFilter(this.util.getRawEvents, 5, 56, this.validate);
    }
    get bizManagerBizInfoNavigateAwayLeaveThisPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 560, this.validate);
    }
    get bizManagerStartLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 561, this.validate);
    }
    get bizManagerFinishLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 562, this.validate);
    }
    get bizManagerModuleLoadingStart() {
        return new EventsFilter(this.util.getRawEvents, 5, 564, this.validate);
    }
    get bizManagerModuleLoadingFinish() {
        return new EventsFilter(this.util.getRawEvents, 5, 565, this.validate);
    }
    get bizManagerNavigateToViewStart() {
        return new EventsFilter(this.util.getRawEvents, 5, 566, this.validate);
    }
    get dashboardClickOnCreatePriorityTicketExperts() {
        return new EventsFilter(this.util.getRawEvents, 5, 57, this.validate);
    }
    get bizManagerClickOnBackInSidebarItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 570, this.validate);
    }
    get bizManagerSetupWidgetIsDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 573, this.validate);
    }
    get bizManagerMinimizeExpandSetupWidget() {
        return new EventsFilter(this.util.getRawEvents, 5, 574, this.validate);
    }
    get bizManagerSetupWidgetShowMessageComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 575, this.validate);
    }
    get bizManagerSetupWidgetClickCtaAfterShowMessageComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 576, this.validate);
    }
    get bizManagerDisplayAnalyticsWidget() {
        return new EventsFilter(this.util.getRawEvents, 5, 577, this.validate);
    }
    get bizManagerClickOnHelpCenterInDashboardFooter() {
        return new EventsFilter(this.util.getRawEvents, 5, 578, this.validate);
    }
    get bizManagerClickOnWidgetCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 579, this.validate);
    }
    get transferSiteResendTransferEmailClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 58, this.validate);
    }
    get bizManagerHtmlLoaded() {
        return new EventsFilter(this.util.getRawEvents, 5, 580, this.validate);
    }
    get bizManagerViewStartLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 581, this.validate);
    }
    get bizManagerViewFinishLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 582, this.validate);
    }
    get bizManagerClickLinkFromTooltip() {
        return new EventsFilter(this.util.getRawEvents, 5, 584, this.validate);
    }
    get bizManagerBusinessInfoMergeDialogShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 585, this.validate);
    }
    get bizManagerBusinessInfoMergeDialogApprove() {
        return new EventsFilter(this.util.getRawEvents, 5, 586, this.validate);
    }
    get bizManagerBusinessInfoMergeDialogClose() {
        return new EventsFilter(this.util.getRawEvents, 5, 587, this.validate);
    }
    get managerFullStoryRecordingStarted() {
        return new EventsFilter(this.util.getRawEvents, 5, 588, this.validate);
    }
    get bizManagerDismissTooltip() {
        return new EventsFilter(this.util.getRawEvents, 5, 589, this.validate);
    }
    get transferSiteGetTransferLinkClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 59, this.validate);
    }
    get bizManagerWhatSNextCloseVideoAsset() {
        return new EventsFilter(this.util.getRawEvents, 5, 592, this.validate);
    }
    get bizManagerShowUrgentMessages() {
        return new EventsFilter(this.util.getRawEvents, 5, 593, this.validate);
    }
    get bizManagerUrgentMessagesClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 594, this.validate);
    }
    get bizManagerDashboardAssetClickOnArrow() {
        return new EventsFilter(this.util.getRawEvents, 5, 597, this.validate);
    }
    get bizManagerOpenPanelStart() {
        return new EventsFilter(this.util.getRawEvents, 5, 598, this.validate);
    }
    get bizManagerOpenPanelFinish() {
        return new EventsFilter(this.util.getRawEvents, 5, 599, this.validate);
    }
    get helpWidgetErrorSearchingItems() {
        return new EventsFilter(this.util.getRawEvents, 5, 6, this.validate);
    }
    get transferSiteCancelTransferClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 60, this.validate);
    }
    get dummyEventForSiteMembers() {
        return new EventsFilter(this.util.getRawEvents, 5, 600, this.validate);
    }
    get transferSiteCancelTransferClickApprove() {
        return new EventsFilter(this.util.getRawEvents, 5, 61, this.validate);
    }
    get transferSiteCancelTransferAbort() {
        return new EventsFilter(this.util.getRawEvents, 5, 62, this.validate);
    }
    get transferSiteReceiverOpenCompleteTransferDialog() {
        return new EventsFilter(this.util.getRawEvents, 5, 64, this.validate);
    }
    get quotesNewQuoteClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 65, this.validate);
    }
    get bizManagerClickOnSetupSkipUndo() {
        return new EventsFilter(this.util.getRawEvents, 5, 656, this.validate);
    }
    get bizManagerOpenPanelInteractive() {
        return new EventsFilter(this.util.getRawEvents, 5, 657, this.validate);
    }
    get dashboardAccountSettingsHoverOverSendConfirmationI() {
        return new EventsFilter(this.util.getRawEvents, 5, 66, this.validate);
    }
    get phoneVerificationHoverOnMotivation() {
        return new EventsFilter(this.util.getRawEvents, 5, 660, this.validate);
    }
    get mobilePhoneNumberConfirmationPopUpClickOkForSendingSms() {
        return new EventsFilter(this.util.getRawEvents, 5, 661, this.validate);
    }
    get mobilePhoneNumberConfirmationPopUpClickConfirmForSmsCode() {
        return new EventsFilter(this.util.getRawEvents, 5, 662, this.validate);
    }
    get phoneVerificationPhoneValidation() {
        return new EventsFilter(this.util.getRawEvents, 5, 665, this.validate);
    }
    get phoneVerificationCodeValidation() {
        return new EventsFilter(this.util.getRawEvents, 5, 666, this.validate);
    }
    get mobilePhoneNumberConfirmationOpenPopUp() {
        return new EventsFilter(this.util.getRawEvents, 5, 667, this.validate);
    }
    get dashboardAccountSettingsClickLearnMoreInI() {
        return new EventsFilter(this.util.getRawEvents, 5, 67, this.validate);
    }
    get bizManagerNewBadgeShow() {
        return new EventsFilter(this.util.getRawEvents, 5, 673, this.validate);
    }
    get mobilePhoneNumberConfirmationClickOnEdit() {
        return new EventsFilter(this.util.getRawEvents, 5, 678, this.validate);
    }
    get accountSettingsClickOnManagePreferences() {
        return new EventsFilter(this.util.getRawEvents, 5, 679, this.validate);
    }
    get dashboardAccountSettingsClickSendConfirmation() {
        return new EventsFilter(this.util.getRawEvents, 5, 68, this.validate);
    }
    get accountSettingsDeactivateAccountClickOnLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 680, this.validate);
    }
    get twoFactorAuthenticationSetupEnableClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 681, this.validate);
    }
    get accountSettingsUpdatesConfirmEnterPassword() {
        return new EventsFilter(this.util.getRawEvents, 5, 682, this.validate);
    }
    get twoFactorAuthenticationSetupSendCode() {
        return new EventsFilter(this.util.getRawEvents, 5, 683, this.validate);
    }
    get twoFactorAuthenticationSetupConfirmCodeClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 684, this.validate);
    }
    get twoFactorAuthenticationSetupEditNumber() {
        return new EventsFilter(this.util.getRawEvents, 5, 685, this.validate);
    }
    get twoFactorAuthenticationSetupDisableAction() {
        return new EventsFilter(this.util.getRawEvents, 5, 686, this.validate);
    }
    get twoFactorAuthenticationSetupDisableClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 687, this.validate);
    }
    get accountSettingsSetPasswordModalLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 688, this.validate);
    }
    get accountSettingsOnRequestToUnlockField() {
        return new EventsFilter(this.util.getRawEvents, 5, 689, this.validate);
    }
    get quotesFilterQuotes() {
        return new EventsFilter(this.util.getRawEvents, 5, 69, this.validate);
    }
    get accountSettingsUpdatesConfirmPasswordModalLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 690, this.validate);
    }
    get exposureEventGoogleConnectSeo() {
        return new EventsFilter(this.util.getRawEvents, 5, 6900, this.validate);
    }
    get accountSettingsForgotPasswordClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 691, this.validate);
    }
    get accountSettingsManagePrivacyPreferences() {
        return new EventsFilter(this.util.getRawEvents, 5, 695, this.validate);
    }
    get accountSettingsTwoFactorAuthenticationSetupSendCodeSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 696, this.validate);
    }
    get accountSettingsTwoFactorAuthenticationSetupSendCodeFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 697, this.validate);
    }
    get dummyNameSrc5Evid7100() {
        return new EventsFilter(this.util.getRawEvents, 5, 7100, this.validate);
    }
    get mySitesClickOnSelectSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 715, this.validate);
    }
    get dashboardViewSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 72, this.validate);
    }
    get quotesClickOnCopyLinkFromMoreActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 727, this.validate);
    }
    get quotesErrorGenerateLinkFromMoreActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 728, this.validate);
    }
    get quotesClickOnCreateLinkFromMoreActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 729, this.validate);
    }
    get dashboardClickOnUpgrade() {
        return new EventsFilter(this.util.getRawEvents, 5, 73, this.validate);
    }
    get quotesSettingsChangeNumbering() {
        return new EventsFilter(this.util.getRawEvents, 5, 730, this.validate);
    }
    get quotesHeaderPreview() {
        return new EventsFilter(this.util.getRawEvents, 5, 732, this.validate);
    }
    get quotesOpenFirstCard() {
        return new EventsFilter(this.util.getRawEvents, 5, 733, this.validate);
    }
    get quotesDropdownOption() {
        return new EventsFilter(this.util.getRawEvents, 5, 734, this.validate);
    }
    get sidebarModalOfFteToTheNewSidebarIsLoaded() {
        return new EventsFilter(this.util.getRawEvents, 5, 735, this.validate);
    }
    get quotesLimit5KCsvDownload() {
        return new EventsFilter(this.util.getRawEvents, 5, 738, this.validate);
    }
    get accountSettingsStep1ToDisablePartnerDashboard() {
        return new EventsFilter(this.util.getRawEvents, 5, 739, this.validate);
    }
    get quotesSearchQuotes() {
        return new EventsFilter(this.util.getRawEvents, 5, 74, this.validate);
    }
    get accountSettingsErrorWhenClickToDisablePartnerDashboard() {
        return new EventsFilter(this.util.getRawEvents, 5, 740, this.validate);
    }
    get accountSettingsDisabledEnabledPartnerDashboard() {
        return new EventsFilter(this.util.getRawEvents, 5, 741, this.validate);
    }
    get crmHomeSendMessageToInbox() {
        return new EventsFilter(this.util.getRawEvents, 5, 745, this.validate);
    }
    get crmHomeAddNewContactWidget() {
        return new EventsFilter(this.util.getRawEvents, 5, 746, this.validate);
    }
    get crmHomeChooseContactFromWidget() {
        return new EventsFilter(this.util.getRawEvents, 5, 747, this.validate);
    }
    get crmHomeSendMessageViaContactsWidget() {
        return new EventsFilter(this.util.getRawEvents, 5, 748, this.validate);
    }
    get quotesEditQuote() {
        return new EventsFilter(this.util.getRawEvents, 5, 75, this.validate);
    }
    get dummyNameSrc5Evid7500() {
        return new EventsFilter(this.util.getRawEvents, 5, 7500, this.validate);
    }
    get quotesPreviewQuote() {
        return new EventsFilter(this.util.getRawEvents, 5, 76, this.validate);
    }
    get enableDisablePartnersDhasboard() {
        return new EventsFilter(this.util.getRawEvents, 5, 7600, this.validate);
    }
    get quotesCreateQuoteSaveDraftClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 77, this.validate);
    }
    get deprecatedUpdateOrCreateLocation() {
        return new EventsFilter(this.util.getRawEvents, 5, 7700, this.validate);
    }
    get deleteLocation() {
        return new EventsFilter(this.util.getRawEvents, 5, 7701, this.validate);
    }
    get setADefaultLocation() {
        return new EventsFilter(this.util.getRawEvents, 5, 7702, this.validate);
    }
    get updateOrCreateMultilocationSrc5Evid7703() {
        return new EventsFilter(this.util.getRawEvents, 5, 7703, this.validate);
    }
    get updateOrCreateMultilocationSrc5Evid7704() {
        return new EventsFilter(this.util.getRawEvents, 5, 7704, this.validate);
    }
    get updateOrCreateMultilocation() {
        return new EventsFilter(this.util.getRawEvents, 5, 7705, this.validate);
    }
    get quotesCreateQuoteSaveDraftComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 78, this.validate);
    }
    get quotesCreateQuoteSaveDraftFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 79, this.validate);
    }
    get automationsSecurityCheckForAmountOfEmailInGetNotifiedAction() {
        return new EventsFilter(this.util.getRawEvents, 5, 7901, this.validate);
    }
    get uouUnsubscribePageView() {
        return new EventsFilter(this.util.getRawEvents, 5, 798, this.validate);
    }
    get uouUnsubscribeClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 799, this.validate);
    }
    get dashboardCreateNewSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 80, this.validate);
    }
    get sampleEventTest() {
        return new EventsFilter(this.util.getRawEvents, 5, 800, this.validate);
    }
    get gdprUouSiteOwnerOpenPrivacyGdprTab() {
        return new EventsFilter(this.util.getRawEvents, 5, 801, this.validate);
    }
    get gdprUouSiteOwnerInitiateGdprRequest() {
        return new EventsFilter(this.util.getRawEvents, 5, 802, this.validate);
    }
    get gdprUouSiteOwnerSelectRequestType() {
        return new EventsFilter(this.util.getRawEvents, 5, 803, this.validate);
    }
    get gdprUouSiteOwnerInsertUouInformation() {
        return new EventsFilter(this.util.getRawEvents, 5, 804, this.validate);
    }
    get gdprUouSiteOwnerReConfirmUouInformation() {
        return new EventsFilter(this.util.getRawEvents, 5, 805, this.validate);
    }
    get gdprUouSiteOwnerUouRequestSubmissionSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 806, this.validate);
    }
    get gdprUouSiteOwnerUouRequestSubmissionFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 807, this.validate);
    }
    get gdprUouSiteOwnerReturnedToPreviousFormStep() {
        return new EventsFilter(this.util.getRawEvents, 5, 808, this.validate);
    }
    get gdprUouSiteOwnerWishToLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 809, this.validate);
    }
    get dashboardSiteSelectorPopupShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 81, this.validate);
    }
    get gdprUsersUserOpenedTheGdprForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 810, this.validate);
    }
    get dummyNameSrc5Evid8100() {
        return new EventsFilter(this.util.getRawEvents, 5, 8100, this.validate);
    }
    get gdprUsersUserSubmitTheGdprForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 811, this.validate);
    }
    get gdprUsersUserGdprFormSubmissionSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 812, this.validate);
    }
    get gdprUsersUserGdprFormSubmissionFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 813, this.validate);
    }
    get cookiesBannerCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 814, this.validate);
    }
    get cookiesBannerSettingsWindowCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 815, this.validate);
    }
    get cookiesBannerPrivacyPolicyClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 816, this.validate);
    }
    get cookiesBannerSettingPageCta() {
        return new EventsFilter(this.util.getRawEvents, 5, 817, this.validate);
    }
    get cookiesBannerSettingPageOptOutConfirmation() {
        return new EventsFilter(this.util.getRawEvents, 5, 818, this.validate);
    }
    get crmHomeViewContactViaContactsWidget() {
        return new EventsFilter(this.util.getRawEvents, 5, 819, this.validate);
    }
    get dashboardSiteSelectorPopupClose() {
        return new EventsFilter(this.util.getRawEvents, 5, 82, this.validate);
    }
    get crmHomeSeeAllContactsViaContactsWidget() {
        return new EventsFilter(this.util.getRawEvents, 5, 820, this.validate);
    }
    get crmHomeImportContactsViaContactsBanner() {
        return new EventsFilter(this.util.getRawEvents, 5, 821, this.validate);
    }
    get crmHomeTasksWidgetAddNewTask() {
        return new EventsFilter(this.util.getRawEvents, 5, 822, this.validate);
    }
    get crmHomeTasksWidgetSeeAllTasks() {
        return new EventsFilter(this.util.getRawEvents, 5, 823, this.validate);
    }
    get dashboardSiteSelectorPopupSelectAction() {
        return new EventsFilter(this.util.getRawEvents, 5, 83, this.validate);
    }
    get quotesEmailQuotesPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 84, this.validate);
    }
    get sendFaqMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 8400, this.validate);
    }
    get crmHomeWorkflowsWidgetGoToWorkflows() {
        return new EventsFilter(this.util.getRawEvents, 5, 844, this.validate);
    }
    get toastShown85() {
        return new EventsFilter(this.util.getRawEvents, 5, 85, this.validate);
    }
    get crmHomeWorkflowsWidgetUseTemplate() {
        return new EventsFilter(this.util.getRawEvents, 5, 852, this.validate);
    }
    get crmHomeWorkflowsWidgetStartFromScratch() {
        return new EventsFilter(this.util.getRawEvents, 5, 853, this.validate);
    }
    get crmHomeIntegrationsCommunicationsViewAll() {
        return new EventsFilter(this.util.getRawEvents, 5, 854, this.validate);
    }
    get crmHomeIntegrationsCommunicationsConnect() {
        return new EventsFilter(this.util.getRawEvents, 5, 855, this.validate);
    }
    get toastClicked86() {
        return new EventsFilter(this.util.getRawEvents, 5, 86, this.validate);
    }
    get crmHomeIntegrationsMarketingViewAll() {
        return new EventsFilter(this.util.getRawEvents, 5, 860, this.validate);
    }
    get dummyNameSrc5Evid8600() {
        return new EventsFilter(this.util.getRawEvents, 5, 8600, this.validate);
    }
    get siteMembersSignupSettingsEditFieldTextSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 864, this.validate);
    }
    get formBuilderGetSubscribersFormManageSubscribersClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 867, this.validate);
    }
    get formBuilderGetSubscribersFormManageSubscribersActionsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 868, this.validate);
    }
    get formBuilderCaptachaStatus() {
        return new EventsFilter(this.util.getRawEvents, 5, 869, this.validate);
    }
    get quotesEmailQuotePageCancelClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 87, this.validate);
    }
    get formBuilderTooltipForAddFieldForbiddenWords() {
        return new EventsFilter(this.util.getRawEvents, 5, 870, this.validate);
    }
    get startLoading8702() {
        return new EventsFilter(this.util.getRawEvents, 5, 8702, this.validate);
    }
    get formBuilderManageChoicesUpdateDisplayName() {
        return new EventsFilter(this.util.getRawEvents, 5, 871, this.validate);
    }
    get formBuilderManageChoicesAddAChoiceClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 872, this.validate);
    }
    get formBuilderManageChoicesEditChoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 873, this.validate);
    }
    get formBuilderManageChoicesDeleteChoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 874, this.validate);
    }
    get formBuilderManageChoicesUpdateChoiceName() {
        return new EventsFilter(this.util.getRawEvents, 5, 875, this.validate);
    }
    get formBuilderManageChoicesSetAsDefaultChoice() {
        return new EventsFilter(this.util.getRawEvents, 5, 876, this.validate);
    }
    get formBuilderManageChoicesDuplicateChoiceClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 877, this.validate);
    }
    get formBuilderManageChoicesDuplicateChoiceComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 878, this.validate);
    }
    get formBuilderManageChoicesReorderChoiceComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 879, this.validate);
    }
    get quotesPrintQuote() {
        return new EventsFilter(this.util.getRawEvents, 5, 88, this.validate);
    }
    get formBuilderMultiStepDragStep() {
        return new EventsFilter(this.util.getRawEvents, 5, 882, this.validate);
    }
    get formBuilderMultiStepClickOnManageSteps() {
        return new EventsFilter(this.util.getRawEvents, 5, 883, this.validate);
    }
    get formBuilderMultiStepEditStepActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 885, this.validate);
    }
    get formBuilderMultiStepAddNewStep() {
        return new EventsFilter(this.util.getRawEvents, 5, 886, this.validate);
    }
    get formBuilderMultiStepAddNewField() {
        return new EventsFilter(this.util.getRawEvents, 5, 887, this.validate);
    }
    get formBuilderBackofficeFormSubscriberUserActions() {
        return new EventsFilter(this.util.getRawEvents, 5, 888, this.validate);
    }
    get formBuilderMultiStepClickOnAscendPlan() {
        return new EventsFilter(this.util.getRawEvents, 5, 889, this.validate);
    }
    get quotesDownloadPdfQuote() {
        return new EventsFilter(this.util.getRawEvents, 5, 89, this.validate);
    }
    get formBuilderMultiStepBackNextButtonChangeText() {
        return new EventsFilter(this.util.getRawEvents, 5, 890, this.validate);
    }
    get finishLoading8900() {
        return new EventsFilter(this.util.getRawEvents, 5, 8900, this.validate);
    }
    get exposureEventForPaginationTest() {
        return new EventsFilter(this.util.getRawEvents, 5, 8906, this.validate);
    }
    get formBuilderMultiStepBackNextButtonFinishUpdateText() {
        return new EventsFilter(this.util.getRawEvents, 5, 891, this.validate);
    }
    get invoiceSettingsSelectTaxDetailsCustomFieldCheckbox() {
        return new EventsFilter(this.util.getRawEvents, 5, 8917, this.validate);
    }
    get invoiceSettingsClickOnLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 8918, this.validate);
    }
    get formBuilderMultiStepUouClickOnBackNextSubmit() {
        return new EventsFilter(this.util.getRawEvents, 5, 892, this.validate);
    }
    get bookingAnywhereNoPermission() {
        return new EventsFilter(this.util.getRawEvents, 5, 8920, this.validate);
    }
    get mySitesMainPageFinishLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 8921, this.validate);
    }
    get addNoteTermClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 8923, this.validate);
    }
    get mySitesSiteLessLoaded() {
        return new EventsFilter(this.util.getRawEvents, 5, 8926, this.validate);
    }
    get wixInvoicesClickingOnGroupsDropDownInSettings() {
        return new EventsFilter(this.util.getRawEvents, 5, 8929, this.validate);
    }
    get formBuilderBackofficeFormDeleteKeepSubscribeItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 893, this.validate);
    }
    get invoicesSettingsSaveButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 8930, this.validate);
    }
    get manageFolderDropDownListShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 8933, this.validate);
    }
    get wixInvoicesLeavingSettingsWithoutSavingPopUp() {
        return new EventsFilter(this.util.getRawEvents, 5, 8934, this.validate);
    }
    get discardModalClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 8935, this.validate);
    }
    get wixInvoicesClickOnTabsToChoseProductFrom() {
        return new EventsFilter(this.util.getRawEvents, 5, 8936, this.validate);
    }
    get invoiceSettingsSaveButtonClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 8937, this.validate);
    }
    get wixInvoicesSavingItemDescription() {
        return new EventsFilter(this.util.getRawEvents, 5, 8938, this.validate);
    }
    get gizaApiCallReceived() {
        return new EventsFilter(this.util.getRawEvents, 5, 8939, this.validate);
    }
    get formBuilderMultiStepClickOnRestoreButtons() {
        return new EventsFilter(this.util.getRawEvents, 5, 894, this.validate);
    }
    get clickDoneToEditFilterView() {
        return new EventsFilter(this.util.getRawEvents, 5, 8940, this.validate);
    }
    get clickOnSaveView() {
        return new EventsFilter(this.util.getRawEvents, 5, 8941, this.validate);
    }
    get mySitesEditCustomizeColumnView() {
        return new EventsFilter(this.util.getRawEvents, 5, 8942, this.validate);
    }
    get mySitesClickOnGoToDashboardLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 8943, this.validate);
    }
    get mySitesClickOnBackToMySitesFromAnEmptyFolder() {
        return new EventsFilter(this.util.getRawEvents, 5, 8944, this.validate);
    }
    get mySitesSiteListPageStartLoading() {
        return new EventsFilter(this.util.getRawEvents, 5, 8945, this.validate);
    }
    get mySitesSiteListFolderNavigation() {
        return new EventsFilter(this.util.getRawEvents, 5, 8946, this.validate);
    }
    get mySitesClickSiteItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 8947, this.validate);
    }
    get mySitesClicksActionsInSidePanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 8948, this.validate);
    }
    get clickStartGuideOrCloseDoItWithMeTool() {
        return new EventsFilter(this.util.getRawEvents, 5, 8949, this.validate);
    }
    get wixInvoicesChargeWithCreditCardClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 8954, this.validate);
    }
    get wixInvoicesPaymentMethodClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 8956, this.validate);
    }
    get wixInvoicesToastIsShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 8957, this.validate);
    }
    get wixInvoicesRequestPaymentClickInAutomation() {
        return new EventsFilter(this.util.getRawEvents, 5, 8958, this.validate);
    }
    get wixInvoicesChargeWithCreditCardClickInTheModalMoto() {
        return new EventsFilter(this.util.getRawEvents, 5, 8959, this.validate);
    }
    get formBuilderSubmitButtonSettingsClickOnSetupPaymentForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 896, this.validate);
    }
    get wixInvoicesMotoCashierWidgetIsShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 8960, this.validate);
    }
    get wixInvoicesMotoPaidModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 8961, this.validate);
    }
    get mySitesRefreshAfterMoveToFolder() {
        return new EventsFilter(this.util.getRawEvents, 5, 8962, this.validate);
    }
    get wixInvoicesAutomationDynamicValueClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 8963, this.validate);
    }
    get mySitesErrorPage() {
        return new EventsFilter(this.util.getRawEvents, 5, 8964, this.validate);
    }
    get wixInvoicesClickingOnNextMobile() {
        return new EventsFilter(this.util.getRawEvents, 5, 8965, this.validate);
    }
    get wixInvoicesBannerToUpgradeToPremiumShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 8966, this.validate);
    }
    get wixInvoicesBannerToUpgradeToPremiumClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 8967, this.validate);
    }
    get wixInvoicesOpenShareLinkModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 8968, this.validate);
    }
    get wixInvoicesBannerToConnectPaymentMethods() {
        return new EventsFilter(this.util.getRawEvents, 5, 8969, this.validate);
    }
    get wixInvoicesBannerToConnectPaymentMethodClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 8970, this.validate);
    }
    get gizaFirstBiteIsShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 8971, this.validate);
    }
    get gizaFirstInteractionInTheSessionIsAvailable() {
        return new EventsFilter(this.util.getRawEvents, 5, 8972, this.validate);
    }
    get wixInvoicesClickOnDynamicValueInAutomation() {
        return new EventsFilter(this.util.getRawEvents, 5, 8973, this.validate);
    }
    get priorityASuggestionCarouselInteraction() {
        return new EventsFilter(this.util.getRawEvents, 5, 8974, this.validate);
    }
    get priorityAEssentialShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 8975, this.validate);
    }
    get clickOnCreateNewFolder() {
        return new EventsFilter(this.util.getRawEvents, 5, 8978, this.validate);
    }
    get createNewSiteCreationMenuIsShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 8989, this.validate);
    }
    get clickOnAddAtTheDynamicFieldModal() {
        return new EventsFilter(this.util.getRawEvents, 5, 8993, this.validate);
    }
    get getBrandedAppAlertIsShownAutomationSendAPushNotification() {
        return new EventsFilter(this.util.getRawEvents, 5, 8995, this.validate);
    }
    get quotesDeleteQuoteClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 90, this.validate);
    }
    get formBuildrUouSubmitFormClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 900, this.validate);
    }
    get sdkLoaded9000() {
        return new EventsFilter(this.util.getRawEvents, 5, 9000, this.validate);
    }
    get gizaPageStartLoading9001() {
        return new EventsFilter(this.util.getRawEvents, 5, 9001, this.validate);
    }
    get gizaPageFinishLoading9002() {
        return new EventsFilter(this.util.getRawEvents, 5, 9002, this.validate);
    }
    get formBuildrUouSubmitFormSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 901, this.validate);
    }
    get mySitesEditTemplateButtonClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 9012, this.validate);
    }
    get mySitesClickOnSeeAllTemplates() {
        return new EventsFilter(this.util.getRawEvents, 5, 9013, this.validate);
    }
    get createNewWorkspace() {
        return new EventsFilter(this.util.getRawEvents, 5, 9014, this.validate);
    }
    get workspaceSelectorSelectWorkspace() {
        return new EventsFilter(this.util.getRawEvents, 5, 9015, this.validate);
    }
    get workspaceActionMenuDropDownListShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 9016, this.validate);
    }
    get workspaceActionMenuItemClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 9017, this.validate);
    }
    get workspaceSelectorClickOnLearnMoreInTooltips() {
        return new EventsFilter(this.util.getRawEvents, 5, 9018, this.validate);
    }
    get workspaceSelectorWorkspaceSettingClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 9019, this.validate);
    }
    get formBuildrUouSubmitFormError() {
        return new EventsFilter(this.util.getRawEvents, 5, 902, this.validate);
    }
    get accountSettingsCloseYourWorkspacePageLoad() {
        return new EventsFilter(this.util.getRawEvents, 5, 9021, this.validate);
    }
    get accountSettingsCloseYourWorkspaceCloseWorkspaceRequestSummited() {
        return new EventsFilter(this.util.getRawEvents, 5, 9022, this.validate);
    }
    get workspaceSettingsDeleteAccountWorkspace() {
        return new EventsFilter(this.util.getRawEvents, 5, 9023, this.validate);
    }
    get workspaceSettingsChangesSaved() {
        return new EventsFilter(this.util.getRawEvents, 5, 9024, this.validate);
    }
    get wixInvoicesBannerWhenCreatingInvoiceOrder() {
        return new EventsFilter(this.util.getRawEvents, 5, 9025, this.validate);
    }
    get allPageLoadersAreDown9029() {
        return new EventsFilter(this.util.getRawEvents, 5, 9029, this.validate);
    }
    get formBuilderManageFieldsAddNewFieldCompleted() {
        return new EventsFilter(this.util.getRawEvents, 5, 903, this.validate);
    }
    get invoicesSettingsSelectPaymentsReceivedOption() {
        return new EventsFilter(this.util.getRawEvents, 5, 9032, this.validate);
    }
    get mySitesDebuggingEvent() {
        return new EventsFilter(this.util.getRawEvents, 5, 9034, this.validate);
    }
    get searchInSitesDropdown() {
        return new EventsFilter(this.util.getRawEvents, 5, 9039, this.validate);
    }
    get formBuildrUouSubmitFormSuccessSendingDataToFormBuilderServer() {
        return new EventsFilter(this.util.getRawEvents, 5, 904, this.validate);
    }
    get searchInSitesDropdownEndProcess() {
        return new EventsFilter(this.util.getRawEvents, 5, 9040, this.validate);
    }
    get devModeOnEntryPoint() {
        return new EventsFilter(this.util.getRawEvents, 5, 9041, this.validate);
    }
    get gizaClickOnCollapseExpandSidebar() {
        return new EventsFilter(this.util.getRawEvents, 5, 9042, this.validate);
    }
    get formBuilderDesignPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 906, this.validate);
    }
    get formBuildrComponentLoadTime() {
        return new EventsFilter(this.util.getRawEvents, 5, 907, this.validate);
    }
    get formBuilderDeleteForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 908, this.validate);
    }
    get formBuilderFormCopyAttempt() {
        return new EventsFilter(this.util.getRawEvents, 5, 909, this.validate);
    }
    get quotesDeleteQuotePageOk() {
        return new EventsFilter(this.util.getRawEvents, 5, 91, this.validate);
    }
    get formBuilderTemplatePickerOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 910, this.validate);
    }
    get dummyNameSrc5Evid9100() {
        return new EventsFilter(this.util.getRawEvents, 5, 9100, this.validate);
    }
    get formBuilderTemplatePickerClickOnTemplate() {
        return new EventsFilter(this.util.getRawEvents, 5, 911, this.validate);
    }
    get formBuilderTemplatePickerChooseTemplate() {
        return new EventsFilter(this.util.getRawEvents, 5, 912, this.validate);
    }
    get formBuilderManageFieldsPanelOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 913, this.validate);
    }
    get formBuilderManageFieldsDeleteField() {
        return new EventsFilter(this.util.getRawEvents, 5, 914, this.validate);
    }
    get formBuilderManageFieldsAddNewFieldDropDownOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 915, this.validate);
    }
    get formBuilderManageFieldsAddNewField() {
        return new EventsFilter(this.util.getRawEvents, 5, 916, this.validate);
    }
    get formBuilderEditFieldPanelOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 917, this.validate);
    }
    get formBuilderEditFieldUpdateValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 918, this.validate);
    }
    get formBuilderEditFieldUpdateMandatoryValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 919, this.validate);
    }
    get quotesDeleteQuotePageCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 92, this.validate);
    }
    get formBuilderEditFieldConnectFieldDropDownOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 920, this.validate);
    }
    get appInstallationInitiate() {
        return new EventsFilter(this.util.getRawEvents, 5, 9201, this.validate);
    }
    get appInstallationCompleted() {
        return new EventsFilter(this.util.getRawEvents, 5, 9202, this.validate);
    }
    get appsBulkInstallationInitiate() {
        return new EventsFilter(this.util.getRawEvents, 5, 9203, this.validate);
    }
    get appsBulkInstallationsCompleted() {
        return new EventsFilter(this.util.getRawEvents, 5, 9204, this.validate);
    }
    get formBuilderEditFieldConnectField() {
        return new EventsFilter(this.util.getRawEvents, 5, 921, this.validate);
    }
    get formBuilderFormSettingsPanelOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 922, this.validate);
    }
    get formBuilderFormSettingsUpdateValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 923, this.validate);
    }
    get formBuilderFormLayoutOpenAndChange() {
        return new EventsFilter(this.util.getRawEvents, 5, 924, this.validate);
    }
    get formBuilderFormSettingsChangeDuration() {
        return new EventsFilter(this.util.getRawEvents, 5, 925, this.validate);
    }
    get formBuilderFormSettingsMsgLink() {
        return new EventsFilter(this.util.getRawEvents, 5, 926, this.validate);
    }
    get formBuilderFormSettingsLinkSelection() {
        return new EventsFilter(this.util.getRawEvents, 5, 927, this.validate);
    }
    get formBuilderFormSettingsCreateSubmissionsTableRequest() {
        return new EventsFilter(this.util.getRawEvents, 5, 928, this.validate);
    }
    get formBuilderFormSettingsCreateSubmissionsTableSuccess() {
        return new EventsFilter(this.util.getRawEvents, 5, 929, this.validate);
    }
    get quotesMarkAsAccepted() {
        return new EventsFilter(this.util.getRawEvents, 5, 93, this.validate);
    }
    get formBuilderBizMgrAddNewFormClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 930, this.validate);
    }
    get formBuilderBizMgrViewFormSubmissionsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 931, this.validate);
    }
    get formBuilderManageFieldsDuplicateField() {
        return new EventsFilter(this.util.getRawEvents, 5, 932, this.validate);
    }
    get formBuilderRemoveSubmitButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 933, this.validate);
    }
    get formBuilderPremiumPanelOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 934, this.validate);
    }
    get formBuilderPremiumPanelClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 935, this.validate);
    }
    get formBuilderUserSaveAnEditorTemplateWithForm() {
        return new EventsFilter(this.util.getRawEvents, 5, 936, this.validate);
    }
    get formBuilderFormCopyAttemptFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 937, this.validate);
    }
    get formBuilderSubmitButtonUploadFileClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 938, this.validate);
    }
    get formBuilderPresetGlobalDesignFailed() {
        return new EventsFilter(this.util.getRawEvents, 5, 939, this.validate);
    }
    get quotesCreateInvoiceFromQuotes() {
        return new EventsFilter(this.util.getRawEvents, 5, 94, this.validate);
    }
    get formBuilderManageFieldsClickOnFieldButtonLearnAllAboutThis() {
        return new EventsFilter(this.util.getRawEvents, 5, 940, this.validate);
    }
    get biProfileDataForSiteDetailsActionsWidget() {
        return new EventsFilter(this.util.getRawEvents, 5, 9400, this.validate);
    }
    get formBuilderAdiEditFormClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 941, this.validate);
    }
    get formBuilderAdiEditFormCompleted() {
        return new EventsFilter(this.util.getRawEvents, 5, 942, this.validate);
    }
    get formBuilderAdiReorderFieldClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 943, this.validate);
    }
    get formBuilderAdiReorderFieldComplete() {
        return new EventsFilter(this.util.getRawEvents, 5, 944, this.validate);
    }
    get formBuilderClickOnGfppOfConnectPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 945, this.validate);
    }
    get formBuilderFormLayoutSpaceBetweenRows() {
        return new EventsFilter(this.util.getRawEvents, 5, 946, this.validate);
    }
    get formBuilderFormLayoutSpaceBetweenColumns() {
        return new EventsFilter(this.util.getRawEvents, 5, 947, this.validate);
    }
    get formBuilderFormLayoutTextAlignment() {
        return new EventsFilter(this.util.getRawEvents, 5, 948, this.validate);
    }
    get formBuilderFormLayoutTitleAndPlaceholderPadding() {
        return new EventsFilter(this.util.getRawEvents, 5, 949, this.validate);
    }
    get quotesMarkAsAcceptedSrc5Evid95() {
        return new EventsFilter(this.util.getRawEvents, 5, 95, this.validate);
    }
    get formBuilderFormLayoutSpaceBetweenFieldTitleAndInput() {
        return new EventsFilter(this.util.getRawEvents, 5, 950, this.validate);
    }
    get formBuilderFormLayoutFieldTitlePadding() {
        return new EventsFilter(this.util.getRawEvents, 5, 951, this.validate);
    }
    get formBuilderGlobalLayoutShowFieldTitleToggle() {
        return new EventsFilter(this.util.getRawEvents, 5, 952, this.validate);
    }
    get formBuilderFormSettingsInnerTabsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 953, this.validate);
    }
    get formBuilderFormSettingsSettingsTabsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 954, this.validate);
    }
    get formBuilderFormSettingsGoToAutomationsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 955, this.validate);
    }
    get formBuilderFormSettingsContactListClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 956, this.validate);
    }
    get formBuilderFormSettingsSupportAboutWixFormsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 957, this.validate);
    }
    get formBuilderFormSettingsSupportWixSupportClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 958, this.validate);
    }
    get formBuilderFormSettingsSupportSendYourFeedbackClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 959, this.validate);
    }
    get quotesCreateInvoiceFromQuotesSrc5Evid96() {
        return new EventsFilter(this.util.getRawEvents, 5, 96, this.validate);
    }
    get formBuilderFormSettingsSupportAddAReviewClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 960, this.validate);
    }
    get formBuilderFormSettingsCreateAutomation() {
        return new EventsFilter(this.util.getRawEvents, 5, 961, this.validate);
    }
    get formBuilderFormSettingsInformationIconHoverClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 962, this.validate);
    }
    get formBuilderFormSettingsPublishYourSitePopupPresented() {
        return new EventsFilter(this.util.getRawEvents, 5, 963, this.validate);
    }
    get formBuilderFormSettingsPublishYourSitePopupSelection() {
        return new EventsFilter(this.util.getRawEvents, 5, 964, this.validate);
    }
    get formBuilderFormSettingsSubmitMessageChooseLinkClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 965, this.validate);
    }
    get formBuilderFormSettingsSubmitMessageRestoreSuccessMessage() {
        return new EventsFilter(this.util.getRawEvents, 5, 966, this.validate);
    }
    get formBuilderFormSettingsSubmitMessageRestoreSubmitButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 967, this.validate);
    }
    get formBuilderPaymentFormSettingGetStarted() {
        return new EventsFilter(this.util.getRawEvents, 5, 968, this.validate);
    }
    get formBuilderPaymentFormSettingLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 969, this.validate);
    }
    get removePhoneNumberRemoveClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 97, this.validate);
    }
    get formBuilderPaymentFormSetupIsDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 970, this.validate);
    }
    get unsubscriptionPageIsShown() {
        return new EventsFilter(this.util.getRawEvents, 5, 9700, this.validate);
    }
    get unsubscriptionPageClickOnUnsubscribeButton() {
        return new EventsFilter(this.util.getRawEvents, 5, 9701, this.validate);
    }
    get formBuilderPaymentFormSetupGetStartedClickOnDemoSite() {
        return new EventsFilter(this.util.getRawEvents, 5, 971, this.validate);
    }
    get formBuilderPaymentFormSetupAddUpdateItem() {
        return new EventsFilter(this.util.getRawEvents, 5, 972, this.validate);
    }
    get formBuilderPaymentFormSetupItemPriceChangeCurrency() {
        return new EventsFilter(this.util.getRawEvents, 5, 973, this.validate);
    }
    get formBuilderPaymentFormSetupConnectPaymentMethod() {
        return new EventsFilter(this.util.getRawEvents, 5, 974, this.validate);
    }
    get formBuilderPaymentFormSetupContinueToSetupFromBizmg() {
        return new EventsFilter(this.util.getRawEvents, 5, 975, this.validate);
    }
    get formBuilderPaymentFormSetupChangePaymentMethod() {
        return new EventsFilter(this.util.getRawEvents, 5, 976, this.validate);
    }
    get formBuilderPaymentFormSaveAndRefreshPopUpDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 977, this.validate);
    }
    get formBuilderPaymentFormSaveAndRefreshPopUpUserSelection() {
        return new EventsFilter(this.util.getRawEvents, 5, 978, this.validate);
    }
    get formBuilderPaymentFormViewPublishSitePopUpDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 979, this.validate);
    }
    get removePhoneNumberRemoveNumberClicked() {
        return new EventsFilter(this.util.getRawEvents, 5, 98, this.validate);
    }
    get formBuilderPaymentFormViewPublishSitePopUpUserSelection() {
        return new EventsFilter(this.util.getRawEvents, 5, 980, this.validate);
    }
    get formBuilderPaymentFormRemovePaymentPopUpDisplayed() {
        return new EventsFilter(this.util.getRawEvents, 5, 981, this.validate);
    }
    get formBuilderPaymentFormRemovePaymentPopUpUserSelection() {
        return new EventsFilter(this.util.getRawEvents, 5, 982, this.validate);
    }
    get formBuilderPaymentFormOpenChangeRemoveDisconnectPayment() {
        return new EventsFilter(this.util.getRawEvents, 5, 983, this.validate);
    }
    get formBuilderPaymentFormClickOnNextInPaymentFormSetup() {
        return new EventsFilter(this.util.getRawEvents, 5, 984, this.validate);
    }
    get formBuilderPaymentFormTooltipHover() {
        return new EventsFilter(this.util.getRawEvents, 5, 985, this.validate);
    }
    get formBuilderSubmissionSettingsPanelOpened() {
        return new EventsFilter(this.util.getRawEvents, 5, 986, this.validate);
    }
    get formBuilderCategoryPickerInAddPanel() {
        return new EventsFilter(this.util.getRawEvents, 5, 987, this.validate);
    }
    get formBuilderPaymentFormClickOnGoToSetup() {
        return new EventsFilter(this.util.getRawEvents, 5, 988, this.validate);
    }
    get formBuilderTemplatePickerChooseTemplateStartToInstall() {
        return new EventsFilter(this.util.getRawEvents, 5, 989, this.validate);
    }
    get removePhoneNumberPhoneRemovalAsked() {
        return new EventsFilter(this.util.getRawEvents, 5, 99, this.validate);
    }
    get formBuilderFormSettingsNotificationsHelpClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 990, this.validate);
    }
    get formBuilderFormSettingsUpdateValueClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 991, this.validate);
    }
    get formBuilderFormSettingsContactListContactLabelsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 992, this.validate);
    }
    get formBuilderFormSettingsContactListContactLabelsUpdateValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 993, this.validate);
    }
    get formBuilderFormSettingsContactListChooseSavedFieldsClick() {
        return new EventsFilter(this.util.getRawEvents, 5, 994, this.validate);
    }
    get formBuilderFormSettingsContactListChooseSavedFieldsUpdateValue() {
        return new EventsFilter(this.util.getRawEvents, 5, 995, this.validate);
    }
    get formBuilderFormSettingsContactListChooseSavedFieldsSave() {
        return new EventsFilter(this.util.getRawEvents, 5, 996, this.validate);
    }
    get formBuilderHoverOnUpgradeAscend() {
        return new EventsFilter(this.util.getRawEvents, 5, 997, this.validate);
    }
    get formBuilderFormSettingsContactListChooseSavedFieldsCancel() {
        return new EventsFilter(this.util.getRawEvents, 5, 998, this.validate);
    }
    get formBuilderFormSettingsContactListChooseSavedFieldsLearnMore() {
        return new EventsFilter(this.util.getRawEvents, 5, 999, this.validate);
    }
    get devCenterWebhookSubscribed() {
        return new EventsFilter(this.util.getRawEvents, 96, 1030, this.validate);
    }
    get dcWebhookUpdated() {
        return new EventsFilter(this.util.getRawEvents, 96, 1031, this.validate);
    }
    get devCenterComponentTranslationAddingTranslatableComponent() {
        return new EventsFilter(this.util.getRawEvents, 96, 11401, this.validate);
    }
    get devCenterComponentTranslationAddLanguageToAnApp() {
        return new EventsFilter(this.util.getRawEvents, 96, 11402, this.validate);
    }
    get devCenterComponentTranslationAddTranslationToAComponent() {
        return new EventsFilter(this.util.getRawEvents, 96, 11403, this.validate);
    }
    get dummyName() {
        return new EventsFilter(this.util.getRawEvents, 96, 11700, this.validate);
    }
    get dcAppVersionsVersionAdded() {
        return new EventsFilter(this.util.getRawEvents, 96, 11701, this.validate);
    }
    get dcAppVersionsStatusUpdated() {
        return new EventsFilter(this.util.getRawEvents, 96, 11702, this.validate);
    }
    get dummyNameSrc96Evid11900() {
        return new EventsFilter(this.util.getRawEvents, 96, 11900, this.validate);
    }
    get dummyNameSrc96Evid12200() {
        return new EventsFilter(this.util.getRawEvents, 96, 12200, this.validate);
    }
}
export const interceptor = (logDir) => new WixBiSchemaNodeTestkit(logDir);
export { EventsFilter, validateSingleEvent };
