// @wix/auto_sdk_ecom_local-delivery-options/build/internal/es/meta.mjs
// src/ecom-v1-local-delivery-option-local-delivery-options.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/wix-ecommerce-shipments-local-delivery-web",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/stores/v1/localDeliveryOptions",
        destPath: "/v1/localDeliveryOptions"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/local-delivery-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/local-delivery-server",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_local-delivery-options";
function createLocalDeliveryOption(payload) {
  function __createLocalDeliveryOption({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
          { path: "localDeliveryOption.radiusConfig.address.geocode.latitude" },
          {
            path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
          },
          { path: "localDeliveryOption.customAreaConfig.geocodes.latitude" },
          { path: "localDeliveryOption.customAreaConfig.geocodes.longitude" },
          { path: "localDeliveryOption.radiusConfig.radius" }
        ]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "localDeliveryOption.createdDate" },
          { path: "localDeliveryOption.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.CreateLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createLocalDeliveryOption;
}
function getLocalDeliveryOption(payload) {
  function __getLocalDeliveryOption({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "GET",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.GetLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/{localDeliveryOptionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
            protoPath: "/v1/local-delivery-options/{localDeliveryOptionId}",
            data: payload,
            host
          }),
          params: toURLSearchParams(payload)
        }
      ]
    };
    return metadata;
  }
  return __getLocalDeliveryOption;
}
function listLocalDeliveryOptions(payload) {
  function __listLocalDeliveryOptions({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "GET",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.ListLocalDeliveryOptions",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/localDeliveryOptions",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOptions.radiusConfig.radius" }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOptions.createdDate" },
            { path: "localDeliveryOptions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listLocalDeliveryOptions;
}
function queryLocalDeliveryOptions(payload) {
  function __queryLocalDeliveryOptions({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.QueryLocalDeliveryOptions",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOptions.radiusConfig.radius" }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOptions.createdDate" },
            { path: "localDeliveryOptions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryLocalDeliveryOptions;
}
function updateLocalDeliveryOption(payload) {
  function __updateLocalDeliveryOption({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
          { path: "localDeliveryOption.radiusConfig.address.geocode.latitude" },
          {
            path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
          },
          { path: "localDeliveryOption.customAreaConfig.geocodes.latitude" },
          { path: "localDeliveryOption.customAreaConfig.geocodes.longitude" },
          { path: "localDeliveryOption.radiusConfig.radius" }
        ]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "localDeliveryOption.createdDate" },
          { path: "localDeliveryOption.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "PATCH",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.UpdateLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/{localDeliveryOption.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateLocalDeliveryOption;
}
function deleteLocalDeliveryOption(payload) {
  function __deleteLocalDeliveryOption({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "DELETE",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.DeleteLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/{localDeliveryOptionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteLocalDeliveryOption;
}
function bulkCreateLocalDeliveryOption(payload) {
  function __bulkCreateLocalDeliveryOption({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
          {
            path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
          },
          {
            path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
          },
          { path: "localDeliveryOptions.customAreaConfig.geocodes.latitude" },
          { path: "localDeliveryOptions.customAreaConfig.geocodes.longitude" },
          { path: "localDeliveryOptions.radiusConfig.radius" }
        ]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "localDeliveryOptions.createdDate" },
          { path: "localDeliveryOptions.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.BulkCreateLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/bulk/local-delivery-options/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOptions.radiusConfig.radius" }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOptions.createdDate" },
            { path: "localDeliveryOptions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateLocalDeliveryOption;
}
function bulkUpdateLocalDeliveryOption(payload) {
  function __bulkUpdateLocalDeliveryOption({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
          {
            path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
          },
          {
            path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
          },
          { path: "localDeliveryOptions.customAreaConfig.geocodes.latitude" },
          { path: "localDeliveryOptions.customAreaConfig.geocodes.longitude" },
          { path: "localDeliveryOptions.radiusConfig.radius" }
        ]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "localDeliveryOptions.createdDate" },
          { path: "localDeliveryOptions.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "PATCH",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.BulkUpdateLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/bulk/local-delivery-options/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOptions.radiusConfig.radius" }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOptions.createdDate" },
            { path: "localDeliveryOptions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateLocalDeliveryOption;
}
function bulkDeleteLocalDeliveryOption(payload) {
  function __bulkDeleteLocalDeliveryOption({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "DELETE",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.BulkDeleteLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/bulk/local-delivery-options/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeleteLocalDeliveryOption;
}
function addDeliveryRegion(payload) {
  function __addDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.AddDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/add-delivery-region",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addDeliveryRegion;
}
function removeDeliveryRegion(payload) {
  function __removeDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.RemoveDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/remove-delivery-region",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removeDeliveryRegion;
}

// src/ecom-v1-local-delivery-option-local-delivery-options.meta.ts
function createLocalDeliveryOption2() {
  const payload = {};
  const getRequestOptions = createLocalDeliveryOption(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/local-delivery-options",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getLocalDeliveryOption2() {
  const payload = { localDeliveryOptionId: ":localDeliveryOptionId" };
  const getRequestOptions = getLocalDeliveryOption(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/local-delivery-options/{localDeliveryOptionId}",
    pathParams: { localDeliveryOptionId: "localDeliveryOptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listLocalDeliveryOptions2() {
  const payload = {};
  const getRequestOptions = listLocalDeliveryOptions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/localDeliveryOptions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryLocalDeliveryOptions2() {
  const payload = {};
  const getRequestOptions = queryLocalDeliveryOptions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/local-delivery-options/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateLocalDeliveryOption2() {
  const payload = {
    localDeliveryOption: { id: ":localDeliveryOptionId" }
  };
  const getRequestOptions = updateLocalDeliveryOption(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/local-delivery-options/{localDeliveryOption.id}",
    pathParams: { localDeliveryOptionId: "localDeliveryOptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteLocalDeliveryOption2() {
  const payload = { localDeliveryOptionId: ":localDeliveryOptionId" };
  const getRequestOptions = deleteLocalDeliveryOption(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/local-delivery-options/{localDeliveryOptionId}",
    pathParams: { localDeliveryOptionId: "localDeliveryOptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateLocalDeliveryOption2() {
  const payload = {};
  const getRequestOptions = bulkCreateLocalDeliveryOption(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/local-delivery-options/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateLocalDeliveryOption2() {
  const payload = {};
  const getRequestOptions = bulkUpdateLocalDeliveryOption(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/bulk/local-delivery-options/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteLocalDeliveryOption2() {
  const payload = {};
  const getRequestOptions = bulkDeleteLocalDeliveryOption(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/bulk/local-delivery-options/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addDeliveryRegion2() {
  const payload = {};
  const getRequestOptions = addDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/local-delivery-options/add-delivery-region",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeDeliveryRegion2() {
  const payload = {};
  const getRequestOptions = removeDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/local-delivery-options/remove-delivery-region",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  addDeliveryRegion2 as addDeliveryRegion,
  bulkCreateLocalDeliveryOption2 as bulkCreateLocalDeliveryOption,
  bulkDeleteLocalDeliveryOption2 as bulkDeleteLocalDeliveryOption,
  bulkUpdateLocalDeliveryOption2 as bulkUpdateLocalDeliveryOption,
  createLocalDeliveryOption2 as createLocalDeliveryOption,
  deleteLocalDeliveryOption2 as deleteLocalDeliveryOption,
  getLocalDeliveryOption2 as getLocalDeliveryOption,
  listLocalDeliveryOptions2 as listLocalDeliveryOptions,
  queryLocalDeliveryOptions2 as queryLocalDeliveryOptions,
  removeDeliveryRegion2 as removeDeliveryRegion,
  updateLocalDeliveryOption2 as updateLocalDeliveryOption
};
