// @wix/headless-stores/cjs/dist/react/BuyNow.js
import { useService } from "@wix/services-manager-react";
import { BuyNowServiceDefinition } from "../services/buy-now-service.js";
/**
 * A headless component that provides buy now functionality using the render props pattern.
 *
 * This component manages the state and actions for a "buy now" flow, allowing consumers
 * to render their own UI while accessing the underlying buy now functionality.
 * @example
 * ```tsx
 * <BuyNow>
 *   {({ isLoading, productName, redirectToCheckout, error, price, currency, inStock, preOrderAvailable }) => (
 *     <div>
 *       <h2>{productName}</h2>
 *       <p>{price} {currency}</p>
 *       {error && <div className="error">{error}</div>}
 *       {inStock && <div>In stock</div>}
 *       {preOrderAvailable && <div>Pre-order available</div>}
 *       <button
 *         onClick={redirectToCheckout}
 *         disabled={isLoading}
 *       >
 *         {isLoading ? 'Processing...' : 'Buy Now'}
 *       </button>
 *     </div>
 *   )}
 * </BuyNow>
 * ```
 * @component
 */
export function BuyNow(props) {
    const { redirectToCheckout, loadingSignal, productName, errorSignal, price, currency, inStockSignal, preOrderAvailableSignal, } = useService(BuyNowServiceDefinition);
    return props.children({
        isLoading: loadingSignal.get(),
        error: errorSignal.get(),
        productName: productName,
        redirectToCheckout,
        price,
        currency,
        inStock: inStockSignal.get(),
        preOrderAvailable: preOrderAvailableSignal.get(),
    });
}
