// @wix/headless-stores/cjs/dist/react/ProductsList.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from "@wix/services-manager-react";
import { createServicesMap } from "@wix/services-manager";
import { ProductListService, ProductsListServiceDefinition, } from "../services/products-list-service.js";
import { ProductService, ProductServiceDefinition, } from "@wix/headless-stores/services";
/**
 * Root component that provides the ProductsList service context to its children.
 * This component sets up the necessary services for managing products list state.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { ProductsList } from '@wix/stores/components';
 *
 * function ProductsListPage() {
 *   return (
 *     <ProductsList.Root
 *       productsListConfig={{
 *         collectionId: 'my-collection-id',
 *         filters: { price: { min: 10, max: 100 } }
 *       }}
 *     >
 *       <ProductsList.Grid>
 *         {({ products, isLoading, error }) => (
 *           <div>
 *             {isLoading && <div>Loading products...</div>}
 *             {error && <div>Error: {error}</div>}
 *             {products.map(product => (
 *               <div key={product.id}>{product.name}</div>
 *             ))}
 *           </div>
 *         )}
 *       </ProductsList.Grid>
 *     </ProductsList.Root>
 *   );
 * }
 * ```
 */
export function Root(props) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(ProductsListServiceDefinition, ProductListService, props.productsListConfig), children: props.children }));
}
/**
 * Component that renders content when the products list is empty.
 * Only displays its children when there are no products, no loading state, and no errors.
 *
 * @component
 * @example
 * ```tsx
 * import { ProductsList } from '@wix/stores/components';
 *
 * function EmptyProductsMessage() {
 *   return (
 *     <ProductsList.EmptyState>
 *       {() => (
 *         <div className="empty-state">
 *           <h3>No products found</h3>
 *           <p>Try adjusting your search or filter criteria</p>
 *           <button>Clear Filters</button>
 *         </div>
 *       )}
 *     </ProductsList.EmptyState>
 *   );
 * }
 * ```
 */
export function EmptyState(props) {
    const { isLoading, error, products } = useService(ProductsListServiceDefinition);
    const isLoadingValue = isLoading.get();
    const errorValue = error.get();
    const productsValue = products.get();
    if (!isLoadingValue && !errorValue && productsValue.length === 0) {
        return typeof props.children === "function"
            ? props.children({})
            : props.children;
    }
    return null;
}
/**
 * Component that renders content during loading state.
 * Only displays its children when the products list is currently loading.
 *
 * @component
 * @example
 * ```tsx
 * import { ProductsList } from '@wix/stores/components';
 *
 * function ProductsLoading() {
 *   return (
 *     <ProductsList.Loading>
 *       {() => (
 *         <div className="loading-spinner">
 *           <div>Loading products...</div>
 *           <div className="spinner"></div>
 *         </div>
 *       )}
 *     </ProductsList.Loading>
 *   );
 * }
 * ```
 */
export function Loading(props) {
    const { isLoading } = useService(ProductsListServiceDefinition);
    const isLoadingValue = isLoading.get();
    if (isLoadingValue) {
        return typeof props.children === "function"
            ? props.children({})
            : props.children;
    }
    return null;
}
;
/**
 * Component that renders content when there's an error loading products.
 * Only displays its children when an error has occurred.
 *
 * @component
 * @example
 * ```tsx
 * import { ProductsList } from '@wix/stores/components';
 *
 * function ProductsError() {
 *   return (
 *     <ProductsList.Error>
 *       {({ error }) => (
 *         <div className="error-state">
 *           <h3>Error loading products</h3>
 *           <p>{error}</p>
 *           <button onClick={() => window.location.reload()}>
 *             Try Again
 *           </button>
 *         </div>
 *       )}
 *     </ProductsList.Error>
 *   );
 * }
 * ```
 */
export function Error(props) {
    const { error } = useService(ProductsListServiceDefinition);
    const errorValue = error.get();
    if (errorValue) {
        return typeof props.children === "function"
            ? props.children({ error: errorValue })
            : props.children;
    }
    return null;
}
/**
 * Component that renders content for each product in the list.
 * Maps over all products and provides each product through a service context.
 * Only renders when products are successfully loaded (not loading, no error, and has products).
 *
 * @component
 * @example
 * ```tsx
 * import { ProductsList } from '@wix/stores/components';
 *
 * function ProductsGrid() {
 *   return (
 *     <ProductsList.ItemContent>
 *       {({ product }) => (
 *         <div className="product-card">
 *           <img src={product.media?.main?.image} alt={product.name} />
 *           <h3>{product.name}</h3>
 *           <p>{product.actualPriceRange?.minValue?.formattedAmount}</p>
 *           <button>View Details</button>
 *         </div>
 *       )}
 *     </ProductsList.ItemContent>
 *   );
 * }
 * ```
 */
export function ItemContent(props) {
    const { products, isLoading, error } = useService(ProductsListServiceDefinition);
    const productsValue = products.get();
    if (isLoading.get() || error.get() || productsValue.length === 0) {
        return null;
    }
    return productsValue.map((product) => (_jsx(WixServices, { servicesMap: createServicesMap().addService(ProductServiceDefinition, ProductService, { product }), children: typeof props.children === "function"
            ? props.children({ product })
            : props.children }, product._id)));
}
