// @wix/headless-stores/cjs/dist/react/Product.js
import { useService } from "@wix/services-manager-react";
import { ProductServiceDefinition } from "../services/product-service.js";
/**
 * Headless component for product name display
 *
 * @component
 * @example
 * ```tsx
 * import { Product } from '@wix/stores/components';
 *
 * function ProductHeader() {
 *   return (
 *     <Product.Name>
 *       {({ name }) => (
 *         <h1>{name}</h1>
 *       )}
 *     </Product.Name>
 *   );
 * }
 * ```
 */
export const Name = (props) => {
    const service = useService(ProductServiceDefinition);
    const product = service.product.get();
    const name = product.name;
    return props.children({
        name,
    });
};
/**
 * Headless component for product description display
 *
 * @component
 * @example
 * ```tsx
 * import { Product } from '@wix/stores/components';
 *
 * function ProductDescription() {
 *   return (
 *     <Product.Description>
 *       {({ plainDescription, description }) => (
 *         <div>
 *           {plainDescription && (
 *             <div
 *               dangerouslySetInnerHTML={{
 *                 __html: plainDescription,
 *               }}
 *             />
 *           )}
 *           {description && (
 *             <div>Rich content description available</div>
 *           )}
 *         </div>
 *       )}
 *     </Product.Description>
 *   );
 * }
 * ```
 */
export const Description = (props) => {
    const service = useService(ProductServiceDefinition);
    const product = service.product.get();
    const descriptionRichContent = product.description;
    const plainDescription = product.plainDescription;
    return props.children({
        description: descriptionRichContent,
        plainDescription: plainDescription,
    });
};
