// @wix/auto_sdk_ecom_pickup-locations/build/es/index.typings.mjs
// src/ecom-v1-pickup-location-pickup-locations.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/ecom-v1-pickup-location-pickup-locations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomV1PickupLocationsUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pickup-locations",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/pickup-locations",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_pickup-locations";
function createPickupLocation(payload) {
  function __createPickupLocation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pickupLocation.createdDate" },
          { path: "pickupLocation.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.CreatePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPickupLocation;
}
function getPickupLocation(payload) {
  function __getPickupLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "GET",
      methodFqn: "wix.ecom.v1.PickupLocations.GetPickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveWixEcomV1PickupLocationsUrl({
            protoPath: "/v1/pickup-locations/{pickupLocationId}",
            data: payload,
            host
          }),
          params: toURLSearchParams(payload)
        }
      ]
    };
    return metadata;
  }
  return __getPickupLocation;
}
function updatePickupLocation(payload) {
  function __updatePickupLocation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pickupLocation.createdDate" },
          { path: "pickupLocation.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "PATCH",
      methodFqn: "wix.ecom.v1.PickupLocations.UpdatePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocation.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePickupLocation;
}
function deletePickupLocation(payload) {
  function __deletePickupLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "DELETE",
      methodFqn: "wix.ecom.v1.PickupLocations.DeletePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deletePickupLocation;
}
function queryPickupLocation(payload) {
  function __queryPickupLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.QueryPickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocations.createdDate" },
            { path: "pickupLocations.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPickupLocation;
}
function addDeliveryRegion(payload) {
  function __addDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.AddDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocationId}/add-delivery-region",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addDeliveryRegion;
}
function removeDeliveryRegion(payload) {
  function __removeDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.RemoveDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/pickup-locations/{pickupLocationId}/remove-delivery-region",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocation.createdDate" },
            { path: "pickupLocation.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removeDeliveryRegion;
}
function bulkCreatePickupLocation(payload) {
  function __bulkCreatePickupLocation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pickupLocations.createdDate" },
          { path: "pickupLocations.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "POST",
      methodFqn: "wix.ecom.v1.PickupLocations.BulkCreatePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/bulk/pickup-locations/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocations.createdDate" },
            { path: "pickupLocations.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreatePickupLocation;
}
function bulkUpdatePickupLocation(payload) {
  function __bulkUpdatePickupLocation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pickupLocations.createdDate" },
          { path: "pickupLocations.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "PATCH",
      methodFqn: "wix.ecom.v1.PickupLocations.BulkUpdatePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/bulk/pickup-locations/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupLocations.createdDate" },
            { path: "pickupLocations.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdatePickupLocation;
}
function bulkDeletePickupLocation(payload) {
  function __bulkDeletePickupLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.pickup_location",
      method: "DELETE",
      methodFqn: "wix.ecom.v1.PickupLocations.BulkDeletePickupLocation",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomV1PickupLocationsUrl({
        protoPath: "/v1/bulk/pickup-locations/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeletePickupLocation;
}

// src/ecom-v1-pickup-location-pickup-locations.universal.ts
import { transformSDKAddressToRESTAddress } from "@wix/sdk-runtime/transformations/address";
import { transformRESTAddressToSDKAddress } from "@wix/sdk-runtime/transformations/address";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var ConditionType = /* @__PURE__ */ ((ConditionType2) => {
  ConditionType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  ConditionType2["BY_TOTAL_WEIGHT"] = "BY_TOTAL_WEIGHT";
  ConditionType2["BY_TOTAL_PRICE"] = "BY_TOTAL_PRICE";
  ConditionType2["BY_TOTAL_QUANTITY"] = "BY_TOTAL_QUANTITY";
  return ConditionType2;
})(ConditionType || {});
var LogicalOperator = /* @__PURE__ */ ((LogicalOperator2) => {
  LogicalOperator2["UNKNOWN_LOGICAL_OPERATOR_TYPE"] = "UNKNOWN_LOGICAL_OPERATOR_TYPE";
  LogicalOperator2["EQ"] = "EQ";
  LogicalOperator2["GT"] = "GT";
  LogicalOperator2["GTE"] = "GTE";
  LogicalOperator2["LT"] = "LT";
  LogicalOperator2["LTE"] = "LTE";
  return LogicalOperator2;
})(LogicalOperator || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createPickupLocation2(pickupLocation) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ pickupLocation }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "pickupLocation.address" }]
      }
    ]
  );
  const reqOpts = createPickupLocation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [{ path: "pickupLocation.address" }]
        }
      ])
    )?.pickupLocation;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { pickupLocation: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["pickupLocation"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPickupLocation2(pickupLocationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    pickupLocationId
  });
  const reqOpts = getPickupLocation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [{ path: "pickupLocation.address" }]
        }
      ])
    )?.pickupLocation;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { pickupLocationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["pickupLocationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePickupLocation2(_id, pickupLocation) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      pickupLocation: { ...pickupLocation, id: _id }
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "pickupLocation.address" }]
      }
    ]
  );
  const reqOpts = updatePickupLocation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [{ path: "pickupLocation.address" }]
        }
      ])
    )?.pickupLocation;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { pickupLocation: "$[1]" },
        explicitPathsToArguments: { "pickupLocation.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "pickupLocation"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deletePickupLocation2(pickupLocationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    pickupLocationId
  });
  const reqOpts = deletePickupLocation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { pickupLocationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["pickupLocationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPickupLocation2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryPickupLocation(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [
          {
            transformFn: transformRESTAddressToSDKAddress,
            paths: [{ path: "pickupLocations.address" }]
          }
        ])
      );
      return {
        items: transformedData?.pickupLocations,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function addDeliveryRegion2(pickupLocationId, deliveryRegionId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    pickupLocationId,
    deliveryRegionId,
    revision: options?.revision
  });
  const reqOpts = addDeliveryRegion(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [{ path: "pickupLocation.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          pickupLocationId: "$[0]",
          deliveryRegionId: "$[1]",
          revision: "$[2].revision"
        },
        singleArgumentUnchanged: false
      },
      ["pickupLocationId", "deliveryRegionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function removeDeliveryRegion2(pickupLocationId, deliveryRegionId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    pickupLocationId,
    deliveryRegionId,
    revision: options?.revision
  });
  const reqOpts = removeDeliveryRegion(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [{ path: "pickupLocation.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          pickupLocationId: "$[0]",
          deliveryRegionId: "$[1]",
          revision: "$[2].revision"
        },
        singleArgumentUnchanged: false
      },
      ["pickupLocationId", "deliveryRegionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreatePickupLocation2(pickupLocations) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ pickupLocations }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "pickupLocations.address" }]
      }
    ]
  );
  const reqOpts = bulkCreatePickupLocation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [{ path: "pickupLocations.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { pickupLocations: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["pickupLocations"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdatePickupLocation2(pickupLocations) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ pickupLocations }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "pickupLocations.address" }]
      }
    ]
  );
  const reqOpts = bulkUpdatePickupLocation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [{ path: "pickupLocations.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { pickupLocations: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["pickupLocations"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeletePickupLocation2(pickupLocationIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    pickupLocationIds
  });
  const reqOpts = bulkDeletePickupLocation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { pickupLocationIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["pickupLocationIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  ConditionType,
  LogicalOperator,
  SortOrder,
  WebhookIdentityType,
  addDeliveryRegion2 as addDeliveryRegion,
  bulkCreatePickupLocation2 as bulkCreatePickupLocation,
  bulkDeletePickupLocation2 as bulkDeletePickupLocation,
  bulkUpdatePickupLocation2 as bulkUpdatePickupLocation,
  createPickupLocation2 as createPickupLocation,
  deletePickupLocation2 as deletePickupLocation,
  getPickupLocation2 as getPickupLocation,
  queryPickupLocation2 as queryPickupLocation,
  removeDeliveryRegion2 as removeDeliveryRegion,
  updatePickupLocation2 as updatePickupLocation
};
