// @wix/auto_sdk_ecom_shipping-rates/build/internal/es/index.mjs
// src/service-plugins-types.ts
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var WeightUnit = /* @__PURE__ */ ((WeightUnit2) => {
  WeightUnit2["UNSPECIFIED_WEIGHT_UNIT"] = "UNSPECIFIED_WEIGHT_UNIT";
  WeightUnit2["KG"] = "KG";
  WeightUnit2["LB"] = "LB";
  return WeightUnit2;
})(WeightUnit || {});
var PickupMethod = /* @__PURE__ */ ((PickupMethod2) => {
  PickupMethod2["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
  PickupMethod2["STORE_PICKUP"] = "STORE_PICKUP";
  PickupMethod2["PICKUP_POINT"] = "PICKUP_POINT";
  return PickupMethod2;
})(PickupMethod || {});
var ChargeType = /* @__PURE__ */ ((ChargeType2) => {
  ChargeType2["HANDLING_FEE"] = "HANDLING_FEE";
  return ChargeType2;
})(ChargeType || {});
var RuleType = /* @__PURE__ */ ((RuleType2) => {
  RuleType2["VALIDATION"] = "VALIDATION";
  RuleType2["OTHER"] = "OTHER";
  RuleType2["MAX"] = "MAX";
  RuleType2["MIN"] = "MIN";
  RuleType2["MAX_LENGTH"] = "MAX_LENGTH";
  RuleType2["MIN_LENGTH"] = "MIN_LENGTH";
  RuleType2["MAX_SIZE"] = "MAX_SIZE";
  RuleType2["MIN_SIZE"] = "MIN_SIZE";
  RuleType2["FORMAT"] = "FORMAT";
  RuleType2["DECIMAL_LTE"] = "DECIMAL_LTE";
  RuleType2["DECIMAL_GTE"] = "DECIMAL_GTE";
  RuleType2["DECIMAL_LT"] = "DECIMAL_LT";
  RuleType2["DECIMAL_GT"] = "DECIMAL_GT";
  RuleType2["DECIMAL_MAX_SCALE"] = "DECIMAL_MAX_SCALE";
  RuleType2["INVALID_ENUM_VALUE"] = "INVALID_ENUM_VALUE";
  RuleType2["REQUIRED_FIELD"] = "REQUIRED_FIELD";
  RuleType2["FIELD_NOT_ALLOWED"] = "FIELD_NOT_ALLOWED";
  RuleType2["ONE_OF_ALIGNMENT"] = "ONE_OF_ALIGNMENT";
  RuleType2["EXACT_LENGTH"] = "EXACT_LENGTH";
  RuleType2["EXACT_SIZE"] = "EXACT_SIZE";
  RuleType2["REQUIRED_ONE_OF_FIELD"] = "REQUIRED_ONE_OF_FIELD";
  return RuleType2;
})(RuleType || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});

// src/interfaces-ecom-v1-shipping-rates-provider.public.ts
import { ServicePluginDefinition } from "@wix/sdk-types";
import { transformSDKAddressToRESTAddress } from "@wix/sdk-runtime/transformations/address";
import { transformRESTAddressToSDKAddress } from "@wix/sdk-runtime/transformations/address";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";
var provideHandlers = ServicePluginDefinition("ECOM_SHIPPING_RATES", [
  {
    name: "getShippingRates",
    primaryHttpMappingPath: "/v1/getRates",
    transformations: {
      toREST: (payload) => {
        const toRestResponse = transformPaths(payload, [
          {
            transformFn: transformSDKAddressToRESTAddress,
            paths: [{ path: "shippingRates.logistics.pickupDetails.address" }]
          },
          {
            transformFn: transformSDKTimestampToRESTTimestamp,
            paths: [
              { path: "shippingRates.logistics.deliveryTimeSlot.from" },
              { path: "shippingRates.logistics.deliveryTimeSlot.to" }
            ]
          }
        ]);
        return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
      },
      fromREST: (payload) => {
        const fromRestRequest = transformPaths(payload, [
          {
            transformFn: transformRESTAddressToSDKAddress,
            paths: [
              { path: "request.shippingDestination" },
              { path: "request.shippingOrigin" }
            ]
          },
          {
            transformFn: transformRESTFloatToSDKFloat,
            paths: [{ path: "request.lineItems.physicalProperties.weight" }]
          }
        ]);
        return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
      }
    }
  }
]);

// src/interfaces-ecom-v1-shipping-rates-provider.context.ts
import { createServicePluginModule } from "@wix/sdk-runtime/service-plugin-modules";
var provideHandlers2 = createServicePluginModule(
  provideHandlers
);

// src/service-plugins-error-classes.ts
var MissingPostalCodeWixError = class extends Error {
  constructor(data = {}) {
    super("MissingPostalCode");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "MISSING_POSTAL_CODE";
    this.name = "MissingPostalCode";
    this.errorSchemaName = "com.wix.ecom.shipping_rates.v1.MissingPostalCodeError";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "MissingPostalCode",
      applicationCode: "MISSING_POSTAL_CODE"
    };
    this.data = data;
  }
};
/** @hidden */
MissingPostalCodeWixError.__type = "wix_spi_error";
var InvalidPostalCodeWixError = class extends Error {
  constructor(data = {}) {
    super("InvalidPostalCode");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "INVALID_POSTAL_CODE";
    this.name = "InvalidPostalCode";
    this.errorSchemaName = "com.wix.ecom.shipping_rates.v1.InvalidPostalCodeError";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "InvalidPostalCode",
      applicationCode: "INVALID_POSTAL_CODE"
    };
    this.data = data;
  }
};
/** @hidden */
InvalidPostalCodeWixError.__type = "wix_spi_error";
var InvalidAddressWixError = class extends Error {
  constructor(data = {}) {
    super("InvalidAddress");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "PARTIAL_ADDRESS";
    this.name = "InvalidAddress";
    this.errorSchemaName = "com.wix.ecom.shipping_rates.v1.InvalidAddressError";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "InvalidAddress",
      applicationCode: "PARTIAL_ADDRESS"
    };
    this.data = data;
  }
};
/** @hidden */
InvalidAddressWixError.__type = "wix_spi_error";
var GenericShippingRatesWixError = class extends Error {
  constructor(data = {}) {
    super("GenericShippingRatesError");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "GENERIC_ERROR";
    this.name = "GenericShippingRatesError";
    this.errorSchemaName = "com.wix.ecom.shipping_rates.v1.GenericShippingRatesError";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "GenericShippingRatesError",
      applicationCode: "GENERIC_ERROR"
    };
    this.data = data;
  }
};
/** @hidden */
GenericShippingRatesWixError.__type = "wix_spi_error";
export {
  ChargeType,
  GenericShippingRatesWixError,
  IdentityType,
  InvalidAddressWixError,
  InvalidPostalCodeWixError,
  MissingPostalCodeWixError,
  PickupMethod,
  RuleType,
  SubdivisionType,
  VatType,
  WeightUnit,
  provideHandlers2 as provideHandlers,
  provideHandlers as publicProvideHandlers
};
