// @wix/auto_sdk_ecom_discount-rules/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  createDiscountRule: () => createDiscountRule2,
  deleteDiscountRule: () => deleteDiscountRule2,
  getDiscountRule: () => getDiscountRule2,
  queryDiscountRules: () => queryDiscountRules2,
  updateDiscountRule: () => updateDiscountRule2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-discounts-v1-discount-rule-discount-rules.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomDiscountsDiscountRuleServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/discount-rules-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/discount-rules",
        destPath: "/v1/discount-rules"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/discount-rules",
        destPath: "/v1/discount-rules"
      },
      {
        srcPath: "/v1/discount-rules",
        destPath: "/v1/discount-rules"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/discounts/v1/discount-rules",
        destPath: "/v1/discount-rules"
      }
    ],
    _: [
      {
        srcPath: "/ecom/discounts/v1/discount-rules",
        destPath: "/v1/discount-rules"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_discount-rules";
function createDiscountRule(payload) {
  function __createDiscountRule({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "discountRule.createdDate" },
          { path: "discountRule.updatedDate" },
          { path: "discountRule.activeTimeInfo.start" },
          { path: "discountRule.activeTimeInfo.end" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "discountRule.discounts.values.percentage" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "POST",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.CreateDiscountRule",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "discountRule.createdDate" },
            { path: "discountRule.updatedDate" },
            { path: "discountRule.activeTimeInfo.start" },
            { path: "discountRule.activeTimeInfo.end" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "discountRule.discounts.values.percentage" }]
        }
      ])
    };
    return metadata;
  }
  return __createDiscountRule;
}
function getDiscountRule(payload) {
  function __getDiscountRule({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "GET",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.GetDiscountRule",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules/{discountRuleId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "discountRule.createdDate" },
            { path: "discountRule.updatedDate" },
            { path: "discountRule.activeTimeInfo.start" },
            { path: "discountRule.activeTimeInfo.end" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "discountRule.discounts.values.percentage" }]
        }
      ])
    };
    return metadata;
  }
  return __getDiscountRule;
}
function updateDiscountRule(payload) {
  function __updateDiscountRule({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "discountRule.createdDate" },
          { path: "discountRule.updatedDate" },
          { path: "discountRule.activeTimeInfo.start" },
          { path: "discountRule.activeTimeInfo.end" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "discountRule.discounts.values.percentage" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "PATCH",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.UpdateDiscountRule",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules/{discountRule.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "discountRule.createdDate" },
            { path: "discountRule.updatedDate" },
            { path: "discountRule.activeTimeInfo.start" },
            { path: "discountRule.activeTimeInfo.end" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "discountRule.discounts.values.percentage" }]
        }
      ])
    };
    return metadata;
  }
  return __updateDiscountRule;
}
function deleteDiscountRule(payload) {
  function __deleteDiscountRule({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "DELETE",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.DeleteDiscountRule",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules/{discountRuleId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteDiscountRule;
}
function queryDiscountRules(payload) {
  function __queryDiscountRules({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "POST",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.QueryDiscountRules",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "discountRules.createdDate" },
            { path: "discountRules.updatedDate" },
            { path: "discountRules.activeTimeInfo.start" },
            { path: "discountRules.activeTimeInfo.end" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "discountRules.discounts.values.percentage" }]
        }
      ])
    };
    return metadata;
  }
  return __queryDiscountRules;
}

// src/ecom-discounts-v1-discount-rule-discount-rules.meta.ts
function createDiscountRule2() {
  const payload = {};
  const getRequestOptions = createDiscountRule(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/discount-rules",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDiscountRule2() {
  const payload = { discountRuleId: ":discountRuleId" };
  const getRequestOptions = getDiscountRule(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/discount-rules/{discountRuleId}",
    pathParams: { discountRuleId: "discountRuleId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateDiscountRule2() {
  const payload = { discountRule: { id: ":discountRuleId" } };
  const getRequestOptions = updateDiscountRule(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/discount-rules/{discountRule.id}",
    pathParams: { discountRuleId: "discountRuleId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteDiscountRule2() {
  const payload = { discountRuleId: ":discountRuleId" };
  const getRequestOptions = deleteDiscountRule(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/discount-rules/{discountRuleId}",
    pathParams: { discountRuleId: "discountRuleId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDiscountRules2() {
  const payload = {};
  const getRequestOptions = queryDiscountRules(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/discount-rules/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createDiscountRule,
  deleteDiscountRule,
  getDiscountRule,
  queryDiscountRules,
  updateDiscountRule
});
