// @wix/error-handler/dist/cjs/public-testkit/ErrorHandlerTestkit.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ErrorHandlerTestkit = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _i18n = require("@wix/fe-essentials/i18n");
var _ErrorHandlerTestkitBase = require("./ErrorHandlerTestkitBase");
class ErrorHandlerTestkit {
  constructor(params) {
    (0, _defineProperty2.default)(this, "base", void 0);
    (0, _defineProperty2.default)(this, "showError", props => {
      this.base.showError(props);
    });
    (0, _defineProperty2.default)(this, "getCommonErrorMessageByStatus", async status => {
      return this.base.getCommonErrorMessageByStatus(status);
    });
    this.base = new _ErrorHandlerTestkitBase.ErrorHandlerTestkitBase({
      ...params,
      createI18n: i18nOptions => (0, _i18n.initI18n)({
        ...i18nOptions,
        locale: params.language
      })
    });
  }
  get showErrorMock() {
    return this.base.showErrorMock;
  }
  async getClientErrorMessages() {
    return this.base.getClientErrorMessages();
  }
}
exports.ErrorHandlerTestkit = ErrorHandlerTestkit;
