// @wix/error-handler/dist/cjs/public-testkit/initI18n.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.initI18n = void 0;
var _i18next = _interopRequireDefault(require("i18next"));
var _i18nextIcu = _interopRequireDefault(require("i18next-icu"));
const initI18n = ({
  locale,
  asyncMessagesLoader
}) => {
  const i18nInstance = _i18next.default.createInstance({
    lng: locale || 'en',
    fallbackLng: 'en',
    keySeparator: false
  }).use(_i18nextIcu.default);
  if (asyncMessagesLoader) {
    i18nInstance.use(customBackend({
      asyncMessagesLoader
    }));
  }
  return i18nInstance;
};
exports.initI18n = initI18n;
function customBackend({
  asyncMessagesLoader
}) {
  return {
    type: 'backend',
    init: () => {},
    read: async (language, _namespace, callback) => {
      try {
        const translation = await asyncMessagesLoader(language);
        return callback(null, translation);
      } catch (err) {
        return callback(err, null);
      }
    }
  };
}
