// @wix/auto_sdk_ecom_delivery-solutions/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getDeliverySolutions: () => getDeliverySolutions2
});
module.exports = __toCommonJS(meta_exports);

// src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.http.ts
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomDeliveryprofileV1DeliveryRatesUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/delivery-rates",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_delivery-solutions";
function getDeliverySolutions(payload) {
  function __getDeliverySolutions({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "lineItems.physicalProperties.weight" },
          { path: "destination.geocode.latitude" },
          { path: "destination.geocode.longitude" },
          { path: "origin.geocode.latitude" },
          { path: "origin.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.gateways.ecom.v1.delivery_rates_gateway",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryRates.GetDeliverySolutions",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryRatesUrl({
        protoPath: "/v1/get-delivery-solutions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "deliverySolutions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "deliverySolutions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        },
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliverySolutions.logistics.deliveryTimeSlot.from" },
            { path: "deliverySolutions.logistics.deliveryTimeSlot.to" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDeliverySolutions;
}

// src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.meta.ts
function getDeliverySolutions2() {
  const payload = {};
  const getRequestOptions = getDeliverySolutions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/get-delivery-solutions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getDeliverySolutions
});
