// @wix/auto_sdk_ecom_checkout-content/build/es/meta.mjs
// src/ecom-api-v1-checkout-content-provider-checkout-content.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomCheckoutContentSpiHostV1CheckoutContentSpiHostServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_checkout-content";
function getCheckoutContent(payload) {
  function __getCheckoutContent({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.api.v1.checkout_content_provider",
      method: "GET",
      methodFqn: "wix.ecom.checkout_content_spi_host.v1.CheckoutContentSPIHostService.GetCheckoutContent",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomCheckoutContentSpiHostV1CheckoutContentSpiHostServiceUrl(
        {
          protoPath: "/v1/checkout-content/{appId}/{componentId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCheckoutContent;
}

// src/ecom-api-v1-checkout-content-provider-checkout-content.meta.ts
function getCheckoutContent2() {
  const payload = { appId: ":appId", componentId: ":componentId" };
  const getRequestOptions = getCheckoutContent(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/checkout-content/{appId}/{componentId}",
    pathParams: { appId: "appId", componentId: "componentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getCheckoutContent2 as getCheckoutContent
};
