// @wix/error-handler/dist/cjs/testkit/createMockSdkError.js
"use strict";

exports.__esModule = true;
exports.createMockSdkApplicationError = createMockSdkApplicationError;
exports.createMockSdkError = createMockSdkError;
exports.createMockSdkValidationError = createMockSdkValidationError;
function createMockSdkError(response, overrides) {
  return {
    status: response.status,
    body: {
      message: response.message ?? 'MockSdkError',
      ...(overrides == null ? void 0 : overrides.body)
    },
    headers: {
      'x-wix-request-id': response.requestId,
      ...(overrides == null ? void 0 : overrides.headers)
    },
    ...overrides
  };
}
function createMockSdkApplicationError(applicationError, response, overrides) {
  var _overrides$body;
  return {
    status: response.status,
    body: {
      message: response.message ?? 'MockSdkApplicationError',
      details: {
        applicationError,
        ...(overrides == null || (_overrides$body = overrides.body) == null ? void 0 : _overrides$body.details)
      }
    },
    headers: {
      'x-wix-request-id': response.requestId,
      ...(overrides == null ? void 0 : overrides.headers)
    },
    ...overrides
  };
}
function createMockSdkValidationError(validationError, response, overrides) {
  var _overrides$body2;
  return {
    status: response.status,
    body: {
      message: response.message ?? 'MockSdkValidationError',
      details: {
        validationError,
        ...(overrides == null || (_overrides$body2 = overrides.body) == null ? void 0 : _overrides$body2.details)
      }
    },
    headers: {
      'x-wix-request-id': response.requestId,
      ...(overrides == null ? void 0 : overrides.headers)
    },
    ...overrides
  };
}
