// @wix/error-handler/dist/cjs/testkit/SdkHttpClient.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ErrorHandlerSdkHttpClient = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
class ErrorHandlerSdkHttpClient {
  constructor(_httpClient, _errorHandler) {
    (0, _defineProperty2.default)(this, "httpClient", void 0);
    (0, _defineProperty2.default)(this, "errorHandler", void 0);
    (0, _defineProperty2.default)(this, "fetchWithAuth", async (input, init) => {
      const {
        httpClient,
        errorHandler
      } = this;
      try {
        const response = await httpClient.fetchWithAuth(input, init);
        errorHandler.handleError(response, {
          requestOptions: {
            url: typeof input === 'string' || input instanceof URL ? input.toString() : input.url,
            method: init == null ? void 0 : init.method
          }
        });
        return response;
      } catch (e) {
        errorHandler.handleError(e);
        throw e;
      }
    });
    this.httpClient = _httpClient;
    this.errorHandler = _errorHandler;
  }
}
exports.ErrorHandlerSdkHttpClient = ErrorHandlerSdkHttpClient;
