// @wix/auto_sdk_ecom_tax-groups/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  createTaxGroup: () => createTaxGroup2,
  deleteTaxGroup: () => deleteTaxGroup2,
  getTaxGroup: () => getTaxGroup2,
  listDefaultTaxGroups: () => listDefaultTaxGroups2,
  listDefaultTaxGroupsByAppIds: () => listDefaultTaxGroupsByAppIds2,
  queryTaxGroups: () => queryTaxGroups2,
  updateTaxGroup: () => updateTaxGroup2
});
module.exports = __toCommonJS(meta_exports);

// src/billing-v1-tax-group-tax-groups.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBillingV1TaxGroupsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/billing/v1/tax-groups",
        destPath: "/v1/tax-groups"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/billing/v1/tax-groups",
        destPath: "/v1/tax-groups"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_tax-groups";
function createTaxGroup(payload) {
  function __createTaxGroup({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "taxGroup.createdDate" },
          { path: "taxGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "POST",
      methodFqn: "wix.billing.v1.TaxGroups.CreateTaxGroup",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroup.createdDate" },
            { path: "taxGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createTaxGroup;
}
function getTaxGroup(payload) {
  function __getTaxGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "GET",
      methodFqn: "wix.billing.v1.TaxGroups.GetTaxGroup",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/{taxGroupId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroup.createdDate" },
            { path: "taxGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTaxGroup;
}
function updateTaxGroup(payload) {
  function __updateTaxGroup({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "taxGroup.createdDate" },
          { path: "taxGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "PATCH",
      methodFqn: "wix.billing.v1.TaxGroups.UpdateTaxGroup",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/{taxGroup.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroup.createdDate" },
            { path: "taxGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateTaxGroup;
}
function deleteTaxGroup(payload) {
  function __deleteTaxGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "DELETE",
      methodFqn: "wix.billing.v1.TaxGroups.DeleteTaxGroup",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/{taxGroupId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteTaxGroup;
}
function queryTaxGroups(payload) {
  function __queryTaxGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "POST",
      methodFqn: "wix.billing.v1.TaxGroups.QueryTaxGroups",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroups.createdDate" },
            { path: "taxGroups.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTaxGroups;
}
function listDefaultTaxGroups(payload) {
  function __listDefaultTaxGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "GET",
      methodFqn: "wix.billing.v1.TaxGroups.ListDefaultTaxGroups",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/default-tax-groups",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "taxGroups.createdDate" },
            { path: "taxGroups.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDefaultTaxGroups;
}
function listDefaultTaxGroupsByAppIds(payload) {
  function __listDefaultTaxGroupsByAppIds({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.tax_group",
      method: "POST",
      methodFqn: "wix.billing.v1.TaxGroups.ListDefaultTaxGroupsByAppIds",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxGroupsUrl({
        protoPath: "/v1/tax-groups/list-default-tax-groups-by-app-ids",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.taxGroup.createdDate" },
            { path: "results.taxGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDefaultTaxGroupsByAppIds;
}

// src/billing-v1-tax-group-tax-groups.meta.ts
function createTaxGroup2() {
  const payload = {};
  const getRequestOptions = createTaxGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tax-groups",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getTaxGroup2() {
  const payload = { taxGroupId: ":taxGroupId" };
  const getRequestOptions = getTaxGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/tax-groups/{taxGroupId}",
    pathParams: { taxGroupId: "taxGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateTaxGroup2() {
  const payload = { taxGroup: { id: ":taxGroupId" } };
  const getRequestOptions = updateTaxGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/tax-groups/{taxGroup.id}",
    pathParams: { taxGroupId: "taxGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteTaxGroup2() {
  const payload = { taxGroupId: ":taxGroupId" };
  const getRequestOptions = deleteTaxGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/tax-groups/{taxGroupId}",
    pathParams: { taxGroupId: "taxGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryTaxGroups2() {
  const payload = {};
  const getRequestOptions = queryTaxGroups(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tax-groups/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDefaultTaxGroups2() {
  const payload = {};
  const getRequestOptions = listDefaultTaxGroups(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/tax-groups/default-tax-groups",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDefaultTaxGroupsByAppIds2() {
  const payload = {};
  const getRequestOptions = listDefaultTaxGroupsByAppIds(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tax-groups/list-default-tax-groups-by-app-ids",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createTaxGroup,
  deleteTaxGroup,
  getTaxGroup,
  listDefaultTaxGroups,
  listDefaultTaxGroupsByAppIds,
  queryTaxGroups,
  updateTaxGroup
});
