// @wix/auto_sdk_ecom_delivery-solutions/build/es/meta.mjs
// src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.http.ts
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomDeliveryprofileV1DeliveryRatesUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/delivery-rates",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/delivery-rates",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_delivery-solutions";
function getDeliverySolutions(payload) {
  function __getDeliverySolutions({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "lineItems.physicalProperties.weight" },
          { path: "destination.geocode.latitude" },
          { path: "destination.geocode.longitude" },
          { path: "origin.geocode.latitude" },
          { path: "origin.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.gateways.ecom.v1.delivery_rates_gateway",
      method: "POST",
      methodFqn: "wix.ecom.deliveryprofile.v1.DeliveryRates.GetDeliverySolutions",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryprofileV1DeliveryRatesUrl({
        protoPath: "/v1/get-delivery-solutions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "deliverySolutions.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "deliverySolutions.logistics.pickupDetails.address.geocode.longitude"
            }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "deliverySolutions.logistics.deliveryTimeSlot.from" },
            { path: "deliverySolutions.logistics.deliveryTimeSlot.to" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDeliverySolutions;
}

// src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.meta.ts
function getDeliverySolutions2() {
  const payload = {};
  const getRequestOptions = getDeliverySolutions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/get-delivery-solutions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getDeliverySolutions2 as getDeliverySolutions
};
