// @wix/auto_sdk_ecom_order-billing/build/internal/es/meta.mjs
// src/ecom-v1-order-billing-order-billing.http.ts
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixEcomOrdersBillingV1OrderBillingServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/order-billing/v1",
        destPath: "/v1/order-billing"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/order-billing",
        destPath: "/v1/order-billing"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/order-billing",
        destPath: "/v1/order-billing"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/order-billing",
        destPath: "/v1/order-billing"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_order-billing";
function getOrderRefundability(payload) {
  function __getOrderRefundability({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_billing",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.billing.v1.OrderBillingService.GetOrderRefundability",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersBillingV1OrderBillingServiceUrl({
        protoPath: "/v1/order-billing/get-order-refundability",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getOrderRefundability;
}
function calculateRefund(payload) {
  function __calculateRefund({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_billing",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.billing.v1.OrderBillingService.CalculateRefund",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersBillingV1OrderBillingServiceUrl({
        protoPath: "/v1/order-billing/calculate-refund",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __calculateRefund;
}
function refundPayments(payload) {
  function __refundPayments({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_billing",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.billing.v1.OrderBillingService.RefundPayments",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersBillingV1OrderBillingServiceUrl({
        protoPath: "/v1/order-billing/refund-payments",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refund.createdDate" },
            { path: "orderTransactions.payments.createdDate" },
            { path: "orderTransactions.payments.updatedDate" },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.updatedDate"
            },
            { path: "orderTransactions.refunds.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __refundPayments;
}
function authorizeChargeWithSavedPaymentMethod(payload) {
  function __authorizeChargeWithSavedPaymentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_billing",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.billing.v1.OrderBillingService.AuthorizeChargeWithSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersBillingV1OrderBillingServiceUrl({
        protoPath: "/v1/order-billing/authorize-charge-with-saved-payment-method",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payment.createdDate" },
            { path: "payment.updatedDate" },
            {
              path: "payment.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "payment.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "payment.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "payment.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            { path: "payment.regularPaymentDetails.chargebacks.createdDate" },
            { path: "payment.regularPaymentDetails.chargebacks.updatedDate" },
            { path: "orderTransactions.payments.createdDate" },
            { path: "orderTransactions.payments.updatedDate" },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.updatedDate"
            },
            { path: "orderTransactions.refunds.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __authorizeChargeWithSavedPaymentMethod;
}
function voidAuthorizedPayments(payload) {
  function __voidAuthorizedPayments({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_billing",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.billing.v1.OrderBillingService.VoidAuthorizedPayments",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersBillingV1OrderBillingServiceUrl({
        protoPath: "/v1/order-billing/void-authorized-payments",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderTransactions.payments.createdDate" },
            { path: "orderTransactions.payments.updatedDate" },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.updatedDate"
            },
            { path: "orderTransactions.refunds.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __voidAuthorizedPayments;
}
function captureAuthorizedPayments(payload) {
  function __captureAuthorizedPayments({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_billing",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.billing.v1.OrderBillingService.CaptureAuthorizedPayments",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersBillingV1OrderBillingServiceUrl({
        protoPath: "/v1/order-billing/capture-authorized-payments",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderTransactions.payments.createdDate" },
            { path: "orderTransactions.payments.updatedDate" },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.updatedDate"
            },
            { path: "orderTransactions.refunds.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __captureAuthorizedPayments;
}
function generateReceipts(payload) {
  function __generateReceipts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_billing",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.billing.v1.OrderBillingService.GenerateReceipts",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersBillingV1OrderBillingServiceUrl({
        protoPath: "/v1/order-billing/generate-receipts",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateReceipts;
}
function redeemGiftCard(payload) {
  function __redeemGiftCard({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_billing",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.billing.v1.OrderBillingService.RedeemGiftCard",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersBillingV1OrderBillingServiceUrl({
        protoPath: "/v1/order-billing/redeem-gift-card",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderTransactions.payments.createdDate" },
            { path: "orderTransactions.payments.updatedDate" },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.updatedDate"
            },
            { path: "orderTransactions.refunds.createdDate" },
            { path: "payment.createdDate" },
            { path: "payment.updatedDate" },
            {
              path: "payment.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "payment.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "payment.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "payment.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            { path: "payment.regularPaymentDetails.chargebacks.createdDate" },
            { path: "payment.regularPaymentDetails.chargebacks.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __redeemGiftCard;
}

// src/ecom-v1-order-billing-order-billing.meta.ts
function getOrderRefundability2() {
  const payload = {};
  const getRequestOptions = getOrderRefundability(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-billing/get-order-refundability",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function calculateRefund2() {
  const payload = {};
  const getRequestOptions = calculateRefund(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-billing/calculate-refund",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function refundPayments2() {
  const payload = {};
  const getRequestOptions = refundPayments(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-billing/refund-payments",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function authorizeChargeWithSavedPaymentMethod2() {
  const payload = {};
  const getRequestOptions = authorizeChargeWithSavedPaymentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-billing/authorize-charge-with-saved-payment-method",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function voidAuthorizedPayments2() {
  const payload = {};
  const getRequestOptions = voidAuthorizedPayments(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-billing/void-authorized-payments",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function captureAuthorizedPayments2() {
  const payload = {};
  const getRequestOptions = captureAuthorizedPayments(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-billing/capture-authorized-payments",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateReceipts2() {
  const payload = {};
  const getRequestOptions = generateReceipts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-billing/generate-receipts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function redeemGiftCard2() {
  const payload = {};
  const getRequestOptions = redeemGiftCard(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-billing/redeem-gift-card",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  authorizeChargeWithSavedPaymentMethod2 as authorizeChargeWithSavedPaymentMethod,
  calculateRefund2 as calculateRefund,
  captureAuthorizedPayments2 as captureAuthorizedPayments,
  generateReceipts2 as generateReceipts,
  getOrderRefundability2 as getOrderRefundability,
  redeemGiftCard2 as redeemGiftCard,
  refundPayments2 as refundPayments,
  voidAuthorizedPayments2 as voidAuthorizedPayments
};
