// @wix/auto_sdk_ecom_order-payment-requests/build/es/meta.mjs
// src/ecom-v1-order-payment-request-order-payment-requests.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/order-payment-request",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/ecom/v1/order-payment-requests",
        destPath: "/v1/order-payment-requests"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/order-payment-requests",
        destPath: "/v1/order-payment-requests"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/order-payment-requests",
        destPath: "/v1/order-payment-requests"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/order-payment-requests",
        destPath: "/v1/order-payment-requests"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_order-payment-requests";
function createOrderPaymentRequest(payload) {
  function __createOrderPaymentRequest({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "orderPaymentRequest.expirationDate" },
          { path: "orderPaymentRequest.createdDate" },
          { path: "orderPaymentRequest.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_payment_request",
      method: "POST",
      methodFqn: "wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.CreateOrderPaymentRequest",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(
        { protoPath: "/v1/order-payment-requests", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderPaymentRequest.expirationDate" },
            { path: "orderPaymentRequest.createdDate" },
            { path: "orderPaymentRequest.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOrderPaymentRequest;
}
function getOrderPaymentRequest(payload) {
  function __getOrderPaymentRequest({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_payment_request",
      method: "GET",
      methodFqn: "wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.GetOrderPaymentRequest",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(
        {
          protoPath: "/v1/order-payment-requests/{orderPaymentRequestId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderPaymentRequest.expirationDate" },
            { path: "orderPaymentRequest.createdDate" },
            { path: "orderPaymentRequest.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOrderPaymentRequest;
}
function updateOrderPaymentRequest(payload) {
  function __updateOrderPaymentRequest({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "orderPaymentRequest.expirationDate" },
          { path: "orderPaymentRequest.createdDate" },
          { path: "orderPaymentRequest.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_payment_request",
      method: "PATCH",
      methodFqn: "wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.UpdateOrderPaymentRequest",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(
        {
          protoPath: "/v1/order-payment-requests/{orderPaymentRequest.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderPaymentRequest.expirationDate" },
            { path: "orderPaymentRequest.createdDate" },
            { path: "orderPaymentRequest.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOrderPaymentRequest;
}
function deleteOrderPaymentRequest(payload) {
  function __deleteOrderPaymentRequest({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_payment_request",
      method: "DELETE",
      methodFqn: "wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.DeleteOrderPaymentRequest",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(
        {
          protoPath: "/v1/order-payment-requests/{orderPaymentRequestId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteOrderPaymentRequest;
}
function queryOrderPaymentRequests(payload) {
  function __queryOrderPaymentRequests({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_payment_request",
      method: "POST",
      methodFqn: "wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.QueryOrderPaymentRequests",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(
        { protoPath: "/v1/order-payment-requests/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderPaymentRequests.expirationDate" },
            { path: "orderPaymentRequests.createdDate" },
            { path: "orderPaymentRequests.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryOrderPaymentRequests;
}
function getOrderPaymentRequestUrl(payload) {
  function __getOrderPaymentRequestUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_payment_request",
      method: "POST",
      methodFqn: "wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.GetOrderPaymentRequestURL",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(
        { protoPath: "/v1/order-payment-requests/url", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __getOrderPaymentRequestUrl;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_payment_request",
      method: "POST",
      methodFqn: "wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(
        {
          protoPath: "/v1/order-payment-requests/{id}/update-extended-fields",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderPaymentRequest.expirationDate" },
            { path: "orderPaymentRequest.createdDate" },
            { path: "orderPaymentRequest.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function voidOrderPaymentRequest(payload) {
  function __voidOrderPaymentRequest({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order_payment_request",
      method: "POST",
      methodFqn: "wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.VoidOrderPaymentRequest",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(
        { protoPath: "/v1/order-payment-requests/void", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __voidOrderPaymentRequest;
}

// src/ecom-v1-order-payment-request-order-payment-requests.meta.ts
function createOrderPaymentRequest2() {
  const payload = {};
  const getRequestOptions = createOrderPaymentRequest(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-payment-requests",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getOrderPaymentRequest2() {
  const payload = { orderPaymentRequestId: ":orderPaymentRequestId" };
  const getRequestOptions = getOrderPaymentRequest(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/order-payment-requests/{orderPaymentRequestId}",
    pathParams: { orderPaymentRequestId: "orderPaymentRequestId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateOrderPaymentRequest2() {
  const payload = {
    orderPaymentRequest: { id: ":orderPaymentRequestId" }
  };
  const getRequestOptions = updateOrderPaymentRequest(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/order-payment-requests/{orderPaymentRequest.id}",
    pathParams: { orderPaymentRequestId: "orderPaymentRequestId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteOrderPaymentRequest2() {
  const payload = { orderPaymentRequestId: ":orderPaymentRequestId" };
  const getRequestOptions = deleteOrderPaymentRequest(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/order-payment-requests/{orderPaymentRequestId}",
    pathParams: { orderPaymentRequestId: "orderPaymentRequestId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryOrderPaymentRequests2() {
  const payload = {};
  const getRequestOptions = queryOrderPaymentRequests(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-payment-requests/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getOrderPaymentRequestUrl2() {
  const payload = {};
  const getRequestOptions = getOrderPaymentRequestUrl(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-payment-requests/url",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-payment-requests/{id}/update-extended-fields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function voidOrderPaymentRequest2() {
  const payload = {};
  const getRequestOptions = voidOrderPaymentRequest(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/order-payment-requests/void",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createOrderPaymentRequest2 as createOrderPaymentRequest,
  deleteOrderPaymentRequest2 as deleteOrderPaymentRequest,
  getOrderPaymentRequest2 as getOrderPaymentRequest,
  getOrderPaymentRequestUrl2 as getOrderPaymentRequestUrl,
  queryOrderPaymentRequests2 as queryOrderPaymentRequests,
  updateExtendedFields2 as updateExtendedFields,
  updateOrderPaymentRequest2 as updateOrderPaymentRequest,
  voidOrderPaymentRequest2 as voidOrderPaymentRequest
};
