// @wix/headless-stores/dist/react/SelectedVariant.js
import { useService } from "@wix/services-manager-react";
import { SelectedVariantServiceDefinition } from "../services/selected-variant-service.js";
/**
 * Headless component for selected variant details display
 *
 * @component
 * @example
 * ```tsx
 * import { SelectedVariant } from '@wix/stores/components';
 *
 * function VariantDetails() {
 *   return (
 *     <SelectedVariant.Details>
 *       {({ sku, weight }) => (
 *         <div>
 *           {sku && <div>SKU: {sku}</div>}
 *           {weight && <div>Weight: {weight}</div>}
 *         </div>
 *       )}
 *     </SelectedVariant.Details>
 *   );
 * }
 * ```
 */
export const Details = (props) => {
    const selectedVariantService = useService(SelectedVariantServiceDefinition);
    const selectedVariant = selectedVariantService.currentVariant?.get();
    let sku = selectedVariant?.sku || null;
    let weight = selectedVariant?.physicalProperties?.weight?.toString() || null;
    return props.children({
        sku,
        weight,
    });
};
/**
 * Headless component for product price display
 *
 * @component
 * @example
 * ```tsx
 * import { SelectedVariant } from '@wix/stores/components';
 *
 * function ProductPrice() {
 *   return (
 *     <SelectedVariant.Price>
 *       {({ price, compareAtPrice, currency }) => (
 *         <div className="price-display">
 *           <span className="current-price">{price}</span>
 *           {compareAtPrice && (
 *             <span className="compare-price">
 *               <s>{compareAtPrice}</s>
 *             </span>
 *           )}
 *           <span className="currency">{currency}</span>
 *         </div>
 *       )}
 *     </SelectedVariant.Price>
 *   );
 * }
 * ```
 */
export const Price = (props) => {
    const variantService = useService(SelectedVariantServiceDefinition);
    const price = variantService.currentPrice.get();
    const compareAtPrice = variantService.currentCompareAtPrice.get();
    const currency = variantService.currency.get();
    return props.children({
        price,
        compareAtPrice,
        currency,
    });
};
/**
 * Headless component for product SKU display
 *
 * @component
 * @example
 * ```tsx
 * import { SelectedVariant } from '@wix/stores/components';
 *
 * function ProductSKU() {
 *   return (
 *     <SelectedVariant.SKU>
 *       {({ sku }) => (
 *         <div>
 *           {sku && (
 *             <div className="product-sku">
 *               <strong>SKU:</strong> {sku}
 *             </div>
 *           )}
 *         </div>
 *       )}
 *     </SelectedVariant.SKU>
 *   );
 * }
 * ```
 */
export const SKU = (props) => {
    const selectedVariantService = useService(SelectedVariantServiceDefinition);
    const selectedVariant = selectedVariantService.currentVariant?.get();
    const sku = selectedVariant?.sku || null;
    return props.children({
        sku,
    });
};
