// @wix/auto_sdk_ecom_tippable-staff/build/es/meta.mjs
// src/tips-staff-v1-staff-tippable-staff.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixTipsStaffV1TipStaffServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/tips-staff",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_tippable-staff";
function listTippableStaff(payload) {
  function __listTippableStaff({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.staff.v1.staff",
      method: "GET",
      methodFqn: "wix.tips.staff.v1.TipStaffService.ListTippableStaff",
      packageName: PACKAGE_NAME,
      url: resolveWixTipsStaffV1TipStaffServiceUrl({
        protoPath: "/v1/eligible-staff",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listTippableStaff;
}

// src/tips-staff-v1-staff-tippable-staff.meta.ts
function listTippableStaff2() {
  const payload = {};
  const getRequestOptions = listTippableStaff(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/eligible-staff",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  listTippableStaff2 as listTippableStaff
};
