// @wix/auto_sdk_ecom_order-invoices/build/internal/es/meta.mjs
// src/ecom-orders-v1-invoice-order-invoices.http.ts
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixEcomOrdersInvoicesV1InvoicesUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/ep-invoices",
        destPath: "/v1/invoices"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/ep-invoices",
        destPath: "/v1/invoices"
      }
    ],
    _: [
      {
        srcPath: "/v1/invoices",
        destPath: "/v1/invoices"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_order-invoices";
function listInvoicesForMultipleOrders(payload) {
  function __listInvoicesForMultipleOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.orders.v1.invoice",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.invoices.v1.Invoices.ListInvoicesForMultipleOrders",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersInvoicesV1InvoicesUrl({
        protoPath: "/v1/invoices/list-by-ids",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listInvoicesForMultipleOrders;
}
function generateInvoice(payload) {
  function __generateInvoice({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.orders.v1.invoice",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.invoices.v1.Invoices.GenerateInvoice",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersInvoicesV1InvoicesUrl({
        protoPath: "/v1/invoices/orders/{orderId}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateInvoice;
}
function bulkGenerateInvoices(payload) {
  function __bulkGenerateInvoices({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.orders.v1.invoice",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.invoices.v1.Invoices.BulkGenerateInvoices",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersInvoicesV1InvoicesUrl({
        protoPath: "/v1/bulk/invoices/generate",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkGenerateInvoices;
}
function addInvoiceToOrder(payload) {
  function __addInvoiceToOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.orders.v1.invoice",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.invoices.v1.Invoices.AddInvoiceToOrder",
      packageName: PACKAGE_NAME,
      url: resolveComWixEcomOrdersInvoicesV1InvoicesUrl({
        protoPath: "/v1/invoices/orders/{orderId}/add-invoice",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __addInvoiceToOrder;
}

// src/ecom-orders-v1-invoice-order-invoices.meta.ts
function listInvoicesForMultipleOrders2() {
  const payload = {};
  const getRequestOptions = listInvoicesForMultipleOrders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/invoices/list-by-ids",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateInvoice2() {
  const payload = { orderId: ":orderId" };
  const getRequestOptions = generateInvoice(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/invoices/orders/{orderId}",
    pathParams: { orderId: "orderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkGenerateInvoices2() {
  const payload = {};
  const getRequestOptions = bulkGenerateInvoices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/invoices/generate",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addInvoiceToOrder2() {
  const payload = { orderId: ":orderId" };
  const getRequestOptions = addInvoiceToOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/invoices/orders/{orderId}/add-invoice",
    pathParams: { orderId: "orderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  addInvoiceToOrder2 as addInvoiceToOrder,
  bulkGenerateInvoices2 as bulkGenerateInvoices,
  generateInvoice2 as generateInvoice,
  listInvoicesForMultipleOrders2 as listInvoicesForMultipleOrders
};
