// @wix/ecom/build/es/meta.mjs
// meta.ts
import * as taxGroups from "@wix/auto_sdk_ecom_tax-groups/meta";
import * as taxRegions from "@wix/auto_sdk_ecom_tax-regions/meta";
import * as checkoutContent from "@wix/auto_sdk_ecom_checkout-content/meta";
import * as discountsCustomTrigger from "@wix/auto_sdk_ecom_discounts-custom-trigger/meta";
import * as discountRules from "@wix/auto_sdk_ecom_discount-rules/meta";
import * as orderInvoices from "@wix/auto_sdk_ecom_order-invoices/meta";
import * as recommendations from "@wix/auto_sdk_ecom_recommendations/meta";
import * as subscriptionContracts from "@wix/auto_sdk_ecom_subscription-contracts/meta";
import * as abandonedCheckouts from "@wix/auto_sdk_ecom_abandoned-checkouts/meta";
import * as backInStockNotifications from "@wix/auto_sdk_ecom_back-in-stock-notifications/meta";
import * as backInStockSettings from "@wix/auto_sdk_ecom_back-in-stock-settings/meta";
import * as cart from "@wix/auto_sdk_ecom_cart/meta";
import * as currentCart from "@wix/auto_sdk_ecom_current-cart/meta";
import * as checkout from "@wix/auto_sdk_ecom_checkout/meta";
import * as checkoutSettings from "@wix/auto_sdk_ecom_checkout-settings/meta";
import * as checkoutTemplates from "@wix/auto_sdk_ecom_checkout-templates/meta";
import * as deliveryProfile from "@wix/auto_sdk_ecom_delivery-profile/meta";
import * as draftOrders from "@wix/auto_sdk_ecom_draft-orders/meta";
import * as orderFulfillments from "@wix/auto_sdk_ecom_order-fulfillments/meta";
import * as giftVouchers from "@wix/auto_sdk_ecom_gift-vouchers/meta";
import * as localDeliveryOptions from "@wix/auto_sdk_ecom_local-delivery-options/meta";
import * as orders from "@wix/auto_sdk_ecom_orders/meta";
import * as orderBilling from "@wix/auto_sdk_ecom_order-billing/meta";
import * as orderPaymentRequests from "@wix/auto_sdk_ecom_order-payment-requests/meta";
import * as orderTransactions from "@wix/auto_sdk_ecom_order-transactions/meta";
import * as ordersSettings from "@wix/auto_sdk_ecom_orders-settings/meta";
import * as pickupLocations from "@wix/auto_sdk_ecom_pickup-locations/meta";
import * as shippingOptions from "@wix/auto_sdk_ecom_shipping-options/meta";
import * as shippoConfigurations from "@wix/auto_sdk_ecom_shippo-configurations/meta";
import * as totalsCalculator from "@wix/auto_sdk_ecom_totals-calculator/meta";
import * as cartV2 from "@wix/auto_sdk_ecom_cart-v-2/meta";
import * as currentCartV2 from "@wix/auto_sdk_ecom_current-cart-v-2/meta";
import * as currencies from "@wix/auto_sdk_ecom_currencies/meta";
import * as taxCalculation from "@wix/auto_sdk_ecom_tax-calculation/meta";
import * as deliverySolutions from "@wix/auto_sdk_ecom_delivery-solutions/meta";
import * as tipSettings from "@wix/auto_sdk_ecom_tip-settings/meta";
import * as tippableStaff from "@wix/auto_sdk_ecom_tippable-staff/meta";
import * as tips from "@wix/auto_sdk_ecom_tips/meta";
export {
  abandonedCheckouts,
  backInStockNotifications,
  backInStockSettings,
  cart,
  cartV2,
  checkout,
  checkoutContent,
  checkoutSettings,
  checkoutTemplates,
  currencies,
  currentCart,
  currentCartV2,
  deliveryProfile,
  deliverySolutions,
  discountRules,
  discountsCustomTrigger,
  draftOrders,
  giftVouchers,
  localDeliveryOptions,
  orderBilling,
  orderFulfillments,
  orderInvoices,
  orderPaymentRequests,
  orderTransactions,
  orders,
  ordersSettings,
  pickupLocations,
  recommendations,
  shippingOptions,
  shippoConfigurations,
  subscriptionContracts,
  taxCalculation,
  taxGroups,
  taxRegions,
  tipSettings,
  tippableStaff,
  tips,
  totalsCalculator
};
