// @wix/ecom/build/es/service-plugins.mjs
// service-plugins.ts
import * as taxCalculationProvider from "@wix/auto_sdk_ecom_tax-calculation-provider";
import * as giftVouchersProvider from "@wix/auto_sdk_ecom_gift-vouchers-provider";
import * as memberships from "@wix/auto_sdk_ecom_memberships";
import * as recommendationsProvider from "@wix/auto_sdk_ecom_recommendations-provider";
import * as shippingRates from "@wix/auto_sdk_ecom_shipping-rates";
import * as additionalFees from "@wix/auto_sdk_ecom_additional-fees";
import * as catalog from "@wix/auto_sdk_ecom_catalog";
import * as customTriggers from "@wix/auto_sdk_ecom_custom-triggers";
import * as discounts from "@wix/auto_sdk_ecom_discounts";
import * as inventory from "@wix/auto_sdk_ecom_inventory";
import * as paymentSettings from "@wix/auto_sdk_ecom_payment-settings";
import * as validations from "@wix/auto_sdk_ecom_validations";
export {
  additionalFees,
  catalog,
  customTriggers,
  discounts,
  giftVouchersProvider,
  inventory,
  memberships,
  paymentSettings,
  recommendationsProvider,
  shippingRates,
  taxCalculationProvider,
  validations
};
