// @wix/auto_sdk_ecom_tippable-staff/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  listTippableStaff: () => listTippableStaff2
});
module.exports = __toCommonJS(meta_exports);

// src/tips-staff-v1-staff-tippable-staff.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixTipsStaffV1TipStaffServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/tips-staff",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_tippable-staff";
function listTippableStaff(payload) {
  function __listTippableStaff({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.staff.v1.staff",
      method: "GET",
      methodFqn: "wix.tips.staff.v1.TipStaffService.ListTippableStaff",
      packageName: PACKAGE_NAME,
      url: resolveWixTipsStaffV1TipStaffServiceUrl({
        protoPath: "/v1/eligible-staff",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listTippableStaff;
}

// src/tips-staff-v1-staff-tippable-staff.meta.ts
function listTippableStaff2() {
  const payload = {};
  const getRequestOptions = listTippableStaff(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/eligible-staff",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  listTippableStaff
});
