// @wix/auto_sdk_ecom_inventory/build/es/index.typings.mjs
// src/service-plugins-error-classes.ts
var DecrementNotPossibleWixError = class extends Error {
  constructor() {
    super("DecrementNotPossible");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "DECREMENT_NOT_POSSIBLE";
    this.name = "DecrementNotPossible";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "DecrementNotPossible",
      applicationCode: "DECREMENT_NOT_POSSIBLE"
    };
  }
};
/** @hidden */
DecrementNotPossibleWixError.__type = "wix_spi_error";
var IncrementNotPossibleWixError = class extends Error {
  constructor() {
    super("IncrementNotPossible");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "INCREMENT_NOT_POSSIBLE";
    this.name = "IncrementNotPossible";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "IncrementNotPossible",
      applicationCode: "INCREMENT_NOT_POSSIBLE"
    };
  }
};
/** @hidden */
IncrementNotPossibleWixError.__type = "wix_spi_error";

// src/service-plugins-types.ts
var Reason = /* @__PURE__ */ ((Reason2) => {
  Reason2["UNKNOWN_REASON"] = "UNKNOWN_REASON";
  Reason2["ORDER_PLACED"] = "ORDER_PLACED";
  Reason2["ORDER_PAID"] = "ORDER_PAID";
  Reason2["ORDER_CANCELED"] = "ORDER_CANCELED";
  Reason2["ORDER_REFUNDED"] = "ORDER_REFUNDED";
  Reason2["ORDER_EDITED"] = "ORDER_EDITED";
  Reason2["ORDER_REJECTED"] = "ORDER_REJECTED";
  return Reason2;
})(Reason || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});
export {
  DecrementNotPossibleWixError,
  IdentityType,
  IncrementNotPossibleWixError,
  Reason
};
