// @wix/error-handler/dist/cjs/state/CommonErrorsMapState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CommonErrorsMapState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _getErrorCodesMap = require("./getErrorCodesMap");
class CommonErrorsMapState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "translations", void 0);
    (0, _defineProperty2.default)(this, "errorMonitor", void 0);
    (0, _defineProperty2.default)(this, "webWindow", void 0);
    (0, _defineProperty2.default)(this, "getErrorCodesMap", void 0);
    (0, _defineProperty2.default)(this, "_initPromise", void 0);
    (0, _defineProperty2.default)(this, "commonErrors", void 0);
    this.translations = params.translations;
    this.errorMonitor = params.errorMonitor;
    this.webWindow = params.webWindow;
    this.getErrorCodesMap = params.getErrorCodesMap ?? _getErrorCodesMap.getErrorCodesMap;
    this._initPromise = this._init();
  }
  async getInitPromise() {
    const error = await this._initPromise;
    if (error) {
      throw error;
    }
  }
  async _init() {
    try {
      const {
        setCommonErrors
      } = await this.getErrorCodesMap();
      await setCommonErrors(this);
      return null;
    } catch (e) {
      return e;
    }
  }
}
exports.CommonErrorsMapState = CommonErrorsMapState;
