// @wix/auto_sdk_ecom_shippo-configurations/build/es/meta.mjs
// src/ecom-v1-shippo-configuration-shippo-configurations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEcomDeliveryV1ShippoServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/shippo-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/shippo-configurations",
        destPath: "/v1/shippo-configurations"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_shippo-configurations";
function createShippoConfiguration(payload) {
  function __createShippoConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "shippoConfiguration.createdDate" },
          { path: "shippoConfiguration.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "shippoConfiguration.shipFromAddress.geocode.latitude" },
          { path: "shippoConfiguration.shipFromAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.shippo_configuration",
      method: "POST",
      methodFqn: "wix.ecom.delivery.v1.ShippoService.CreateShippoConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryV1ShippoServiceUrl({
        protoPath: "/v1/shippo-configurations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "shippoConfiguration.createdDate" },
            { path: "shippoConfiguration.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "shippoConfiguration.shipFromAddress.geocode.latitude" },
            { path: "shippoConfiguration.shipFromAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createShippoConfiguration;
}
function getShippoConfiguration(payload) {
  function __getShippoConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.shippo_configuration",
      method: "GET",
      methodFqn: "wix.ecom.delivery.v1.ShippoService.GetShippoConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryV1ShippoServiceUrl({
        protoPath: "/v1/shippo-configurations/{shippoConfigurationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "shippoConfiguration.createdDate" },
            { path: "shippoConfiguration.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "shippoConfiguration.shipFromAddress.geocode.latitude" },
            { path: "shippoConfiguration.shipFromAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getShippoConfiguration;
}
function updateShippoConfiguration(payload) {
  function __updateShippoConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "shippoConfiguration.createdDate" },
          { path: "shippoConfiguration.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "shippoConfiguration.shipFromAddress.geocode.latitude" },
          { path: "shippoConfiguration.shipFromAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.shippo_configuration",
      method: "PATCH",
      methodFqn: "wix.ecom.delivery.v1.ShippoService.UpdateShippoConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryV1ShippoServiceUrl({
        protoPath: "/v1/shippo-configurations/{shippoConfiguration.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "shippoConfiguration.createdDate" },
            { path: "shippoConfiguration.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "shippoConfiguration.shipFromAddress.geocode.latitude" },
            { path: "shippoConfiguration.shipFromAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateShippoConfiguration;
}
function deleteShippoConfiguration(payload) {
  function __deleteShippoConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.shippo_configuration",
      method: "DELETE",
      methodFqn: "wix.ecom.delivery.v1.ShippoService.DeleteShippoConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryV1ShippoServiceUrl({
        protoPath: "/v1/shippo-configurations/{shippoConfigurationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteShippoConfiguration;
}
function queryShippoConfigurations(payload) {
  function __queryShippoConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.shippo_configuration",
      method: "GET",
      methodFqn: "wix.ecom.delivery.v1.ShippoService.QueryShippoConfigurations",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryV1ShippoServiceUrl({
        protoPath: "/v1/shippo-configurations/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "shippoConfigurations.createdDate" },
            { path: "shippoConfigurations.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "shippoConfigurations.shipFromAddress.geocode.latitude" },
            {
              path: "shippoConfigurations.shipFromAddress.geocode.longitude"
            }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixEcomDeliveryV1ShippoServiceUrl({
            protoPath: "/v1/shippo-configurations/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryShippoConfigurations;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.shippo_configuration",
      method: "POST",
      methodFqn: "wix.ecom.delivery.v1.ShippoService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryV1ShippoServiceUrl({
        protoPath: "/v1/shippo-configurations/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "shippoConfiguration.createdDate" },
            { path: "shippoConfiguration.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "shippoConfiguration.shipFromAddress.geocode.latitude" },
            { path: "shippoConfiguration.shipFromAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function addDeliveryRegion(payload) {
  function __addDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.shippo_configuration",
      method: "POST",
      methodFqn: "wix.ecom.delivery.v1.ShippoService.AddDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryV1ShippoServiceUrl({
        protoPath: "/v1/shippo-configurations/{shippoConfigurationId}/delivery-regions",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "shippoConfiguration.createdDate" },
            { path: "shippoConfiguration.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "shippoConfiguration.shipFromAddress.geocode.latitude" },
            { path: "shippoConfiguration.shipFromAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addDeliveryRegion;
}
function removeDeliveryRegion(payload) {
  function __removeDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.shippo_configuration",
      method: "DELETE",
      methodFqn: "wix.ecom.delivery.v1.ShippoService.RemoveDeliveryRegion",
      packageName: PACKAGE_NAME,
      url: resolveWixEcomDeliveryV1ShippoServiceUrl({
        protoPath: "/v1/shippo-configurations/{shippoConfigurationId}/delivery-regions/{deliveryRegionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "shippoConfiguration.createdDate" },
            { path: "shippoConfiguration.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "shippoConfiguration.shipFromAddress.geocode.latitude" },
            { path: "shippoConfiguration.shipFromAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removeDeliveryRegion;
}

// src/ecom-v1-shippo-configuration-shippo-configurations.meta.ts
function createShippoConfiguration2() {
  const payload = {};
  const getRequestOptions = createShippoConfiguration(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/shippo-configurations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getShippoConfiguration2() {
  const payload = { shippoConfigurationId: ":shippoConfigurationId" };
  const getRequestOptions = getShippoConfiguration(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/shippo-configurations/{shippoConfigurationId}",
    pathParams: { shippoConfigurationId: "shippoConfigurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateShippoConfiguration2() {
  const payload = {
    shippoConfiguration: { id: ":shippoConfigurationId" }
  };
  const getRequestOptions = updateShippoConfiguration(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/shippo-configurations/{shippoConfiguration.id}",
    pathParams: { shippoConfigurationId: "shippoConfigurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteShippoConfiguration2() {
  const payload = { shippoConfigurationId: ":shippoConfigurationId" };
  const getRequestOptions = deleteShippoConfiguration(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/shippo-configurations/{shippoConfigurationId}",
    pathParams: { shippoConfigurationId: "shippoConfigurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryShippoConfigurations2() {
  const payload = {};
  const getRequestOptions = queryShippoConfigurations(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/shippo-configurations/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/shippo-configurations/{id}/update-extended-fields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addDeliveryRegion2() {
  const payload = { shippoConfigurationId: ":shippoConfigurationId" };
  const getRequestOptions = addDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/shippo-configurations/{shippoConfigurationId}/delivery-regions",
    pathParams: { shippoConfigurationId: "shippoConfigurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeDeliveryRegion2() {
  const payload = {
    shippoConfigurationId: ":shippoConfigurationId",
    deliveryRegionId: ":deliveryRegionId"
  };
  const getRequestOptions = removeDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/shippo-configurations/{shippoConfigurationId}/delivery-regions/{deliveryRegionId}",
    pathParams: {
      shippoConfigurationId: "shippoConfigurationId",
      deliveryRegionId: "deliveryRegionId"
    },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  addDeliveryRegion2 as addDeliveryRegion,
  createShippoConfiguration2 as createShippoConfiguration,
  deleteShippoConfiguration2 as deleteShippoConfiguration,
  getShippoConfiguration2 as getShippoConfiguration,
  queryShippoConfigurations2 as queryShippoConfigurations,
  removeDeliveryRegion2 as removeDeliveryRegion,
  updateExtendedFields2 as updateExtendedFields,
  updateShippoConfiguration2 as updateShippoConfiguration
};
