// @wix/auto_sdk_ecom_tax-calculation/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  calculateTax: () => calculateTax2,
  listTaxCalculators: () => listTaxCalculators2
});
module.exports = __toCommonJS(meta_exports);

// src/gateways-billing-v1-tax-calculation-gateway-tax-calculation.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBillingV1TaxCalculationUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/billing/v1/calculate-tax",
        destPath: "/v1/calculate-tax"
      },
      {
        srcPath: "/billing/v1/list-tax-calculators",
        destPath: "/v1/list-tax-calculators"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/billing/v1/calculate-tax",
        destPath: "/v1/calculate-tax"
      },
      {
        srcPath: "/billing/v1/list-tax-calculators",
        destPath: "/v1/list-tax-calculators"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_tax-calculation";
function calculateTax(payload) {
  function __calculateTax({ host }) {
    const metadata = {
      entityFqdn: "wix.gateways.billing.v1.tax_calculation_gateway",
      method: "POST",
      methodFqn: "wix.billing.v1.TaxCalculation.CalculateTax",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxCalculationUrl({
        protoPath: "/v1/calculate-tax",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __calculateTax;
}
function listTaxCalculators(payload) {
  function __listTaxCalculators({ host }) {
    const metadata = {
      entityFqdn: "wix.gateways.billing.v1.tax_calculation_gateway",
      method: "GET",
      methodFqn: "wix.billing.v1.TaxCalculation.ListTaxCalculators",
      packageName: PACKAGE_NAME,
      url: resolveWixBillingV1TaxCalculationUrl({
        protoPath: "/v1/list-tax-calculators",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listTaxCalculators;
}

// src/gateways-billing-v1-tax-calculation-gateway-tax-calculation.meta.ts
function calculateTax2() {
  const payload = {};
  const getRequestOptions = calculateTax(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/calculate-tax",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listTaxCalculators2() {
  const payload = {};
  const getRequestOptions = listTaxCalculators(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/list-tax-calculators",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  calculateTax,
  listTaxCalculators
});
