// @wix/headless-stores/dist/enums/social-platform-enums.js
/**
 * Business logic enums for social sharing platforms (headless layer)
 */
/**
 * Enum for social sharing platforms (business identifiers)
 */
export var SocialPlatform;
(function (SocialPlatform) {
    SocialPlatform["FACEBOOK"] = "facebook";
    SocialPlatform["TWITTER"] = "twitter";
    SocialPlatform["LINKEDIN"] = "linkedin";
    SocialPlatform["WHATSAPP"] = "whatsapp";
    SocialPlatform["EMAIL"] = "email";
    SocialPlatform["CLIPBOARD"] = "clipboard";
    SocialPlatform["NATIVE"] = "native";
})(SocialPlatform || (SocialPlatform = {}));
/**
 * Enum for social platform share URLs (business logic)
 */
export var SocialPlatformShareUrl;
(function (SocialPlatformShareUrl) {
    SocialPlatformShareUrl["FACEBOOK"] = "https://www.facebook.com/sharer/sharer.php";
    SocialPlatformShareUrl["TWITTER"] = "https://twitter.com/intent/tweet";
    SocialPlatformShareUrl["LINKEDIN"] = "https://www.linkedin.com/sharing/share-offsite/";
    SocialPlatformShareUrl["WHATSAPP"] = "https://wa.me/";
    SocialPlatformShareUrl["EMAIL"] = "mailto:";
})(SocialPlatformShareUrl || (SocialPlatformShareUrl = {}));
